

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>19. Model from Template protocol &mdash; Scipion 3.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../../../../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../../../" src="../../../../../_static/documentation_options.js"></script>
        <script data-url_root="../../../../../" id="documentation_options" src="../../../../../_static/documentation_options.js"></script>
        <script src="../../../../../_static/jquery.js"></script>
        <script src="../../../../../_static/underscore.js"></script>
        <script src="../../../../../_static/doctools.js"></script>
    
    <script type="text/javascript" src="../../../../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/banner.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/custom.css" type="text/css" />
    <link rel="index" title="Index" href="../../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../../search.html" />
    <link rel="next" title="20. Phenix EMRinger protocol" href="a120-emRingerProtocol.html" />
    <link rel="prev" title="18. Local MonoRes protocol" href="a102-localMonoRes.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../../../index.html" class="icon icon-home"> Scipion
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/how-to-install.html">Installing Scipion</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">User documentation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../user-documentation.html#tutorials">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#single-particle-tutorials">Single particle tutorials</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#tomography-tutorials">Tomography tutorials</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../../user-documentation.html#model-building">Model Building</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="index.html">Model Building by Homology</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="index.html#contents">Contents</a></li>
<li class="toctree-l4 current"><a class="reference internal" href="index.html#appendices">Appendices</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../Alphafold/index.html">Build initial model with Alphafold</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html#processing-how-to-s">Processing How To’s</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities.html">On-the-fly processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/customize-report.html">Summary Reports</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/scipion-API.html">Scipion API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/plugins-API.html">Plugins API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/xmipp-API.html">Xmipp API</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../misc/contact-us.html">Contact Us</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Licences</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../license.html">Licenses</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../../index.html">Scipion</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../../index.html">Docs</a> &raquo;</li>
        
          <li><a href="../../../user-documentation.html">Graphical interface manuals</a> &raquo;</li>
        
          <li><a href="index.html">Model Building Tutorial</a> &raquo;</li>
        
      <li><span class="section-number">19. </span>Model from Template protocol</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../../../../../_sources/docs/user/tutorials/modelBuilding/Homology/a110-modelFromTemplate.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="model-from-template-protocol">
<span id="app-modelfromtemplate"></span><h1><span class="section-number">19. </span>Model from Template protocol<a class="headerlink" href="#model-from-template-protocol" title="Permalink to this headline">¶</a></h1>
<div class="line-block">
<div class="line">Protocol designed to obtain a structure model for a target sequence in <em>Scipion</em>. Target structure is predicted by sequence homology using <em>Modeller</em> <span id="id1">[<a class="reference internal" href="z010-references.html#id13" title="A. Sali and T. L. Blundell. Comparative protein modelling by satisfaction of spatial restraints. J. Mol. Biol., 234(3):779–815, Dec 1993.">Sali and Blundell, 1993</a>]</span> web service in <em>ChimeraX</em>.</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">WARNING:</span></code> Working with <em>Modeller</em> requires a license key, which can be requested
free of charge for academic users. Try to have this license key before
starting the protocol execution.</div>
</div>
<ul>
<li><div class="line-block">
<div class="line">Requirements to run this protocol and visualize results:</div>
</div>
<ul>
<li><div class="line-block">
<div class="line"><em>Scipion</em> plugin: <strong>scipion-em</strong></div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Scipion</em> plugin: <strong>scipion-em-chimera</strong></div>
</div>
</li>
<li><div class="line-block">
<div class="line">Multiple sequence alignment tools: <strong>Clustal Omega</strong>, <strong>MUSCLE</strong></div>
</div>
</li>
</ul>
</li>
<li><div class="line-block">
<div class="line"><em>Scipion</em> menu: <em>Model building -&gt; Initial model</em> (<a class="reference internal" href="#model-building-app-protocol-seqhomology-1"><span class="std std-numref">Fig. 19.1</span></a> (A))</div>
</div>
<div class="figure align-center" id="model-building-app-protocol-seqhomology-1">
<a class="reference internal image-reference" href="../../../../../_images/Fig111.svg"><img alt="Protocol **chimerax-model from template**. A: Protocol location in *Scipion* menu. B: Protocol form: Option “template available”. C: Protocol form: Option “looking for template”." src="../../../../../_images/Fig111.svg" width="90.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 19.1 </span><span class="caption-text">Protocol <strong>chimerax-model from template</strong>. A: Protocol location in <em>Scipion</em> menu. B: Protocol form: Option “template available”. C: Protocol form: Option “looking for template”.</span><a class="headerlink" href="#model-building-app-protocol-seqhomology-1" title="Permalink to this image">¶</a></p>
</div>
</li>
<li><div class="line-block">
<div class="line">Protocol form parameters (<a class="reference internal" href="#model-building-app-protocol-seqhomology-1"><span class="std std-numref">Fig. 19.1</span></a> (B)):</div>
</div>
<ul>
<li><div class="line-block">
<div class="line"><em>Input</em> section</div>
</div>
<ul>
<li><div class="line-block">
<div class="line"><em>Do you already have a template?</em>: Select <em>“Yes”</em> if you have found your <em>template</em> in a previous similarity searching step. Select <em>“No”</em> if you do not have any <em>template</em> to start the homology modeling and you would like to search for one.</div>
</div>
</li>
<li><div class="line-block">
<div class="line">Option <em>“Yes”</em> (<a class="reference internal" href="#model-building-app-protocol-seqhomology-1"><span class="std std-numref">Fig. 19.1</span></a> (B))</div>
</div>
<ul>
<li><div class="line-block">
<div class="line"><em>Atomic structure used as template</em>: Atomic structure previously downloaded in <em>Scipion</em>. This structure was selected by sequence homology, i.e. by looking for the structurally characterized sequence more similar (with higher identity) to the target sequence.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Chain</em>: Specific monomer of the macromolecule that has to be used as structure <em>template</em> of the <em>target sequence</em>. Use the wizard on the right side of <em>Chain</em> parameter to select that chain.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Target sequence</em>: Sequence previously downloaded in <em>Scipion</em>. This sequence has to be modeled following the structure skeleton of the selected <em>template</em>.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Options to improve the alignment</em>: <span class="math">Modeller</span> provides structural models of the <em>target sequence</em> based on a sequence alignment, in which at least sequences of <em>template</em> and <em>target</em> have to be included. Three options can be considered to improve this alignment:</div>
</div>
<ol class="arabic">
<li><div class="line-block">
<div class="line"><em>None</em>: No more sequences are going to be included in
the alignment except <em>model</em> and <em>target</em> sequences.</div>
<div class="line-block">
<div class="line">Correlative param:</div>
</div>
<div class="line">*** <em>Alignment tool for two sequences</em>: Select one of
the three available alignment methods, <em>Bio.parirwise2</em>
(by default), <em>Clustal Omega</em>, <em>MUSCLE</em>.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Additional sequences to align</em> if you want to perform
a multiple sequence alignment (MSA) adding other sequences
already downloaded in <em>Scipion</em>. Additional sequences, others
than <em>template</em> and <em>target</em> sequences, are required to
accomplish this MSA.</div>
<div class="line">Correlative params:</div>
<div class="line">*** <em>Other sequences to align</em>: Box to complete with
the additional sequences used to perform the multiple
sequence alignment. All of them were previously
downloaded in <em>Scipion</em>.</div>
<div class="line">*** <em>Multiple alignment tool</em>: Select between
<em>Clustal Omega</em> and <em>MUSCLE</em> methods.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Provide your own sequence alignment</em>: If you want to
include other sequences in the alignment by providing
your own sequence alignment.</div>
<div class="line">Correlative param:</div>
<div class="line">*** <em>Sequence alignment input</em>: Complete this box
with the help of the right side browser including the
sequence alignment file that you already have saved in
your computer. Different <a class="reference external" href="https://www.cgl.ucsf.edu/chimerax/docs/user/commands/open.html">alignment formats</a> are
available. An example of alignment in <em>fasta</em> format can be seen
below (Use case 3).</div>
</div>
</li>
</ol>
</li>
<li><div class="line-block">
<div class="line"><em>Additional target sequence to include?</em>: Select <em>“Yes”</em> if you’d like to obtain a multimer <em>model</em> by using two <em>target</em> sequences and the same multimer <em>template</em>. The params to complete the option <em>“Yes”</em> are identical to those already shown, with the exception of the <em>Atomic structure used as template</em>, already completed. However, no one of those params will appear in case you select <em>“No”</em> in order to obtain a <em>model</em> by using only one <em>target</em> sequence.</div>
</div>
</li>
</ul>
</li>
<li><div class="line-block">
<div class="line">Option <em>“No”</em> (<a class="reference internal" href="#model-building-app-protocol-seqhomology-1"><span class="std std-numref">Fig. 19.1</span></a> (C))</div>
</div>
<ul>
<li><div class="line-block">
<div class="line"><em>Target sequence</em>: Sequence previously downloaded in <em>Scipion</em>. This sequence has to be modeled following the structure skeleton of the <em>template</em> that you are going to select among the retrieved entries found by the similarity searching tool.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Protein sequence database</em>: Select one of the two suggested protein sequence databases, PDB and NR. Press the <em>“?”</em> symbol on the right to see the meaning of each one. Remark that the NR database allows you to get entries with, as well as without, atomic structure associated. These ones, which do not provide <em>templates</em>, could be useful to build a better sequence alignment.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Similarity matrix</em>: Select one of the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/blast/html/sub_matrix.html">“substitution matrix”</a> to assign a score to any couple of residues in the alignment.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>cutoff evalue</em>: Maximum statistic value required to include a retrieved element in the hit list.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Maximum number of sequences</em> that you’d like to retrieve from the database.</div>
</div>
</li>
</ul>
</li>
</ul>
</li>
<li><div class="line-block">
<div class="line"><em>Help</em> section</div>
</div>
<div class="line-block">
<div class="line">Follow this section steps to run <span class="math">Modeller</span> via web service in <em>ChimeraX</em> and to select and save one of the retrieved models in <em>Scipion</em> framework.</div>
</div>
</li>
</ul>
</li>
<li><div class="line-block">
<div class="line">Protocol execution:</div>
</div>
<div class="line-block">
<div class="line">Adding specific template-target label is recommended in <em>Run name</em>
section, at the form top. To add the label, open the protocol form,
press the pencil symbol on the right side of <em>Run name</em> box,
complete the label in the new opened window, press OK and, finally,
close the protocol. This label will be shown in the output summary
content (see below). If you want to run again this protocol, do not
forget to set to <em>Restart</em> the <em>Run mode</em>.</div>
</div>
<div class="line-block">
<div class="line">Press the <em>Execute</em> red button at the form bottom.</div>
</div>
<div class="line-block">
<div class="line">Several <em>ChimeraX</em> windows will be opened after executing the protocol with
different contents according to the distinct form param options.
Although we are going to detail some of them through several use
cases (see below), designed to ilustrate different applications of
this protocol, as well as the procedure to follow in each case, in
general we can predict the opening of <em>ChimeraX</em> graphics window and a
sequence alignment window. Usually, in both windows the <em>template</em>
sequence is green highlighted (see an example of these windows in <a class="reference internal" href="070-movingFromSequence.html#model-building-chimera-alignment"><span class="std std-numref">Fig. 9.6</span></a>). Main steps to follow ahead are:</div>
</div>
<ul>
<li><div class="line-block">
<div class="line">Ask for model(s) to <span class="math">Modeller</span> by selecting <em>Tools -&gt; Sequence -&gt; Modeller Comparative</em> in the main menu of <em>ChimeraX</em> graphics window.</div>
</div>
</li>
<li><div class="line-block">
<div class="line">Complete the new window opened for <em>Modeller Comparative</em> with the sequence alignment that includes the <em>template</em> and with the <em>target(s)</em> sequence(s), <span class="math">Modeller</span> license key, multichain model, number of models retrieved by <em>Modeller</em>, and <em>Advanced</em> options like the building of models with hydrogens, as well as <em>model</em> inclusion of heteroatoms or water molecules. An example of completed <span class="math">Modeller</span> window can be observed in <a class="reference internal" href="070-movingFromSequence.html#model-building-modeller"><span class="std std-numref">Fig. 9.7</span></a> (A). By pressing <em>OK</em> the computation starts. The status of the job can be checked in the lower left corner of <em>ChimeraX</em> graphics window.</div>
</div>
</li>
<li><div class="line-block">
<div class="line">After a while a new panel window will show the retrieved models of the <em>target</em> sequence (<a class="reference internal" href="070-movingFromSequence.html#model-building-modeller"><span class="std std-numref">Fig. 9.7</span></a> (B)). Two statistics assess these models: <em>GA341</em>, statistical potentials derived-score, and <em>zDOPE</em>, normalized Discrete Optimized Protein Energy, atomic distance depending-score. Reliable models show <em>GA341</em> values higher than 0.7, and negative <em>zDOPE</em> values correspond to better models. Retrieved models can be checked in <em>ChimeraX Tools -&gt; Models</em>. One of them should be selected (<a class="reference internal" href="070-movingFromSequence.html#model-building-modeller"><span class="std std-numref">Fig. 9.7</span></a> (C)).</div>
</div>
</li>
<li><div class="line-block">
<div class="line">Rename the selected model, for example <em>#n_initial</em> to <em>#n_final</em> with <em>ChimeraX</em> command line:</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rename</span> <span class="c1">#n_initial id #n_final</span>
</pre></div>
</div>
</li>
<li><div class="line-block">
<div class="line">Save the retrieved model selected according to the new model number in the <em>Scipion</em> track system (<em>#n_final</em>) shown in <em>ChimeraX Tools -&gt; Models</em> by writing in command line:</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">scipionwrite</span> <span class="c1">#n_final prefix user_defined_name_</span>
</pre></div>
</div>
</li>
</ul>
</li>
<li><div class="line-block">
<div class="line">Visualization of protocol results:</div>
</div>
<div class="line-block">
<div class="line">After executing the protocol, press <em>Analyze Results</em> and <em>ChimeraX</em> graphics window will be opened by default. Atomic structures are referred to the origin of coordinates in <em>ChimeraX</em>. To show the relative position of the atomic structure, the three coordinate axes are represented; X axis (red), Y axis (yellow), and Z axis (blue) (<a class="reference internal" href="a020-chimeraOperate.html#model-building-app-protocol-volume-3"><span class="std std-numref">Fig. 5.3</span></a>). Coordinate axes and selected atomic structure <em>model</em> are model numbers <em>#1</em> and <em>#2</em>, respectively, in <em>ChimeraX Models</em> panel if only one structure has been saved.</div>
</div>
</li>
<li><div class="line-block">
<div class="line">Summary content:</div>
</div>
<ul>
<li><div class="line-block">
<div class="line">Protocol output (below <em>Scipion</em> framework):</div>
<div class="line"><em>chimerax - model from template -&gt; name of the new atomic
structure</em>;</div>
<div class="line"><em>AtomStruct (pseudoatoms=True/ False, volume=True/ False)</em>.</div>
<div class="line">Pseudoatoms is set to <em>True</em> when the structure is made of
pseudoatoms instead of atoms. Volume is set to <em>True</em> when an
electron density map is associated to the atomic structure.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>SUMMARY</em> box:</div>
<div class="line">Produced files:</div>
<div class="line">we have some result</div>
</div>
</li>
</ul>
</li>
</ul>
<div class="section" id="use-cases">
<h2><span class="section-number">19.1. </span>USE CASES<a class="headerlink" href="#use-cases" title="Permalink to this headline">¶</a></h2>
<ul>
<li><div class="line-block">
<div class="line"><strong>Use Case 1:</strong>  <em>Input atomic structure as template, 1 target sequence, Option “None” to improve the alignment</em></div>
</div>
<div class="line-block">
<div class="line">Aim: To model a <em>target</em> sequence using one chain of a homologous atomic structure as <em>template</em>, using only the sequences of <em>target</em> and <em>template</em> in the sequence alignment.</div>
</div>
<div class="figure align-center" id="model-building-app-protocol-seqhomology-2">
<a class="reference internal image-reference" href="../../../../../_images/Fig305.svg"><img alt="(A) Protocol form of **chimerax-model from template**. (B) *ChimeraX* view of the *template* atomic structure, highlighted in green the *chain B* selected to perform the modeling. (C) *ChimeraX* sequence view panel showing the sequence alignment between the *template* *chain B* sequence, greeen highlighted, and the *target* sequence. (D) Upper part of the *ChimeraX Modeller comparative* panel showing the sequence alignment selected, shown in (C), and the selected *target* sequence." src="../../../../../_images/Fig305.svg" width="95.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 19.2 </span><span class="caption-text"><em>(A)</em> Protocol form of <strong>chimerax-model from template</strong>. (B) <em>ChimeraX</em> view of the <em>template</em> atomic structure, highlighted in green the <em>chain B</em> selected to perform the modeling. (C) <em>ChimeraX</em> sequence view panel showing the sequence alignment between the <em>template</em> <em>chain B</em> sequence, greeen highlighted, and the <em>target</em> sequence. (D) Upper part of the <em>ChimeraX Modeller comparative</em> panel showing the sequence alignment selected, shown in (C), and the selected <em>target</em> sequence.</span><a class="headerlink" href="#model-building-app-protocol-seqhomology-2" title="Permalink to this image">¶</a></p>
</div>
<div class="line-block">
<div class="line">Protocol execution: Complete the protocol form as indicated in <a class="reference internal" href="#model-building-app-protocol-seqhomology-2"><span class="std std-numref">Fig. 19.2</span></a> (A). Follow the general procedure shown above (Protocol execution section). Windows (B) and (C) will appear. Open and complete the <em>Modeller Comparative</em> panel as indicated in (D) and wait for a while. After getting the retrieved models, if you want to select, for example, the <em>target</em> model <em>#3.2</em>, write in <em>ChimeraX</em> command line:</div>
</div>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rename</span> <span class="c1">#3.2 id #4</span>
<span class="n">scipionwrite</span> <span class="c1">#4 prefix model_3_2_</span>
</pre></div>
</div>
</div></blockquote>
<div class="line-block">
<div class="line">And <em>ChimeraX Quit</em> to close the protocol. Visualize your results.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>Use Case 2:</strong> <em>Input atomic structure as template, 2 target sequences, Option “Additional sequences to align” to improve the alignment, multichain modeling option in Modeller</em></div>
</div>
<div class="line-block">
<div class="line">Aim: To model simultaneously two <em>target</em> sequences to obtain a multichain model using two chains of a homologous atomic structure as <em>templates</em>, using other additional sequences than <em>target</em> and <em>template</em> in the sequence alignment.</div>
</div>
<div class="figure align-left" id="model-building-app-protocol-seqhomology-3">
<a class="reference internal image-reference" href="../../../../../_images/Fig306.svg"><img alt="(A) Protocol form of **chimerax-model from template**. Remark that two additional sequences other than *template* and *target* sequences have been selected to improve both alignments. (B) Upper part of the *ChimeraX Modeller comparative* panel showing the two sequence alignment selected, shown in (D) and (E), and the two selected *target* sequences.(C) *ChimeraX* view of the *template* atomic structure, highlighted in yellow the *chain A* and in green the *chain B* selected to perform the modeling. (D) *ChimeraX* sequence view panel showing the sequence alignment between the *template* *chain A* sequence, greeen highlighted, the *target* sequence (*Q15116*) and two more additional sequences. (E) *ChimeraX* sequence view panel showing the sequence alignment between the *template* *chain B* sequence, greeen highlighted, the *target* sequence (*Q12LC89*) and two more additional sequences." src="../../../../../_images/Fig306.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 19.3 </span><span class="caption-text"><em>(A)</em> Protocol form of <strong>chimerax-model from template</strong>. Remark that two additional sequences other than <em>template</em> and <em>target</em> sequences have been selected to improve both alignments. (B) Upper part of the <em>ChimeraX Modeller comparative</em> panel showing the two sequence alignment selected, shown in (D) and (E), and the two selected <em>target</em> sequences.(C) <em>ChimeraX</em> view of the <em>template</em> atomic structure, highlighted in yellow the <em>chain A</em> and in green the <em>chain B</em> selected to perform the modeling. (D) <em>ChimeraX</em> sequence view panel showing the sequence alignment between the <em>template</em> <em>chain A</em> sequence, greeen highlighted, the <em>target</em> sequence (<em>Q15116</em>) and two more additional sequences. (E) <em>ChimeraX</em> sequence view panel showing the sequence alignment between the <em>template</em> <em>chain B</em> sequence, greeen highlighted, the <em>target</em> sequence (<em>Q12LC89</em>) and two more additional sequences.</span><a class="headerlink" href="#model-building-app-protocol-seqhomology-3" title="Permalink to this image">¶</a></p>
</div>
<div class="line-block">
<div class="line">Protocol execution: Complete the protocol form as indicated in <a class="reference internal" href="#model-building-app-protocol-seqhomology-3"><span class="std std-numref">Fig. 19.3</span></a> (A).
Follow the general procedure shown above (Protocol execution
section). Windows (C), (D) and (E) will appear. Open and complete
the <em>Modeller Comparative</em> panel as indicated in (B) and wait for a
while. After getting the retrieved models, if you want to select,
for example, the <em>target</em> model <em>#3.2</em>, write in <em>ChimeraX</em> command line:</div>
</div>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rename</span> <span class="c1">#3.2 id #4</span>
<span class="n">scipionwrite</span> <span class="c1">#4 prefix model_3_2_</span>
</pre></div>
</div>
</div></blockquote>
<div class="line-block">
<div class="line">And <em>ChimeraX Quit</em> to close the protocol. Visualize your results.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>Use Case 3:</strong> <em>Input atomic structure as template, 2 target sequences, Option “Provide your own sequence alignment” to improve the alignment, multichain modeling option in Modeller</em></div>
</div>
<div class="line-block">
<div class="line">Aim: To model simultaneously two <em>target</em> sequences to obtain a multichain model using two chains of a homologous atomic structure as <em>templates</em>, using your own MSA.</div>
</div>
<div class="figure align-left" id="model-building-app-protocol-seqhomology-4">
<a class="reference internal image-reference" href="../../../../../_images/Fig307.svg"><img alt="(A) Protocol form of **chimerax-model from template**. Remark that your own sequence alignment containing the sequences of *template* and *target*, among others, have been selected. (B) Example of sequence alignment in *fasta* format (file “aligned_2.fasta”) that includes the *template* *chain B* sequence, the *target* sequence (*Q12LC89*) and two more additional sequences." src="../../../../../_images/Fig307.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 19.4 </span><span class="caption-text"><em>(A)</em> Protocol form of <strong>chimerax-model from template</strong>. Remark that your own sequence alignment containing the sequences of <em>template</em> and <em>target</em>, among others, have been selected. (B) Example of sequence alignment in <em>fasta</em> format (file “aligned_2.fasta”) that includes the <em>template</em> <em>chain B</em> sequence, the <em>target</em> sequence (<em>Q12LC89</em>) and two more additional sequences.</span><a class="headerlink" href="#model-building-app-protocol-seqhomology-4" title="Permalink to this image">¶</a></p>
</div>
<div class="line-block">
<div class="line">Protocol execution: Complete the protocol form as indicated in <a class="reference internal" href="#model-building-app-protocol-seqhomology-4"><span class="std std-numref">Fig. 19.4</span></a> (A).
Follow the general procedure shown above (Protocol execution
section). Remark that you already have a MSA file
for each <em>target</em> saved in your computer. An example of MSA can be seen in
(B). Windows (C), (D) and (E) of previous . Open and complete the
<em>Modeller Comparative</em> panel as indicated in <a class="reference internal" href="#model-building-app-protocol-seqhomology-4"><span class="std std-numref">Fig. 19.4</span></a> (B) and wait for a
while. After getting the retrieved models, if you want to select,
for example, the <em>target</em> model <em>#3.2</em>, write in the command line:</div>
</div>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rename</span> <span class="c1">#3.2 id #4</span>
<span class="n">scipionwrite</span> <span class="c1">#4 prefix model_3_2_</span>
</pre></div>
</div>
</div></blockquote>
<div class="line-block">
<div class="line">And <em>ChimeraX Quit</em> to close the protocol. Visualize your results.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>Use Case 4:</strong> <em>Input 1 target sequence, PDB searching database</em></div>
</div>
<div class="line-block">
<div class="line">Aim: To model a <em>target</em> sequence without previous information of a possible atomic structure <em>template</em>.</div>
</div>
<div class="figure align-left" id="model-building-app-protocol-seqhomology-5">
<a class="reference internal image-reference" href="../../../../../_images/Fig308.svg"><img alt="(A) Protocol form of **chimerax-model from template** that includes the *target* sequence that we’d like to model. (B) Upper part of the *ChimeraX Modeller comparative* panel showing the sequence alignment and the *target* sequence selected. (B) *ChimeraX* graphics window empty before opening the *template* atomic structure. (D) *ChimeraX* sequence view panel showing the *target* sequence. (E) *ChimeraX* BlastProtein panel showing the retrieved results from PDB database." src="../../../../../_images/Fig308.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 19.5 </span><span class="caption-text"><em>(A)</em> Protocol form of <strong>chimerax-model from template</strong> that includes the <em>target</em> sequence that we’d like to model. (B) Upper part of the <em>ChimeraX Modeller comparative</em> panel showing the sequence alignment and the <em>target</em> sequence selected. (B) <em>ChimeraX</em> graphics window empty before opening the <em>template</em> atomic structure. (D) <em>ChimeraX</em> sequence view panel showing the <em>target</em> sequence. (E) <em>ChimeraX</em> BlastProtein panel showing the retrieved results from PDB database.</span><a class="headerlink" href="#model-building-app-protocol-seqhomology-5" title="Permalink to this image">¶</a></p>
</div>
<div class="line-block">
<div class="line">Protocol execution: Complete the protocol form as indicated in <a class="reference internal" href="#model-building-app-protocol-seqhomology-5"><span class="std std-numref">Fig. 19.5</span></a> (A).
Follow the general procedure shown above (Protocol execution
section). Remark that in this case there is no <em>template</em> atomic
structure. Instead, an empty <em>ChimeraX</em> graphics window will appear (C)
together with the <em>target</em> sequence that we’d like to model and
<em>BLASTP</em> retrieved results (E). Note that it could take some
seconds the opening of the <em>BlastProtein</em> panel. Have a look to
these results and select one of them as possible <em>template</em> of your
<em>target</em> sequence. In this particular case, for example, we are
going to choose the first one (<em>5WT9</em>). Open this atomic structure
in <em>ChimeraX</em> by writing in the command line:</div>
</div>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="nb">open</span> <span class="mi">5</span><span class="n">wt9</span>
</pre></div>
</div>
</div></blockquote>
<div class="line-block">
<div class="line">At this point, open the <em>Modeller Comparative</em> panel and complete
it as indicated in <a class="reference internal" href="#model-building-app-protocol-seqhomology-5"><span class="std std-numref">Fig. 19.5</span></a> (B). Wait for a while. After getting the retrieved models, if you want to select, for example, the <em>target</em>
model <em>#3.2</em>, write in the command line:</div>
</div>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rename</span> <span class="c1">#3.2 id #4</span>
<span class="n">scipionwrite</span> <span class="c1">#4 prefix model_3_2_</span>
</pre></div>
</div>
</div></blockquote>
<div class="line-block">
<div class="line">And <em>ChimeraX Quit</em> to close the protocol. Visualize your results.</div>
</div>
</li>
</ul>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="a120-emRingerProtocol.html" class="btn btn-neutral float-right" title="20. Phenix EMRinger protocol" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="a102-localMonoRes.html" class="btn btn-neutral float-left" title="18. Local MonoRes protocol" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Scipion team

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  
<div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
        <span class="fa fa-book"> Other Versions</span>
        v: release-3.0.0
        <span class="fa fa-caret-down"></span>
    </span>
    <div class="rst-other-versions">
        <dl>
            <dt>Branches</dt>
            <dd><a href="../../../../../../gh-pages/index.html">gh-pages</a></dd>
            <dd><a href="../../../../../../release-2.0.0/index.html">release-2.0.0</a></dd>
            <dd><a href="a110-modelFromTemplate.html">release-3.0.0</a></dd>
            <dd><a href="../../../../../../rm_model_building/index.html">rm_model_building</a></dd>
        </dl>
    </div>
</div>


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>