<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>19. References &mdash; Scipion 3.0.0 documentation</title>
      <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../../../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../../../../_static/custom.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../../../../" id="documentation_options" src="../../../../../_static/documentation_options.js"></script>
        <script src="../../../../../_static/jquery.js"></script>
        <script src="../../../../../_static/underscore.js"></script>
        <script src="../../../../../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../../../../../_static/doctools.js"></script>
    <script src="../../../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../../search.html" />
    <link rel="next" title="1. Answers to Questions" href="a010-answers.html" />
    <link rel="prev" title="18. TODO" href="140-todo.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../../../../../index.html" class="icon icon-home"> Scipion
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/how-to-install.html">Installing Scipion</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">User documentation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../user-documentation.html#tutorials">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#single-particle-tutorials">Single particle tutorials</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#tomography-tutorials">Tomography tutorials</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../../user-documentation.html#model-building">Model Building</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="index.html">Model Building by Homology</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="index.html#contents">Contents</a></li>
<li class="toctree-l4"><a class="reference internal" href="index.html#appendices">Appendices</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../Alphafold/index.html">Build initial model with Alphafold</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html#processing-how-to-s">Processing How To’s</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities.html">On-the-fly processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/customize-report.html">Summary Reports</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/scipion-API.html">Scipion API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/plugins-API.html">Plugins API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/xmipp-API.html">Xmipp API</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../misc/contact-us.html">Contact Us</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Licences</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../license.html">Licenses</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../../index.html">Scipion</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content style-external-links">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../../../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../../../user-documentation.html">Graphical interface manuals</a> &raquo;</li>
          <li><a href="index.html">Model Building Tutorial</a> &raquo;</li>
      <li><span class="section-number">19. </span>References</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../../../../_sources/docs/user/tutorials/modelBuilding/Homology/z010-references.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="references">
<h1><span class="section-number">19. </span>References<a class="headerlink" href="#references" title="Permalink to this heading"></a></h1>
<p id="id1"><dl class="citation">
<dt class="label" id="id2"><span class="brackets">1</span></dt>
<dd><p>M. Khoshouei, M. Radjainia, W. Baumeister, and R. Danev. Cryo-EM structure of haemoglobin at 3.2 Å determined with the Volta phase plate. <em>Nat Commun</em>, 8:16099, Jun 2017.</p>
</dd>
<dt class="label" id="id3"><span class="brackets">2</span></dt>
<dd><p>M. F. Perutz, M. G. Rossmann, A. F. Cullis, H. Muirhead, G. Will, and A. C. North. Structure of haemoglobin: a three-dimensional Fourier synthesis at 5.5-A. resolution, obtained by X-ray analysis. <em>Nature</em>, 185(4711):416–422, Feb 1960.</p>
</dd>
<dt class="label" id="id4"><span class="brackets">3</span></dt>
<dd><p>M. Martínez, A. Jiménez-Moreno, D. Maluenda, E. Ramírez-Aportela, R. Melero, A. Cuervo, P. Conesa, L. DelCano, Y.C. Fonseca, R. Sanchez-García, D. Strelak, J.J. Conesa, E. Fernandez-Giménez, F. de Isidro, C.O.S. Sorzano, J.M. Carazo, and R. Marabini. Integration of Cryo-EM Model Building Software in Scipion. <em>J Chem Inf Model</em>, 60(5):2533–2540, 2020.</p>
</dd>
<dt class="label" id="id5"><span class="brackets">4</span></dt>
<dd><p>Jose Luis Vilas, Josué Gómez-Blanco, Pablo Conesa, Roberto Melero, José Miguel de la Rosa-Trevín, Joaquin Otón, Jesús Cuenca, Roberto Marabini, José María Carazo, Javier Vargas, and others. Monores: automatic and accurate estimation of local resolution for electron microscopy maps. <em>Structure</em>, 26(2):337–344, 2018.</p>
</dd>
<dt class="label" id="id7"><span class="brackets">5</span></dt>
<dd><p>Erney Ramírez-Aportela, Jose Luis Vilas, Roberto Melero, Pablo Conesa, Marta Martínez, David Maluenda, Javier Mota, Amaya Jiménez, Javier Vargas, Roberto Marabini, Jose Maria Carazo, and Carlos Oscar S. Sorzano. Automatic local resolution-based sharpening of cryo-em maps. <em>bioRxiv</em>, 2018. URL: <a class="reference external" href="https://www.biorxiv.org/content/early/2018/10/02/433284">https://www.biorxiv.org/content/early/2018/10/02/433284</a>, <a class="reference external" href="https://arxiv.org/abs/https://www.biorxiv.org/content/early/2018/10/02/433284.full.pdf">arXiv:https://www.biorxiv.org/content/early/2018/10/02/433284.full.pdf</a>, <a class="reference external" href="https://doi.org/10.1101/433284">doi:10.1101/433284</a>.</p>
</dd>
<dt class="label" id="id6"><span class="brackets">6</span></dt>
<dd><p>R Sanchez-Garcia, J Gomez-Blanco, A Cuervo, JM Carazo, COS Sorzano, and J Vargas. Deepemhancer: a deep learning solution for cryo-em volume post-processing. <em>bioRxiv</em>, 2020. URL: <a class="reference external" href="https://www.biorxiv.org/content/early/2020/08/17/2020.06.12.148296">https://www.biorxiv.org/content/early/2020/08/17/2020.06.12.148296</a>, <a class="reference external" href="https://arxiv.org/abs/https://www.biorxiv.org/content/early/2020/08/17/2020.06.12.148296.full.pdf">arXiv:https://www.biorxiv.org/content/early/2020/08/17/2020.06.12.148296.full.pdf</a>, <a class="reference external" href="https://doi.org/10.1101/2020.06.12.148296">doi:10.1101/2020.06.12.148296</a>.</p>
</dd>
<dt class="label" id="id8"><span class="brackets">7</span></dt>
<dd><p>T. D. Goddard, C. C. Huang, E. C. Meng, E. F. Pettersen, G. S. Couch, J. H. Morris, and T. E. Ferrin. UCSF ChimeraX: Meeting modern challenges in visualization and analysis. <em>Protein Sci.</em>, 27(1):14–25, 01 2018.</p>
</dd>
<dt class="label" id="id9"><span class="brackets">8</span></dt>
<dd><p>W. R. Pearson. An introduction to sequence similarity (&quot;homology&quot;) searching. <em>Curr Protoc Bioinformatics</em>, Chapter 3:Unit3.1, Jun 2013.</p>
</dd>
<dt class="label" id="id10"><span class="brackets">9</span></dt>
<dd><p>A. Kryshtafovych, B. Monastyrskyy, K. Fidelis, J. Moult, T. Schwede, and A. Tramontano. Evaluation of the template-based modeling in CASP12. <em>Proteins</em>, 86 Suppl 1:321–334, Mar 2018.</p>
</dd>
<dt class="label" id="id11"><span class="brackets">10</span></dt>
<dd><p>S. F. Altschul, T. L. Madden, A. A. Schaffer, J. Zhang, Z. Zhang, W. Miller, and D. J. Lipman. Gapped BLAST and PSI-BLAST: a new generation of protein database search programs. <em>Nucleic Acids Res.</em>, 25(17):3389–3402, Sep 1997.</p>
</dd>
<dt class="label" id="id12"><span class="brackets">11</span></dt>
<dd><p>L. Camardella, C. Caruso, R. D'Avino, G. di Prisco, B. Rutigliano, M. Tamburrini, G. Fermi, and M. F. Perutz. Haemoglobin of the antarctic fish Pagothenia bernacchii. Amino acid sequence, oxygen equilibria and crystal structure of its carbonmonoxy derivative. <em>J. Mol. Biol.</em>, 224(2):449–460, Mar 1992.</p>
</dd>
<dt class="label" id="id13"><span class="brackets">12</span></dt>
<dd><p>A. Sali and T. L. Blundell. Comparative protein modelling by satisfaction of spatial restraints. <em>J. Mol. Biol.</em>, 234(3):779–815, Dec 1993.</p>
</dd>
<dt class="label" id="id14"><span class="brackets">13</span></dt>
<dd><p>D. Liebschner, P. V. Afonine, M. L. Baker, G. Bunk?czi, V. B. Chen, T. I. Croll, B. Hintze, L. W. Hung, S. Jain, A. J. McCoy, N. W. Moriarty, R. D. Oeffner, B. K. Poon, M. G. Prisant, R. J. Read, J. S. Richardson, D. C. Richardson, M. D. Sammito, O. V. Sobolev, D. H. Stockwell, T. C. Terwilliger, A. G. Urzhumtsev, L. L. Videau, C. J. Williams, and P. D. Adams. Macromolecular structure determination using X-rays, neutrons and electrons: recent developments in Phenix. <em>Acta Crystallogr D Struct Biol</em>, 75(Pt 10):861–877, Oct 2019.</p>
</dd>
<dt class="label" id="id15"><span class="brackets">14</span></dt>
<dd><p>P. Emsley, B. Lohkamp, W. G. Scott, and K. Cowtan. Features and development of Coot. <em>Acta Crystallogr. D Biol. Crystallogr.</em>, 66(Pt 4):486–501, Apr 2010.</p>
</dd>
<dt class="label" id="id16"><span class="brackets">15</span></dt>
<dd><p>Pavel V. Afonine, Billy K. Poon, Randy J. Read, Oleg V. Sobolev, Thomas C. Terwilliger, Alexandre Urzhumtsev, and Paul D. Adams. Real-space refinement in \it PHENIX for cryo-EM and crystallography. <em>Acta Crystallographica Section D</em>, 74(6):531–544, Jun 2018. URL: <a class="reference external" href="https://doi.org/10.1107/S2059798318006551">https://doi.org/10.1107/S2059798318006551</a>, <a class="reference external" href="https://doi.org/10.1107/S2059798318006551">doi:10.1107/S2059798318006551</a>.</p>
</dd>
<dt class="label" id="id17"><span class="brackets">16</span></dt>
<dd><p>A. A. Vagin, R. A. Steiner, A. A. Lebedev, L. Potterton, S. McNicholas, F. Long, and G. N. Murshudov. REFMAC5 dictionary: organization of prior chemical knowledge and guidelines for its use. <em>Acta Crystallogr. D Biol. Crystallogr.</em>, 60(Pt 12 Pt 1):2184–2195, Dec 2004.</p>
</dd>
<dt class="label" id="id18"><span class="brackets">17</span></dt>
<dd><p>B. A. Barad, N. Echols, R. Y. Wang, Y. Cheng, F. DiMaio, P. D. Adams, and J. S. Fraser. EMRinger: side chain-directed model and map validation for 3D cryo-electron microscopy. <em>Nat. Methods</em>, 12(10):943–946, Oct 2015.</p>
</dd>
<dt class="label" id="id19"><span class="brackets">18</span></dt>
<dd><p>I. W. Davis, L. W. Murray, J. S. Richardson, and D. C. Richardson. MOLPROBITY: structure validation and all-atom contact analysis for nucleic acids and their complexes. <em>Nucleic Acids Res.</em>, 32(Web Server issue):W615–619, Jul 2004.</p>
</dd>
<dt class="label" id="id20"><span class="brackets">19</span></dt>
<dd><p>P. V. Afonine, B. P. Klaholz, N. W. Moriarty, B. K. Poon, O. V. Sobolev, T. C. Terwilliger, P. D. Adams, and A. Urzhumtsev. New tools for the analysis and validation of cryo-EM maps and atomic models. <em>Acta Crystallogr D Struct Biol</em>, 74(Pt 9):814–840, Sep 2018.</p>
</dd>
<dt class="label" id="id21"><span class="brackets">20</span></dt>
<dd><p>P. Zwart, P. Afonine, and R.W. Grosse-Kunstleve. Superimposing two PDB files with superpose_pdbs. 2017. <a class="reference external" href="https://www.phenix-online.org/documentation/reference/superpose_pdbs.html">https://www.phenix-online.org/documentation/reference/superpose_pdbs.html</a>, Accessed: 2018-10-31.</p>
</dd>
<dt class="label" id="id22"><span class="brackets">21</span></dt>
<dd><p>C. J. Williams, J. J. Headd, N. W. Moriarty, M. G. Prisant, L. L. Videau, L. N. Deis, V. Verma, D. A. Keedy, B. J. Hintze, V. B. Chen, S. Jain, S. M. Lewis, W. B. Arendall, J. Snoeyink, P. D. Adams, S. C. Lovell, J. S. Richardson, and D. C. Richardson. MolProbity: More and better reference data for improved all-atom structure validation. <em>Protein Sci.</em>, 27(1):293–315, 01 2018.</p>
</dd>
<dt class="label" id="id23"><span class="brackets">22</span></dt>
<dd><p>M. Pérez-Illana, M. Martínez, G. N. Condezo, M. Hernando-Pérez, C. Mangroo, M. Brown, R. Marabini, and C. San Martín. Cryo-EM structure of enteric adenovirus HAdV-F41 highlights structural variations among human adenoviruses. <em>Sci Adv</em>, 02 2021.</p>
</dd>
<dt class="label" id="id24"><span class="brackets">23</span></dt>
<dd><p>A. Brown, F. Long, R. A. Nicholls, J. Toots, P. Emsley, and G. Murshudov. Tools for macromolecular model building and refinement into electron cryo-microscopy reconstructions. <em>Acta Crystallogr. D Biol. Crystallogr.</em>, 71(Pt 1):136–153, Jan 2015.</p>
</dd>
<dt class="label" id="id25"><span class="brackets">24</span></dt>
<dd><p>O. Kovalevskiy, R. A. Nicholls, F. Long, A. Carlon, and G. N. Murshudov. Overview of refinement procedures within REFMAC5: utilizing data from different sources. <em>Acta Crystallogr D Struct Biol</em>, 74(Pt 3):215–227, 03 2018.</p>
</dd>
<dt class="label" id="id28"><span class="brackets">25</span></dt>
<dd><p>J. Jumper, R. Evans, A. Pritzel, T. Green, M. Figurnov, O. Ronneberger, K. Tunyasuvunakool, R. Bates, A. Žídek, A. Potapenko, A. Bridgland, C. Meyer, S. A. A. Kohl, A. J. Ballard, A. Cowie, B. Romera-Paredes, S. Nikolov, R. Jain, J. Adler, T. Back, S. Petersen, D. Reiman, E. Clancy, M. Zielinski, M. Steinegger, M. Pacholska, T. Berghammer, S. Bodenstein, D. Silver, O. Vinyals, A. W. Senior, K. Kavukcuoglu, P. Kohli, and D. Hassabis. Highly accurate protein structure prediction with AlphaFold. <em>Nature</em>, 596(7873):583–589, 08 2021.</p>
</dd>
<dt class="label" id="id30"><span class="brackets">26</span></dt>
<dd><p>Y. Rong, J. Jiang, Y. Gao, J. Guo, D. Song, W. Liu, M. Zhang, Y. Zhao, B. Xiao, and Z. Liu. TMEM120A contains a specific coenzyme A-binding site and might not mediate poking- or stretch-induced channel activities in cells. <em>Elife</em>, 08 2021.</p>
</dd>
<dt class="label" id="id31"><span class="brackets">27</span></dt>
<dd><p>J. Xue, Y. Han, H. Baniasadi, W. Zeng, J. Pei, N. V. Grishin, J. Wang, B. P. Tu, and Y. Jiang. TMEM120A is a coenzyme A-binding membrane protein with structural similarities to ELOVL fatty acid elongase. <em>Elife</em>, 08 2021.</p>
</dd>
<dt class="label" id="id32"><span class="brackets">28</span></dt>
<dd><p>X. Chen, Y. Wang, Y. Li, X. Lu, J. Chen, M. Li, T. Wen, N. Liu, S. Chang, X. Zhang, X. Yang, and Y. Shen. Cryo-EM structure of the human TACAN in a closed state. <em>Cell Rep</em>, 38(9):110445, 03 2022.</p>
</dd>
<dt class="label" id="id33"><span class="brackets">29</span></dt>
<dd><p>R. Evans, Michael O'Neill, Alexander Pritzel, Natasha Antropova, Andrew Senior, Tim Green, Augustin Žídek, Russ Bates, Sam Blackwell, Jason Yim, Olaf Ronneberger, Sebastian Bodenstein, Michal Zielinski, Alex Bridgland, Anna Potapenko, Andrew Cowie, Kathryn Tunyasuvunakool, Rishub Jain, Ellen Clancy, Pushmeet Kohli, John Jumper, and Demis Hassabis. Protein complex prediction with AlphaFold-Multimer. <em>bioRxiv</em>, 2021. URL: <a class="reference external" href="https://www.biorxiv.org/content/early/2021/10/04/2021.10.04.463034">https://www.biorxiv.org/content/early/2021/10/04/2021.10.04.463034</a>, <a class="reference external" href="https://arxiv.org/abs/https://www.biorxiv.org/content/early/2021/10/04/2021.10.04.463034.full.pdf">arXiv:https://www.biorxiv.org/content/early/2021/10/04/2021.10.04.463034.full.pdf</a>, <a class="reference external" href="https://doi.org/10.1101/2021.10.04.463034">doi:10.1101/2021.10.04.463034</a>.</p>
</dd>
<dt class="label" id="id29"><span class="brackets">30</span></dt>
<dd><p>M. Baek, F. DiMaio, I. Anishchenko, J. Dauparas, S. Ovchinnikov, G. R. Lee, J. Wang, Q. Cong, L. N. Kinch, R. D. Schaeffer, C. Millán, H. Park, C. Adams, C. R. Glassman, A. DeGiovanni, J. H. Pereira, A. V. Rodrigues, A. A. van Dijk, A. C. Ebrecht, D. J. Opperman, T. Sagmeister, C. Buhlheller, T. Pavkov-Keller, M. K. Rathinaswamy, U. Dalwadi, C. K. Yip, J. E. Burke, K. C. Garcia, N. V. Grishin, P. D. Adams, R. J. Read, and D. Baker. Accurate prediction of protein structures and interactions using a three-track neural network. <em>Science</em>, 373(6557):871–876, 08 2021.</p>
</dd>
</dl>
</p>
</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="140-todo.html" class="btn btn-neutral float-left" title="18. TODO" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="a010-answers.html" class="btn btn-neutral float-right" title="1. Answers to Questions" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Scipion team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>