

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>9. Moving from sequence to atomic structure scenario &mdash; Scipion 3.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../../../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
        <script data-url_root="../../../../" id="documentation_options" src="../../../../_static/documentation_options.js"></script>
        <script src="../../../../_static/jquery.js"></script>
        <script src="../../../../_static/underscore.js"></script>
        <script src="../../../../_static/doctools.js"></script>
    
    <script type="text/javascript" src="../../../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/banner.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/custom.css" type="text/css" />
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" />
    <link rel="next" title="10. Merging 3D Maps and Atomic Structures: Rigid Fitting" href="080-rigidFit.html" />
    <link rel="prev" title="8. Structure Prediction by Sequence Homology. Searching for Homologues" href="060-sequenceScenario.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../../index.html" class="icon icon-home"> Scipion
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../scipion-modes/how-to-install.html">Installing Scipion</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">User documentation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../user-documentation.html#tutorials">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../user-documentation.html#single-particle-tutorials">Single particle tutorials</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../user-documentation.html#tomography-tutorials">Tomography tutorials</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../user-documentation.html#model-building">Model Building</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="index.html">Model Building Tutorial</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="index.html#contents">Contents</a></li>
<li class="toctree-l4"><a class="reference internal" href="index.html#appendices">Appendices</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../user-documentation.html#processing-how-to-s">Processing How To’s</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../facilities/facilities.html">On-the-fly processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../facilities/customize-report.html">Summary Reports</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../api/scipion-API.html">Scipion API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/plugins-API.html">Plugins API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/xmipp-API.html">Xmipp API</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../misc/contact-us.html">Contact Us</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Licences</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../license.html">Licenses</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">Scipion</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../index.html">Docs</a> &raquo;</li>
        
          <li><a href="../../user-documentation.html">Graphical interface manuals</a> &raquo;</li>
        
          <li><a href="index.html">Model Building Tutorial</a> &raquo;</li>
        
      <li><span class="section-number">9. </span>Moving from sequence to atomic structure scenario</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../../../../_sources/docs/user/tutorials/modelBuilding/070-movingFromSequence.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="moving-from-sequence-to-atomic-structure-scenario">
<h1><span class="section-number">9. </span>Moving from sequence to atomic structure scenario<a class="headerlink" href="#moving-from-sequence-to-atomic-structure-scenario" title="Permalink to this headline">¶</a></h1>
<p>In this section we are going to obtain the initial <em>model</em> of our <em>target</em> sequence, in
this case the <em>Hgb</em> <span class="math">\alpha</span> subunit. To perform this task we are going
to use the <em>Scipion</em> protocol <strong>chimerax-model from template</strong>. Although this protocol offers several different
possibilities to get the right result (see Appendix <a class="reference internal" href="a110-modelFromTemplate.html#app-modelfromtemplate"><span class="std std-ref">Model from template</span></a> for details and use cases), we are going to consider that we already have a <em>template</em>, that we have
found in the previous step and that will be used as protocol input. In
addition, although the protocol also allows to get a model including two
chains modeled simultaneously using the same <em>template</em> (multichain modeling), in
our example we are going to model only one chain, the <em>Hgb</em> <span class="math">\alpha</span>
subunit. <a class="reference internal" href="#model-building-scipion-workflow-from-sequence"><span class="std std-numref">Fig. 9.1</span></a> shows the workflow that we are going to detail in this section. Other possibilities of the protocol usage will be suggested at the end
of the chapter.</p>
<div class="figure align-center" id="model-building-scipion-workflow-from-sequence">
<a class="reference internal image-reference" href="../../../../_images/Fig66.svg"><img alt="*Scipion* framework detailing the workflow to generate the first model of the human *Hgb* :math:`\alpha` subunit." src="../../../../_images/Fig66.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.1 </span><span class="caption-text"><em>Scipion</em> framework detailing the workflow to generate the first model of the human <em>Hgb</em> <span class="math">\alpha</span> subunit.</span><a class="headerlink" href="#model-building-scipion-workflow-from-sequence" title="Permalink to this image">¶</a></p>
</div>
<div class="section" id="downloading-the-atomic-structure">
<h2><span class="section-number">9.1. </span>Downloading the atomic structure<a class="headerlink" href="#downloading-the-atomic-structure" title="Permalink to this headline">¶</a></h2>
<p>Once identified the <em>template</em> that we are going to use as structural skeleton of
our sequence, we import it into <em>Scipion</em> with the protocol <strong>import atomic structure</strong> (see <a class="reference internal" href="#model-building-import-atomic-structure"><span class="std std-numref">Fig. 9.2</span></a> (1) and Appendix <a class="reference internal" href="a080-importAtomicStructure.html#app-importatomicstructure"><span class="std std-ref">Import atomic structure</span></a>). Select the option for importing the atomic structure from ID (2), write the <em>PDB</em>
accession code (3) and execute the protocol (4).</p>
<div class="figure align-center" id="model-building-import-atomic-structure">
<a class="reference internal image-reference" href="../../../../_images/Fig10.svg"><img alt="Importing the atomic structure *1PBX*." src="../../../../_images/Fig10.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.2 </span><span class="caption-text">Importing the atomic structure <em>1PBX</em>.</span><a class="headerlink" href="#model-building-import-atomic-structure" title="Permalink to this image">¶</a></p>
</div>
<p>You can visualize the imported structure (5) in <em>ChimeraX</em> (<a class="reference internal" href="#model-building-chimera-visualization-structure"><span class="std std-numref">Fig. 9.3</span></a>). By selecting chain <em>A</em>
in the upper menu (1) you can distinguish the <em>Hgb</em> <span class="math">\alpha</span> subunit
(2).</p>
<div class="figure align-center" id="model-building-chimera-visualization-structure">
<a class="reference internal image-reference" href="../../../../_images/Fig11.svg"><img alt="Atomic structure *1PBX* visualized with :math:`ChimeraX`. *Hgb* :math:`\alpha` subunit (selected chain *A*) is shown green-highlighted." src="../../../../_images/Fig11.svg" width="80.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.3 </span><span class="caption-text">Atomic structure <em>1PBX</em> visualized with <span class="math">ChimeraX</span>. <em>Hgb</em> <span class="math">\alpha</span>
subunit (selected chain ) is shown green-highlighted.</span><a class="headerlink" href="#model-building-chimera-visualization-structure" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="structural-models-of-human-methgb-subunits-from-templates">
<span id="section-structural-models"></span><h2><span class="section-number">9.2. </span>Structural models of human <em>metHgb</em> subunits from templates<a class="headerlink" href="#structural-models-of-human-methgb-subunits-from-templates" title="Permalink to this headline">¶</a></h2>
<p><em>Modeller</em> <span id="id1">[<a class="reference internal" href="z010-references.html#id13" title="A. Sali and T. L. Blundell. Comparative protein modelling by satisfaction of spatial restraints. J. Mol. Biol., 234(3):779–815, Dec 1993.">Sali and Blundell, 1993</a>]</span> is one of the computational web services
used by <em>ChimeraX</em>, which provides the interface to run the program. Working with <em>Modeller</em>
requires a license key, which is provided free of charge for academic
users. <em>Modeller</em> allows two types of modeling computations to generate theoretical
models, <em>template</em>-based (sequence homology) and <em>template</em>-free (<em>de novo</em>, only for missing
segments). In this tutorial we are going to consider the first one:
structure prediction by sequence homology. Requirements for this type of
modeling are the <em>template</em> structure and a sequence alignment including sequences
of <em>target</em> and <em>template</em>.</p>
<p><code class="docutils literal notranslate"><span class="pre">NOTE</span> <span class="pre">before</span> <span class="pre">starting:</span></code> We are going to use a <em>ChimeraX</em>-derived <em>Scipion</em> protocol for the first time in this
tutorial (<strong>chimerax-model from template</strong>, Appendix <a class="reference internal" href="a110-modelFromTemplate.html#app-modelfromtemplate"><span class="std std-ref">Model from template</span></a>). Remark that this use of <em>ChimeraX</em> is completely different from the use of <em>ChimeraX</em> as a visualization tool,
as we have seen previously. By using the <em>ChimeraX</em> graphics window, opening it
from the <em>Scipion</em> button <em>Analyze Results</em> we can observe protocol results but we CANNOT save
anything in <em>Scipion</em>. However, using <em>ChimeraX</em> as a tool, as it is the case in <em>Scipion</em> <em>ChimeraX</em>-derived
protocols, we can perform different tasks, taking advantage of the
available <em>ChimeraX</em> tools and, finally, we CAN save the obtained results and the
working session in <em>Scipion</em>.</p>
<ul>
<li><div class="line-block">
<div class="line">Preparing your sequence alignment:</div>
<div class="line">In addition to the ways to obtain the sequence alignment using <em>ChimeraX</em>,
this alignment can be also generated in the <em>Scipion</em> protocol <strong>chimerax-model from template</strong> (Appendix <a class="reference internal" href="a110-modelFromTemplate.html#app-modelfromtemplate"><span class="std std-ref">Model from template</span></a>). This
protocol allows selecting between pairwise and multiple sequence
alignments. Besides producing more reliable alignments, especially
for more distantly related sequences, multiple sequence alignments <em>(MSA)</em>
provide more structural information than pairwise alignments; they
locate conserved regions in the molecule, thus improving
predictions of structural arrangements due to mutant residues or
residues that differ between <em>template</em> and <em>target</em> sequences <span id="id2">[<a class="reference internal" href="z010-references.html#id9" title="W. R. Pearson. An introduction to sequence similarity (&quot;homology&quot;) searching. Curr Protoc Bioinformatics, Chapter 3:Unit3.1, Jun 2013.">Pearson, 2013</a>]</span>. For this reason, in this tutorial
we are going to perform a <em>MSA</em>.
Additionally, you can also test the available tools to perform
pairwise alignments.</div>
</div>
<div class="line-block">
<div class="line">Besides <em>target</em> and <em>template</em> sequences, other sequences are needed to accomplish a
<em>MSA</em>. The type and number of the sequences
included depends on the sequence conservation, although they have
to allow differentiating conserved regions. As an example, our
<em>MSA</em> will include four more <em>Hgb</em> <span class="math">\alpha</span>
subunit sequences from organisms located between human and fish in
the evolutionary scale: <em>Equus caballus</em> (Horse), <em>Oryctolagus cuniculus</em> (Rabbit), <em>Meleagris gallopavo</em> (Wild turkey), <em>Aldabrachelys gigantea</em> (Aldabra
giant tortoise). Download these sequences one by one from <em>UniProtKB</em> database
filling in the <strong>import sequence</strong> protocol form with the appropriate accession codes, <em>P01958</em>, <em>P01948</em>, <em>P01948</em>, and <em>P83134</em>, respectively (<a class="reference internal" href="#model-building-multialignment-sequences"><span class="std std-numref">Fig. 9.4</span></a>). A similar process has to be followed for <em>Hgb</em>
<span class="math">\beta</span> subunit, importing <em>UniProtKB</em> sequences <em>P02062 (HBB HORSE)</em>, <em>P02057(HBB RABIT)</em>, <em>G1U9Q8 (G1U9Q8 MELGA)</em> and <em>P83133 (HBB ALDGI)</em>.</div>
</div>
<div class="figure align-center" id="model-building-multialignment-sequences">
<a class="reference internal image-reference" href="../../../../_images/Fig12.svg"><img alt="Importing additional sequences to perform the *MSA*." src="../../../../_images/Fig12.svg" width="95.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.4 </span><span class="caption-text">Importing additional sequences to perform the <em>MSA</em>.</span><a class="headerlink" href="#model-building-multialignment-sequences" title="Permalink to this image">¶</a></p>
</div>
</li>
<li><div class="line-block">
<div class="line">Access to <em>Modeller</em> in <em>ChimeraX</em>:</div>
<div class="line">The protocol <strong>chimerax-model from template</strong> allows direct opening of the <em>MSA</em> in <em>ChimeraX</em> and then, access to <em>Modeller</em> via web service. Fill in the
protocol form (<a class="reference internal" href="#model-building-model-from-template-protocol"><span class="std std-numref">Fig. 9.5</span></a> (1)), including the <em>template 1PBX</em>  previously imported (2), the
particular chain of interest (use the wizard to select it (3)) and
the <em>target</em> sequence of human <em>Hgb</em> <span class="math">\alpha</span> subunit (4). Since we plan to
improve the alignment by including additional sequences to align
(5), they will have to be added next (6). Finally, select one of
the multiple sequence alignment tools (7).</div>
</div>
<div class="figure align-center" id="model-building-model-from-template-protocol">
<a class="reference internal image-reference" href="../../../../_images/Fig13.svg"><img alt="Importing the multiple sequence alignment in *ChimeraX*." src="../../../../_images/Fig13.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.5 </span><span class="caption-text">Importing the multiple sequence alignment in <em>ChimeraX</em>.</span><a class="headerlink" href="#model-building-model-from-template-protocol" title="Permalink to this image">¶</a></p>
</div>
<p><em>ChimeraX</em> will be opened including this time the <em>MSA</em>
together with the <em>ChimeraX</em> graphics window (<a class="reference internal" href="#model-building-chimera-alignment"><span class="std std-numref">Fig. 9.6</span></a>). The <em>template</em> selected chain is shown
green-highlighted in both windows. As you may observe in the
alignment, <em>Hgb</em> <span class="math">\alpha</span> subunit is a quite conserved macromolecule;
there is only one gap in the alignment because <em>PRO</em> (Proline) 47 residue
disappeared throughout the evolutionary process.</p>
<div class="figure align-center" id="model-building-chimera-alignment">
<a class="reference internal image-reference" href="../../../../_images/Fig14.svg"><img alt="Opening the multiple sequence alignment in *ChimeraX*." src="../../../../_images/Fig14.svg" width="80.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.6 </span><span class="caption-text">Opening the multiple sequence alignment in <em>ChimeraX</em>.</span><a class="headerlink" href="#model-building-chimera-alignment" title="Permalink to this image">¶</a></p>
</div>
</li>
</ul>
<div class="line-block">
<div class="line">To complete the form that will allow us to get some atomic models of
the <em>target</em> sequence in <em>Modeller</em> web service, we have two possibilities: a) to select <em>Tools</em>
(<a class="reference internal" href="#model-building-chimera-alignment"><span class="std std-numref">Fig. 9.6</span></a>, red arrow) <em>-&gt; Sequence -&gt; Modeller Comparative</em> , or b) clicking with the right mouse inside the <em>Seqview</em> box (<a class="reference internal" href="#model-building-chimera-alignment"><span class="std std-numref">Fig. 9.6</span></a>,
green arrow) and selecting <em>Structure -&gt; Modeller Comparative</em> in the pop up window. A new window of <em>Modeller Comparative</em> will be opened (<a class="reference internal" href="#model-building-modeller"><span class="std std-numref">Fig. 9.7</span></a> (A)), that we have to fill in. <em>Sequence alignments</em> (1) should include the <em>template</em>
sequence. In the <em>Target sequences</em> section (2) we should include the <em>target</em> sequence that we
would like to model, <em>HBA_HUMAN_P69905</em> in this particular case. <em>Modeller</em> license key has to be
included here (3). The number of output models, 5 by default, can be
also specified (4). Since the target sequence that we would like to
model should include <em>non-water HETATM residues</em> <em>(HEME</em> group) we are going to select this choice as <em>Advanced option</em>
(5). Finally, press <em>OK</em> (6) to start the computation of potential models
for your <em>target</em> sequence.</div>
</div>
<div class="figure align-center" id="model-building-modeller">
<a class="reference internal image-reference" href="../../../../_images/Fig15.svg"><img alt="*(A)* Completing the form to access to homology modeling with *Modeller*. (B) Resulting *model scores*. (C) *ChimeraX Models* panel." src="../../../../_images/Fig15.svg" width="70.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.7 </span><span class="caption-text"><em>(A)</em> Completing the form to access to homology modeling with <em>Modeller</em>. (B) Resulting <em>model scores</em>. (C) <em>ChimeraX Models</em> panel.</span><a class="headerlink" href="#model-building-modeller" title="Permalink to this image">¶</a></p>
</div>
<p>In <em>ChimeraX</em> main graphics window, lower left corner, you may see the status of
your job. After a while, five possible atomic structures, from now ahead
<em>models</em>, are retrieved for the <em>target</em> sequence (<a class="reference internal" href="#model-building-modeller"><span class="std std-numref">Fig. 9.7</span></a> (B)) together with their assessment
scores. Column <em>GA341</em> of <em>Modeller Results</em> indicates the score derived from statistical
potentials (values in <em>[0, 1]</em>; <em>&gt; 0,7</em> for reliable <em>models</em>). Column <em>zDOPE</em> (normalized Discrete
Optimized Protein Energy) score depends on the atomic distance (negative
values for the better <em>models</em>). You can check every model number in <em>ChimeraX</em>’s main
menu (<em>Tools -&gt; Models</em> (C)).</p>
<p>For this tutorial we are going to select <em>model #3.2</em> (<a class="reference internal" href="#model-building-modeller"><span class="std std-numref">Fig. 9.7</span></a> (B), green arrow). Renaming
this model is the first step to save it. We can rename the model by
typing in the <em>ChimeraX</em> command line:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rename</span> <span class="c1">#3.2 id #4</span>
</pre></div>
</div>
<p>The renamed atomic structure will appear in the <em>Models</em> panel (<a class="reference internal" href="#model-building-modeller"><span class="std std-numref">Fig. 9.7</span></a> (C), green
arrow). To track this new atomic structure in the <em>Scipion</em> workflow, we can write
in the <em>ChimeraX</em> command line:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">scipionwrite</span> <span class="c1">#4 prefix model_from_modeller_3_2_</span>
</pre></div>
</div>
<p>In case that the <em>Advanced option “Include non-water HETATM residues from template”</em> (<a class="reference internal" href="#model-building-modeller"><span class="std std-numref">Fig. 9.7</span></a> (A, 5)) didn’t include the <em>HEME</em> group in the retrieved
models, an alternative option to have the model with the <em>HEME</em> group (residue
144 from the atomic structure <em>#2</em> chain A) could be:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rename</span> <span class="c1">#3.2 id #4</span>
<span class="n">save</span> <span class="o">/</span><span class="n">tmp</span><span class="o">/</span><span class="n">chainA</span><span class="o">.</span><span class="n">cif</span> <span class="nb">format</span> <span class="n">mmcif</span> <span class="n">models</span> <span class="c1">#4</span>
<span class="nb">open</span> <span class="o">/</span><span class="n">tmp</span><span class="o">/</span><span class="n">chainA</span><span class="o">.</span><span class="n">cif</span>
<span class="n">select</span> <span class="c1">#2/A:144</span>
<span class="n">save</span> <span class="o">/</span><span class="n">tmp</span><span class="o">/</span><span class="n">HEME</span><span class="o">.</span><span class="n">cif</span> <span class="nb">format</span> <span class="n">mmcif</span> <span class="n">models</span> <span class="c1">#2 selectedOnly true</span>
<span class="nb">open</span> <span class="o">/</span><span class="n">tmp</span><span class="o">/</span><span class="n">HEME</span><span class="o">.</span><span class="n">cif</span>
<span class="n">scipioncombine</span> <span class="c1">#4,5</span>

<span class="n">scipionwrite</span> <span class="c1">#6 prefix Hgb_alpha_</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">NOTE:</span></code> We have saved the <em>HEME</em> group of the <em>template chain A</em> in a new file that will be opened as <em>model #5</em>.
Finally, the combination of <em>models #4</em> (retrieved aminoacid <em>model</em> of the <em>target</em> sequence) and <em>#5 (HEME</em>
group of the <em>template chain A</em>) generates a new model <em>#6</em> that will be saved in <em>Scipion</em>. A different <em>model ID</em> could be selected by the user adding to the last command line <em>modelid n</em>.</p>
<p>After closing <em>ChimeraX</em>, you can visualize (<a class="reference internal" href="#model-building-model-from-template-protocol"><span class="std std-numref">Fig. 9.5</span></a> (8)) your full predicted <em>model</em> (<a class="reference internal" href="#model-building-chimera-model"><span class="std std-numref">Fig. 9.8</span></a>) that
includes the <em>HEME</em> group (1). The string that we have included as <em>prefix</em> in the
command line <em>scipionwrite</em> will allow us to follow the atomic structure in a more
simple manner. You can check the <em>prefix</em> in the name of the saved atomic
structure <em>(Hgb_alpha_Atom__struct_6_006815)</em> in the <em>Models</em> panel of <a class="reference internal" href="#model-building-chimera-model"><span class="std std-numref">Fig. 9.8</span></a> (1). Interestingly, the suffix number of
the saved atomic structure <em>(006815)</em> stands for the ID protocol number.</p>
<div class="figure align-center" id="model-building-chimera-model">
<a class="reference internal image-reference" href="../../../../_images/Fig16.svg"><img alt="Initial *model* of human *metHgb* :math:`\alpha` subunit, including the *HEME* group (blue)." src="../../../../_images/Fig16.svg" width="80.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.8 </span><span class="caption-text">Initial <em>model</em> of human <em>metHgb</em> <span class="math">\alpha</span> subunit, including the <em>HEME</em> group (blue).</span><a class="headerlink" href="#model-building-chimera-model" title="Permalink to this image">¶</a></p>
</div>
<p>In a similar process, you can also obtain the initial atomic structure
of the human <em>metHgb</em> <span class="math">\beta</span> subunit. Take into account that in this last
case the HEME group is the residue <em>148</em> of the chain <em>B</em>. The command lines are
similar to the previous case of the <em>metHgb</em> <span class="math">\alpha</span> subunit if you also
select the <em>model #3.2</em>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rename</span> <span class="c1">#3.2 id #4</span>
<span class="n">save</span> <span class="o">/</span><span class="n">tmp</span><span class="o">/</span><span class="n">chainB</span><span class="o">.</span><span class="n">cif</span> <span class="nb">format</span> <span class="n">mmcif</span> <span class="n">models</span> <span class="c1">#4</span>
<span class="nb">open</span> <span class="o">/</span><span class="n">tmp</span><span class="o">/</span><span class="n">chainB</span><span class="o">.</span><span class="n">cif</span>
<span class="n">select</span> <span class="c1">#2/B:148</span>
<span class="n">save</span> <span class="o">/</span><span class="n">tmp</span><span class="o">/</span><span class="n">HEME_B</span><span class="o">.</span><span class="n">cif</span> <span class="nb">format</span> <span class="n">mmcif</span> <span class="n">models</span> <span class="c1">#2 selectedOnly true</span>
<span class="nb">open</span> <span class="o">/</span><span class="n">tmp</span><span class="o">/</span><span class="n">HEME_B</span><span class="o">.</span><span class="n">cif</span>
<span class="n">scipioncombine</span> <span class="c1">#4,5</span>
<span class="nb">setattr</span> <span class="c1">#6/A c chain_id B</span>
<span class="n">scipionwrite</span> <span class="c1">#6 prefix Hgb_beta_</span>
</pre></div>
</div>
<p>In addition, we have included a command to change the chain <em>id</em> of the
second polypeptide from <em>A</em> to <em>B</em>. In general, to change the chain <em>ID</em> you have to
write:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="nb">setattr</span> <span class="c1">#model_number/old_ID c chain_id new_ID</span>
<span class="nb">setattr</span> <span class="c1">#model_number/old_ID r chain_id new_ID</span>
</pre></div>
</div>
<p>Note that in the new version of <em>ChimeraX</em> you have to remove <em>c</em> from <em>setattr 6/A c chain id B</em>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="nb">setattr</span> <span class="c1">#model_number/old_ID chain_id new_ID</span>
</pre></div>
</div>
<div class="section" id="additional-exercises-for-practising">
<h3><span class="section-number">9.2.1. </span>Additional exercises for practising<a class="headerlink" href="#additional-exercises-for-practising" title="Permalink to this headline">¶</a></h3>
<div class="line-block">
<div class="line">Since the protocol <strong>chimerax-model from template</strong> allows to use other options, inspect by your own
the possible result obtained by:</div>
</div>
<ol class="arabic simple">
<li><p>Using as input only the <em>target</em> sequence of the human <em>metHgb</em> <span class="math">\alpha</span> subunit.</p></li>
<li><p>Using as input the same atomic structure <em>template</em> and the <em>target</em> sequences of both
the human <em>metHgb</em> <span class="math">\alpha</span> and <span class="math">\beta</span> subunits. Improve the
alignment of the human <em>metHgb</em> <span class="math">\alpha</span> subunit with additional
sequences and improve the alignment of the human <em>metHgb</em> <span class="math">\beta</span>
subunit with your own sequence alignment that contains about 30
sequences.</p></li>
</ol>
</div>
</div>
<div class="section" id="option-of-recovering-the-chimerax-session">
<h2><span class="section-number">9.3. </span>Option of recovering the <em>ChimeraX</em> session<a class="headerlink" href="#option-of-recovering-the-chimerax-session" title="Permalink to this headline">¶</a></h2>
<p>If for any reason you decide to go back and check a different <em>model</em> from the
five <em>models</em> initially provided by <em>Modeller</em>, you can do it by using <strong>chimerax-restore session</strong> protocol (Appendix <a class="reference internal" href="a030-chimeraRestoreSession.html#app-chimerarestoresession"><span class="std std-ref">ChimeraX restore session</span></a>). This
protocol may be used whenever session had been saved, specifically after
using protocols <strong>chimerax-rigid fit</strong>, <strong>chimerax-model from template</strong>, and <strong>chimerax-map subtraction</strong>. In addition to the <em>ChimeraX</em> command line <code class="docutils literal notranslate"><span class="pre">scipionss</span></code>, command
lines <code class="docutils literal notranslate"><span class="pre">scipionwrite</span></code> and <code class="docutils literal notranslate"><span class="pre">scipioncombine</span></code> also save <em>ChimeraX</em> session by default. So, if you want to restore a
previous session just open the form (<a class="reference internal" href="#model-building-restore-session-protocol"><span class="std std-numref">Fig. 9.9</span></a>, 1), and include the session that
you’d like to restore (2).</p>
<div class="figure align-center" id="model-building-restore-session-protocol">
<a class="reference internal image-reference" href="../../../../_images/Fig17.svg"><img alt="Restoring session in *ChimeraX*." src="../../../../_images/Fig17.svg" width="90.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.9 </span><span class="caption-text">Restoring session in <em>ChimeraX</em>.</span><a class="headerlink" href="#model-building-restore-session-protocol" title="Permalink to this image">¶</a></p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="080-rigidFit.html" class="btn btn-neutral float-right" title="10. Merging 3D Maps and Atomic Structures: Rigid Fitting" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="060-sequenceScenario.html" class="btn btn-neutral float-left" title="8. Structure Prediction by Sequence Homology. Searching for Homologues" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Scipion team

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  
<div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
        <span class="fa fa-book"> Other Versions</span>
        v: release-3.0.0
        <span class="fa fa-caret-down"></span>
    </span>
    <div class="rst-other-versions">
        <dl>
            <dt>Branches</dt>
            <dd><a href="../../../../../gh-pages/index.html">gh-pages</a></dd>
            <dd><a href="../../../../../release-2.0.0/index.html">release-2.0.0</a></dd>
            <dd><a href="070-movingFromSequence.html">release-3.0.0</a></dd>
            <dd><a href="../../../../../rm_model_building/docs/user/tutorials/modelBuilding/070-movingFromSequence.html">rm_model_building</a></dd>
        </dl>
    </div>
</div>


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>