

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>chimera.protocols.protocol_modeller_search &mdash; Scipion 3.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../../../_static/jquery.js"></script>
        <script type="text/javascript" src="../../../_static/underscore.js"></script>
        <script type="text/javascript" src="../../../_static/doctools.js"></script>
        <script type="text/javascript" src="../../../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../_static/banner.css" type="text/css" />
  <link rel="stylesheet" href="../../../_static/custom.css" type="text/css" />
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../index.html" class="icon icon-home"> Scipion
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/scipion-modes/install-from-sources.html">Installing Scipion v3.0</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/user/troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption"><span class="caption-text">User documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/user/user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/user/user-documentation.html#tutorials">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/user/user-documentation.html#processing-how-to-s">Processing How Toâ€™s</a></li>
</ul>
<p class="caption"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/facilities/facilities.html">Streaming Processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/facilities/acquisition-simulation.html">Tutorial for Facilities</a></li>
</ul>
<p class="caption"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../api/appion/appion.html">appion package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/atomstructutils/atomstructutils.html">atomstructutils package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/atsas/atsas.html">atsas package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/bamfordlab/bamfordlab.html">bamfordlab package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/bsoft/bsoft.html">bsoft package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/ccp4/ccp4.html">ccp4 package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/chimera/chimera.html">chimera package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/cistem/cistem.html">cistem package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/cryoef/cryoef.html">cryoef package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/cryosparc2/cryosparc2.html">cryosparc2 package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/dynamo/dynamo.html">dynamo package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/eman2/eman2.html">eman2 package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/emxlib/emxlib.html">emxlib package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/fsc3d/fsc3d.html">fsc3d package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/gautomatch/gautomatch.html">gautomatch package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/gctf/gctf.html">gctf package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/imod/imod.html">imod package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/localrec/localrec.html">localrec package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/locscale/locscale.html">locscale package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/motioncorr/motioncorr.html">motioncorr package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/novactf/novactf.html">novactf package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/phenix/phenix.html">phenix package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/pyworkflow/pyworkflow.html">pyworkflow package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/pwem/pwem.html">pwem package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/relion/relion.html">relion package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/resmap/resmap.html">resmap package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/scipion/scipion.html">scipion package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/sphire/sphire.html">sphire package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/spider/spider.html">spider package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/tomo/tomo.html">tomo package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/topaz/topaz.html">topaz package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/xmipp3/xmipp3.html">xmipp3 package</a></li>
</ul>
<p class="caption"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/misc/contact-us.html">Contact Us</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">Scipion</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../index.html">Docs</a> &raquo;</li>
        
          <li><a href="../../index.html">Module code</a> &raquo;</li>
        
          <li><a href="../../chimera.html">chimera</a> &raquo;</li>
        
      <li>chimera.protocols.protocol_modeller_search</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <h1>Source code for chimera.protocols.protocol_modeller_search</h1><div class="highlight"><pre>
<span></span><span class="c1"># **************************************************************************</span>
<span class="c1"># *</span>
<span class="c1"># * Authors:     Marta Martinez (mmmtnez@cnb.csic.es)</span>
<span class="c1"># *              Roberto Marabini (roberto@cnb.csic.es)</span>
<span class="c1"># *</span>
<span class="c1"># * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC</span>
<span class="c1"># *</span>
<span class="c1"># * This program is free software; you can redistribute it and/or modify</span>
<span class="c1"># * it under the terms of the GNU General Public License as published by</span>
<span class="c1"># * the Free Software Foundation; either version 2 of the License, or</span>
<span class="c1"># * (at your option) any later version.</span>
<span class="c1"># *</span>
<span class="c1"># * This program is distributed in the hope that it will be useful,</span>
<span class="c1"># * but WITHOUT ANY WARRANTY; without even the implied warranty of</span>
<span class="c1"># * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the</span>
<span class="c1"># * GNU General Public License for more details.</span>
<span class="c1"># *</span>
<span class="c1"># * You should have received a copy of the GNU General Public License</span>
<span class="c1"># * along with this program; if not, write to the Free Software</span>
<span class="c1"># * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA</span>
<span class="c1"># * 02111-1307  USA</span>
<span class="c1"># *</span>
<span class="c1"># *  All comments concerning this program package may be sent to the</span>
<span class="c1"># *  e-mail address &#39;scipion@cnb.csic.es&#39;</span>
<span class="c1"># *</span>
<span class="c1"># **************************************************************************</span>

<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">from</span> <span class="nn">pwem</span> <span class="k">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">pyworkflow</span> <span class="k">import</span> <span class="n">VERSION_1_2</span>
<span class="kn">from</span> <span class="nn">pyworkflow.utils</span> <span class="k">import</span> <span class="n">copyFile</span>

<span class="kn">from</span> <span class="nn">.</span> <span class="k">import</span> <span class="n">ChimeraProtBase</span>

<span class="kn">from</span> <span class="nn">pwem.convert.atom_struct</span> <span class="k">import</span> <span class="n">AtomicStructHandler</span>

<span class="kn">from</span> <span class="nn">pyworkflow.protocol.params</span> <span class="k">import</span> <span class="p">(</span><span class="n">PointerParam</span><span class="p">,</span>
                                        <span class="n">StringParam</span><span class="p">,</span>
                                        <span class="n">MultiPointerParam</span><span class="p">,</span>
                                        <span class="n">BooleanParam</span><span class="p">,</span>
                                        <span class="n">EnumParam</span><span class="p">,</span>
                                        <span class="n">PathParam</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">pwem.convert.sequence</span> <span class="k">import</span> <span class="p">(</span><span class="n">SequenceHandler</span><span class="p">,</span>
                                   <span class="n">saveFileSequencesToAlign</span><span class="p">,</span>
                                   <span class="n">alignClustalSequences</span><span class="p">,</span>
                                   <span class="n">alignBioPairwise2Sequences</span><span class="p">,</span>
                                   <span class="n">alignMuscleSequences</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="k">import</span> <span class="n">OrderedDict</span>
<span class="kn">from</span> <span class="nn">..constants</span> <span class="k">import</span> <span class="n">CLUSTALO</span><span class="p">,</span> <span class="n">MUSCLE</span>


<div class="viewcode-block" id="ChimeraModelFromTemplate"><a class="viewcode-back" href="../../../api/chimera/chimera.protocols.protocol_modeller_search.html#chimera.protocols.protocol_modeller_search.ChimeraModelFromTemplate">[docs]</a><span class="k">class</span> <span class="nc">ChimeraModelFromTemplate</span><span class="p">(</span><span class="n">ChimeraProtBase</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;Protocol to model three-dimensional structures of proteins using Modeller.</span>
<span class="sd">        Execute command *scipionwrite [model #n]* from command line in order</span>
<span class="sd">        to transfer the selected</span>
<span class="sd">        pdb to scipion. Default value is model=#0,</span>
<span class="sd">        model refers to the pdb file.&quot;&quot;&quot;</span>
    <span class="n">_label</span> <span class="o">=</span> <span class="s1">&#39;model from template&#39;</span>
    <span class="n">_program</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
    <span class="n">_version</span> <span class="o">=</span> <span class="n">VERSION_1_2</span>

    <span class="n">INFILE1</span> <span class="o">=</span> <span class="s2">&quot;unaligned_1.fasta&quot;</span>
    <span class="n">OUTFILE1</span> <span class="o">=</span> <span class="s2">&quot;aligned_1.fasta&quot;</span>
    <span class="n">INFILE2</span> <span class="o">=</span> <span class="s2">&quot;unaligned_2.fasta&quot;</span>
    <span class="n">OUTFILE2</span> <span class="o">=</span> <span class="s2">&quot;aligned_2.fasta&quot;</span>
    <span class="n">TWOSEQUENCES</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">MULTIPLESEQUENCES</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="n">ProgramToAlign1</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Bio.pairwise2&#39;</span><span class="p">,</span> <span class="s1">&#39;Clustal Omega&#39;</span><span class="p">,</span> <span class="s1">&#39;MUSCLE&#39;</span><span class="p">]</span>
    <span class="n">ProgramToAlign2</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Clustal Omega&#39;</span><span class="p">,</span> <span class="s1">&#39;MUSCLE&#39;</span><span class="p">]</span>
    <span class="n">OptionForAligning</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;None&#39;</span><span class="p">,</span> <span class="s1">&#39;Additional sequences to align&#39;</span><span class="p">,</span>
                          <span class="s1">&#39;Provide your own sequence alignment&#39;</span><span class="p">]</span>

    <span class="c1"># --------------------------- DEFINE param functions --------------------</span>
    <span class="k">def</span> <span class="nf">_defineParams</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">form</span><span class="p">,</span> <span class="n">doHelp</span><span class="o">=</span><span class="kc">False</span><span class="p">):</span>
        <span class="n">formBase</span> <span class="o">=</span> <span class="nb">super</span><span class="p">(</span><span class="n">ChimeraModelFromTemplate</span><span class="p">,</span> <span class="bp">self</span><span class="p">)</span><span class="o">.</span><span class="n">_defineParams</span><span class="p">(</span><span class="n">form</span><span class="p">,</span>
                                                                       <span class="n">doHelp</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
        <span class="n">param</span> <span class="o">=</span> <span class="n">form</span><span class="o">.</span><span class="n">getParam</span><span class="p">(</span><span class="s1">&#39;pdbFileToBeRefined&#39;</span><span class="p">)</span>
        <span class="n">param</span><span class="o">.</span><span class="n">label</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="s1">&#39;Atomic structure used as template&#39;</span><span class="p">)</span>
        <span class="n">param</span><span class="o">.</span><span class="n">help</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="s2">&quot;PDBx/mmCIF file template used as basic atomic &quot;</span>
                       <span class="s2">&quot;structure to model your specific sequence.&quot;</span><span class="p">)</span>
        <span class="n">param</span> <span class="o">=</span> <span class="n">form</span><span class="o">.</span><span class="n">getParam</span><span class="p">(</span><span class="s1">&#39;inputVolume&#39;</span><span class="p">)</span>
        <span class="n">param</span><span class="o">.</span><span class="n">condition</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="s1">&#39;False&#39;</span><span class="p">)</span>
        <span class="n">param</span> <span class="o">=</span> <span class="n">form</span><span class="o">.</span><span class="n">getParam</span><span class="p">(</span><span class="s1">&#39;inputVolumes&#39;</span><span class="p">)</span>
        <span class="n">param</span><span class="o">.</span><span class="n">condition</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="s1">&#39;False&#39;</span><span class="p">)</span>
        <span class="c1"># hide inputPdbFiles</span>
        <span class="n">param</span> <span class="o">=</span> <span class="n">form</span><span class="o">.</span><span class="n">getParam</span><span class="p">(</span><span class="s1">&#39;inputPdbFiles&#39;</span><span class="p">)</span>
        <span class="n">param</span><span class="o">.</span><span class="n">condition</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="s1">&#39;False&#39;</span><span class="p">)</span>
        <span class="n">param</span><span class="o">.</span><span class="n">allowsNull</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="s1">&#39;True&#39;</span><span class="p">)</span>
        <span class="n">section</span> <span class="o">=</span> <span class="n">formBase</span><span class="o">.</span><span class="n">getSection</span><span class="p">(</span><span class="s1">&#39;Input&#39;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputStructureChain&#39;</span><span class="p">,</span> <span class="n">StringParam</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Chain &quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select a particular chain of the atomic &quot;</span>
                              <span class="s2">&quot;structure.&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputSequence1&#39;</span><span class="p">,</span> <span class="n">PointerParam</span><span class="p">,</span> <span class="n">pointerClass</span><span class="o">=</span><span class="s2">&quot;Sequence&quot;</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Target sequence&#39;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Input the aminoacid sequence to align with the &quot;</span>
                              <span class="s2">&quot;structure template sequence.&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;optionForAligning1&#39;</span><span class="p">,</span> <span class="n">EnumParam</span><span class="p">,</span>
                         <span class="n">choices</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">OptionForAligning</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Options to improve the alignment:&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;None: Option by default. Only the template and the &quot;</span>
                              <span class="s2">&quot;target sequences will be included in the alignment. &quot;</span>
                              <span class="s2">&quot;This option is recommendable when these two sequences &quot;</span>
                              <span class="s2">&quot;are very similar. Otherwise, select any of the two &quot;</span>
                              <span class="s2">&quot;additional options:</span><span class="se">\n</span><span class="s2">&quot;</span>
                              <span class="s2">&quot;Additional sequences to align: Select this option &quot;</span>
                              <span class="s2">&quot;if you want to add some more sequences to accomplish &quot;</span>
                              <span class="s2">&quot;the alignment.</span><span class="se">\n</span><span class="s2">&quot;</span>
                              <span class="s2">&quot;Provide your own sequence alignment: Your alignment&quot;</span>
                              <span class="s2">&quot;should include both the target and the template &quot;</span>
                              <span class="s2">&quot;sequences.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputYourOwnSequenceAlignment1&#39;</span><span class="p">,</span> <span class="n">PathParam</span><span class="p">,</span>
                         <span class="n">pointerClass</span><span class="o">=</span><span class="s2">&quot;File&quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;optionForAligning1 == 2&#39;</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Sequence alignment input&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Input your own sequence alignment.</span><span class="se">\n</span><span class="s2">&quot;</span>
                              <span class="s2">&quot;ChimeraX allowed formats accessible here: &quot;</span>
                              <span class="s2">&quot;https://www.cgl.ucsf.edu/chimerax/docs/user/commands/open.html#sequence &quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputSequencesToAlign1&#39;</span><span class="p">,</span> <span class="n">MultiPointerParam</span><span class="p">,</span>
                         <span class="n">pointerClass</span><span class="o">=</span><span class="s2">&quot;Sequence&quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;optionForAligning1 == 1&#39;</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Other sequences to align&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;In case you need to load more sequences to &quot;</span>
                              <span class="s2">&quot;align, you can load them here.&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputProgramToAlign1_1&#39;</span><span class="p">,</span> <span class="n">EnumParam</span><span class="p">,</span>
                         <span class="n">choices</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">ProgramToAlign1</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Alignment tool for two sequences:&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;optionForAligning1 == 0&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select a program to accomplish the sequence&quot;</span>
                              <span class="s2">&quot;alignment:</span><span class="se">\n\n</span><span class="s2">Biophyton module &quot;</span>
                              <span class="s2">&quot;Bio.pairwise2 (&quot;</span>
                              <span class="s2">&quot;http://biopython.org/DIST/docs/api/&quot;</span>
                              <span class="s2">&quot;Bio.pairwise2-module.html). Built-in &quot;</span>
                              <span class="s2">&quot;program to align two &quot;</span>
                              <span class="s2">&quot;sequences. The global &quot;</span>
                              <span class="s2">&quot;alignment algorithm from the EMBOSS suite &quot;</span>
                              <span class="s2">&quot;has been implemented with match/mismatch &quot;</span>
                              <span class="s2">&quot;scores of 3/-1 and gap penalties &quot;</span>
                              <span class="s2">&quot;(open/extend) of &quot;</span>
                              <span class="s2">&quot;3/2.</span><span class="se">\n\n</span><span class="s2">Clustal Omega &quot;</span>
                              <span class="s2">&quot;program (http://www.clustal.org/omega/, &quot;</span>
                              <span class="s2">&quot;https://doi.org/10.1038/msb.2011.75): &quot;</span>
                              <span class="s2">&quot;Multiple sequence alignment tool. Install &quot;</span>
                              <span class="s2">&quot;clustalo if you choose this option for &quot;</span>
                              <span class="s2">&quot;the first time by &#39;sudo apt-get install &quot;</span>
                              <span class="s2">&quot;clustalo&#39;.</span><span class="se">\n\n</span><span class="s2">MUSCLE program stands for &quot;</span>
                              <span class="s2">&quot;MUltiple Sequence Comparison by &quot;</span>
                              <span class="s2">&quot;Log- Expectation(&quot;</span>
                              <span class="s2">&quot;http://www.drive5.com/muscle/muscle.html, &quot;</span>
                              <span class="s2">&quot;https://doi.org/10.1093/nar/gkh340). &quot;</span>
                              <span class="s2">&quot;Install muscle if you choose this option &quot;</span>
                              <span class="s2">&quot;for the first time by &#39;sudo apt install &quot;</span>
                              <span class="s2">&quot;muscle&#39;.&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputProgramToAlign2_1&#39;</span><span class="p">,</span> <span class="n">EnumParam</span><span class="p">,</span>
                         <span class="n">choices</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">ProgramToAlign2</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Multiple alignment tool:&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;optionForAligning1 == 1&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select a program to accomplish the sequence&quot;</span>
                              <span class="s2">&quot;alignment:</span><span class="se">\n\n</span><span class="s2">Clustal Omega &quot;</span>
                              <span class="s2">&quot;program (http://www.clustal.org/omega/, &quot;</span>
                              <span class="s2">&quot;https://doi.org/10.1038/msb.2011.75): &quot;</span>
                              <span class="s2">&quot;Multiple sequence alignment tool. Install &quot;</span>
                              <span class="s2">&quot;clustalo if you choose this option for &quot;</span>
                              <span class="s2">&quot;the first time by &#39;sudo apt-get install &quot;</span>
                              <span class="s2">&quot;clustalo&#39;.</span><span class="se">\n\n</span><span class="s2">MUSCLE program stands for &quot;</span>
                              <span class="s2">&quot;MUltiple Sequence Comparison by &quot;</span>
                              <span class="s2">&quot;Log- Expectation(&quot;</span>
                              <span class="s2">&quot;http://www.drive5.com/muscle/muscle.html, &quot;</span>
                              <span class="s2">&quot;https://doi.org/10.1093/nar/gkh340). &quot;</span>
                              <span class="s2">&quot;Install muscle if you choose this option &quot;</span>
                              <span class="s2">&quot;for the first time by &#39;sudo apt install &quot;</span>
                              <span class="s2">&quot;muscle&#39;.&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;additionalTargetSequence&#39;</span><span class="p">,</span> <span class="n">BooleanParam</span><span class="p">,</span>
                         <span class="n">default</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Additional target sequence to include?&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Select YES if you want to add an additional &#39;</span>
                              <span class="s1">&#39;target sequence to model according a different &#39;</span>
                              <span class="s1">&#39;chain of the structure template. This &#39;</span>
                              <span class="s1">&#39;option is recommendable when you want to model &#39;</span>
                              <span class="s1">&#39;the two interacting elements of a particular complex&#39;</span>
                              <span class="s1">&#39; at the same time.&#39;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;selectStructureChain&#39;</span><span class="p">,</span> <span class="n">StringParam</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;additionalTargetSequence == True&#39;</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Chain &quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select a particular chain of the atomic &quot;</span>
                              <span class="s2">&quot;structure.&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputSequence2&#39;</span><span class="p">,</span> <span class="n">PointerParam</span><span class="p">,</span> <span class="n">pointerClass</span><span class="o">=</span><span class="s2">&quot;Sequence&quot;</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;additionalTargetSequence == True&#39;</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Target sequence&#39;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Input the aminoacid sequence to align with the &quot;</span>
                              <span class="s2">&quot;structure template sequence.&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;optionForAligning2&#39;</span><span class="p">,</span> <span class="n">EnumParam</span><span class="p">,</span>
                         <span class="n">choices</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">OptionForAligning</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;additionalTargetSequence == True&#39;</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Options to improve the alignment:&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;None: Option by default. Only the template and the &quot;</span>
                              <span class="s2">&quot;target sequences will be included in the alignment. &quot;</span>
                              <span class="s2">&quot;This option is recommendable when these two sequences &quot;</span>
                              <span class="s2">&quot;are very similar. Otherwise, select any of the two &quot;</span>
                              <span class="s2">&quot;additional options:</span><span class="se">\n</span><span class="s2">&quot;</span>
                              <span class="s2">&quot;Additional sequences to align: Select this option &quot;</span>
                              <span class="s2">&quot;if you want to add some more sequences to accomplish &quot;</span>
                              <span class="s2">&quot;the alignment.</span><span class="se">\n</span><span class="s2">&quot;</span>
                              <span class="s2">&quot;Provide your own sequence alignment: Your alignment&quot;</span>
                              <span class="s2">&quot;should include both the target and the template &quot;</span>
                              <span class="s2">&quot;sequences.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputYourOwnSequenceAlignment2&#39;</span><span class="p">,</span> <span class="n">PathParam</span><span class="p">,</span>
                         <span class="n">pointerClass</span><span class="o">=</span><span class="s2">&quot;File&quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;optionForAligning2 == 2 and &#39;</span>
                                   <span class="s1">&#39;additionalTargetSequence == True&#39;</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Sequence alignment input&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Input your own sequence alignment.</span><span class="se">\n</span><span class="s2">&quot;</span>
                              <span class="s2">&quot;ChimeraX allowed formats accessible here: &quot;</span>
                              <span class="s2">&quot;https://www.cgl.ucsf.edu/chimerax/docs/user/commands/open.html#sequence &quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputSequencesToAlign2&#39;</span><span class="p">,</span> <span class="n">MultiPointerParam</span><span class="p">,</span>
                         <span class="n">pointerClass</span><span class="o">=</span><span class="s2">&quot;Sequence&quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;optionForAligning2 == 1 and &#39;</span>
                                   <span class="s1">&#39;additionalTargetSequence == True&#39;</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Other sequences to align&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;In case you need to load more sequences to &quot;</span>
                              <span class="s2">&quot;align, you can load them here.&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputProgramToAlign1_2&#39;</span><span class="p">,</span> <span class="n">EnumParam</span><span class="p">,</span>
                         <span class="n">choices</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">ProgramToAlign1</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Alignment tool for two sequences:&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;optionForAligning2 == 0 and &#39;</span>
                                   <span class="s1">&#39;additionalTargetSequence == True&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select a program to accomplish the sequence&quot;</span>
                              <span class="s2">&quot;alignment:</span><span class="se">\n\n</span><span class="s2">Biophyton module &quot;</span>
                              <span class="s2">&quot;Bio.pairwise2 (&quot;</span>
                              <span class="s2">&quot;http://biopython.org/DIST/docs/api/&quot;</span>
                              <span class="s2">&quot;Bio.pairwise2-module.html). Built-in &quot;</span>
                              <span class="s2">&quot;program to align two &quot;</span>
                              <span class="s2">&quot;sequences. The global &quot;</span>
                              <span class="s2">&quot;alignment algorithm from the EMBOSS suite &quot;</span>
                              <span class="s2">&quot;has been implemented with match/mismatch &quot;</span>
                              <span class="s2">&quot;scores of 3/-1 and gap penalties &quot;</span>
                              <span class="s2">&quot;(open/extend) of &quot;</span>
                              <span class="s2">&quot;3/2.</span><span class="se">\n\n</span><span class="s2">Clustal Omega &quot;</span>
                              <span class="s2">&quot;program (http://www.clustal.org/omega/, &quot;</span>
                              <span class="s2">&quot;https://doi.org/10.1038/msb.2011.75): &quot;</span>
                              <span class="s2">&quot;Multiple sequence alignment tool. Install &quot;</span>
                              <span class="s2">&quot;clustalo if you choose this option for &quot;</span>
                              <span class="s2">&quot;the first time by &#39;sudo apt-get install &quot;</span>
                              <span class="s2">&quot;clustalo&#39;.</span><span class="se">\n\n</span><span class="s2">MUSCLE program stands for &quot;</span>
                              <span class="s2">&quot;MUltiple Sequence Comparison by &quot;</span>
                              <span class="s2">&quot;Log- Expectation(&quot;</span>
                              <span class="s2">&quot;http://www.drive5.com/muscle/muscle.html, &quot;</span>
                              <span class="s2">&quot;https://doi.org/10.1093/nar/gkh340). &quot;</span>
                              <span class="s2">&quot;Install muscle if you choose this option &quot;</span>
                              <span class="s2">&quot;for the first time by &#39;sudo apt install &quot;</span>
                              <span class="s2">&quot;muscle&#39;.&quot;</span><span class="p">)</span>
        <span class="n">section</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputProgramToAlign2_2&#39;</span><span class="p">,</span> <span class="n">EnumParam</span><span class="p">,</span>
                         <span class="n">choices</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">ProgramToAlign2</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Multiple alignment tool:&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;optionForAligning2 == 1 and &#39;</span>
                                   <span class="s1">&#39;additionalTargetSequence == True&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select a program to accomplish the sequence&quot;</span>
                              <span class="s2">&quot;alignment:</span><span class="se">\n\n</span><span class="s2">Clustal Omega &quot;</span>
                              <span class="s2">&quot;program (http://www.clustal.org/omega/, &quot;</span>
                              <span class="s2">&quot;https://doi.org/10.1038/msb.2011.75): &quot;</span>
                              <span class="s2">&quot;Multiple sequence alignment tool. Install &quot;</span>
                              <span class="s2">&quot;clustalo if you choose this option for &quot;</span>
                              <span class="s2">&quot;the first time by &#39;sudo apt-get install &quot;</span>
                              <span class="s2">&quot;clustalo&#39;.</span><span class="se">\n\n</span><span class="s2">MUSCLE program stands for &quot;</span>
                              <span class="s2">&quot;MUltiple Sequence Comparison by &quot;</span>
                              <span class="s2">&quot;Log- Expectation(&quot;</span>
                              <span class="s2">&quot;http://www.drive5.com/muscle/muscle.html, &quot;</span>
                              <span class="s2">&quot;https://doi.org/10.1093/nar/gkh340). &quot;</span>
                              <span class="s2">&quot;Install muscle if you choose this option &quot;</span>
                              <span class="s2">&quot;for the first time by &#39;sudo apt install &quot;</span>
                              <span class="s2">&quot;muscle&#39;.&quot;</span><span class="p">)</span>
        <span class="n">formBase</span><span class="o">.</span><span class="n">addLine</span><span class="p">(</span><span class="s2">&quot;Step 1:</span><span class="se">\n</span><span class="s2">In the sequence window your target &quot;</span>
                         <span class="s2">&quot;sequence (and other additional sequences that you &quot;</span>
                         <span class="s2">&quot;want to use in  the alignment) will appear aligned to &quot;</span>
                         <span class="s2">&quot;the template&#39;s sequence. Select in the sequence window &quot;</span>
                         <span class="s2">&quot;menu:</span><span class="se">\n</span><span class="s2">Structure -&gt; Modeller (homology)...;</span><span class="se">\n</span><span class="s2">A new  &quot;</span>
                         <span class="s2">&quot;window for Comparative Modeling with Modeller will &quot;</span>
                         <span class="s2">&quot;appear. Select your specific sequence as the sequence &quot;</span>
                         <span class="s2">&quot;to be modeled (target), and the input atomic structure&quot;</span>
                         <span class="o">+</span> <span class="s1">&#39;&#39;&#39;</span>
<span class="s1">        used as template for modeling. Select Run Modeller via web service </span>
<span class="s1">        and write the Modeller license key supplied (Academic user can </span>
<span class="s1">        register free of charge to receive a license key). Finally, press OK.</span>
<span class="s1">        </span><span class="se">\n</span><span class="s1">WAITING TIME: (you may see the status of your job in chimera main </span>
<span class="s1">        window, lower left corner.)</span><span class="se">\n\n</span><span class="s1">Step 2:</span><span class="se">\n</span><span class="s1">When the process finished, </span>
<span class="s1">        5 models will </span>
<span class="s1">        be automatically superimposed onto the template and model scores</span>
<span class="s1">        will appear in Modeller Results window. In Chimera main menu -&gt; </span>
<span class="s1">        Favorites -&gt; Model panel will show you: #0 (coordinate axes); #1 (</span>
<span class="s1">        template); #2.1 to 2.5 (models).Choose the one you like the best, </span>
<span class="s1">        for example model #2.1. To save it in Scipion, we need to change the </span>
<span class="s1">        model ID. In Chimera main menu: Favorites -&gt; Command Line, write </span>
<span class="s1">        *combine #2.1 model #3 close t*. Then, you will see in Model panel </span>
<span class="s1">        that selected model #2.1 renamed to combination with ID #3. Save it </span>
<span class="s1">        as first guess in Scipion by executing the Chimera command </span>
<span class="s1">        *scipionwrite [model #n]*. In our example *scipionwrite model #3*.</span><span class="se">\n</span><span class="s1"> </span>
<span class="s1">        When you use the command line scipionwrite, the Chimera session will </span>
<span class="s1">        be saved by default. Additionally, you can save the Chimera session </span>
<span class="s1">        whenever you want by executing the command *scipionss*. You will be </span>
<span class="s1">        able to restore the saved session by using the protocol chimera </span>
<span class="s1">        restore session (SCIPION menu: Tools/Calculators/chimera restore </span>
<span class="s1">        session). Once you have save your favorite model you can press </span>
<span class="s1">        Quit in the Modeller Results window.&#39;&#39;&#39;</span><span class="p">)</span>

    <span class="c1"># --------------------------- INSERT steps functions --------------------</span>

<div class="viewcode-block" id="ChimeraModelFromTemplate.prerequisitesStep"><a class="viewcode-back" href="../../../api/chimera/chimera.protocols.protocol_modeller_search.html#chimera.protocols.protocol_modeller_search.ChimeraModelFromTemplate.prerequisitesStep">[docs]</a>    <span class="k">def</span> <span class="nf">prerequisitesStep</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>

        <span class="c1"># read PDB</span>
        <span class="n">fileName</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_readPDB</span><span class="p">()</span>

        <span class="c1"># get pdb sequence</span>
        <span class="kn">import</span> <span class="nn">json</span>
        <span class="n">chainIdDict</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">loads</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">inputStructureChain</span><span class="o">.</span><span class="n">get</span><span class="p">())</span>

        <span class="n">userSeq</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequence1</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>  <span class="c1"># SEQ object from Scipion</span>

        <span class="n">inFile</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">INFILE1</span>
        <span class="n">outFile</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">OUTFILE1</span>

        <span class="n">addSeq</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">optionForAligning1</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>

        <span class="n">yourAlignment</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputYourOwnSequenceAlignment1</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>

        <span class="n">inputSeqAlign</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequencesToAlign1</span>

        <span class="n">programToAlign1</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProgramToAlign1_1</span>

        <span class="n">programToAlign2</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProgramToAlign2_1</span>

        <span class="bp">self</span><span class="o">.</span><span class="n">prePreRequisites</span><span class="p">(</span><span class="n">fileName</span><span class="p">,</span> <span class="n">chainIdDict</span><span class="p">,</span> <span class="n">userSeq</span><span class="p">,</span>
                              <span class="n">inFile</span><span class="p">,</span> <span class="n">outFile</span><span class="p">,</span> <span class="n">addSeq</span><span class="p">,</span> <span class="n">yourAlignment</span><span class="p">,</span>
                              <span class="n">inputSeqAlign</span><span class="p">,</span> <span class="n">programToAlign1</span><span class="p">,</span>
                              <span class="n">programToAlign2</span><span class="p">)</span>

        <span class="bp">self</span><span class="o">.</span><span class="n">selectedChain1</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">selectedChain</span>

        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">additionalTargetSequence</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="ow">is</span> <span class="kc">True</span><span class="p">:</span>
            <span class="n">chainIdDict</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">loads</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">selectStructureChain</span><span class="o">.</span><span class="n">get</span><span class="p">())</span>

            <span class="n">userSeq</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequence2</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>  <span class="c1"># SEQ object from Scipion</span>

            <span class="n">inFile</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">INFILE2</span>
            <span class="n">outFile</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">OUTFILE2</span>

            <span class="n">addSeq</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">optionForAligning2</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>

            <span class="n">yourAlignment</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputYourOwnSequenceAlignment2</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>

            <span class="n">inputSeqAlign</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequencesToAlign2</span>

            <span class="n">programToAlign1</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProgramToAlign1_2</span>

            <span class="n">programToAlign2</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProgramToAlign2_2</span>

            <span class="bp">self</span><span class="o">.</span><span class="n">prePreRequisites</span><span class="p">(</span><span class="n">fileName</span><span class="p">,</span> <span class="n">chainIdDict</span><span class="p">,</span> <span class="n">userSeq</span><span class="p">,</span>
                                  <span class="n">inFile</span><span class="p">,</span> <span class="n">outFile</span><span class="p">,</span> <span class="n">addSeq</span><span class="p">,</span> <span class="n">yourAlignment</span><span class="p">,</span>
                                  <span class="n">inputSeqAlign</span><span class="p">,</span> <span class="n">programToAlign1</span><span class="p">,</span>
                                  <span class="n">programToAlign2</span><span class="p">)</span>

        <span class="bp">self</span><span class="o">.</span><span class="n">selectedChain2</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">selectedChain</span></div>

<div class="viewcode-block" id="ChimeraModelFromTemplate.prePreRequisites"><a class="viewcode-back" href="../../../api/chimera/chimera.protocols.protocol_modeller_search.html#chimera.protocols.protocol_modeller_search.ChimeraModelFromTemplate.prePreRequisites">[docs]</a>    <span class="k">def</span> <span class="nf">prePreRequisites</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">fileName</span><span class="p">,</span> <span class="n">chainIdDict</span><span class="p">,</span> <span class="n">userSeq</span><span class="p">,</span> <span class="n">inFile</span><span class="p">,</span> \
                         <span class="n">outFile</span><span class="p">,</span> <span class="n">addSeq</span><span class="p">,</span> <span class="n">yourAlignment</span><span class="p">,</span> <span class="n">inputSeqAlign</span><span class="p">,</span> \
                         <span class="n">programToAlign1</span><span class="p">,</span> <span class="n">programToAlign2</span><span class="p">):</span>
        <span class="c1"># get sequence of structure chain with id chainId (selected by the user)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">selectedModel</span> <span class="o">=</span> <span class="n">chainIdDict</span><span class="p">[</span><span class="s1">&#39;model&#39;</span><span class="p">]</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">selectedChain</span> <span class="o">=</span> <span class="n">chainIdDict</span><span class="p">[</span><span class="s1">&#39;chain&#39;</span><span class="p">]</span>
        <span class="c1"># self.selectedModel = chainId.split(&#39;,&#39;)[0].split(&#39;:&#39;)[1].strip()</span>
        <span class="c1"># self.selectedChain = chainId.split(&#39;,&#39;)[1].split(&#39;:&#39;)[1].strip()</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Selected chain: </span><span class="si">%s</span><span class="s2"> from model: </span><span class="si">%s</span><span class="s2"> from structure: </span><span class="si">%s</span><span class="s2">&quot;</span> \
              <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">selectedChain</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">selectedModel</span><span class="p">,</span>
                 <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">fileName</span><span class="p">)))</span>

        <span class="c1"># Bio.Seq.Seq object</span>
        <span class="n">structureSeq</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span><span class="o">.</span><span class="n">getSequenceFromChain</span><span class="p">(</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">selectedModel</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">selectedChain</span><span class="p">)</span>

        <span class="c1"># obtain a seqID for our PDB sequence</span>
        <span class="n">structSeqID</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span><span class="o">.</span><span class="n">getFullID</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">selectedModel</span><span class="p">,</span>
                                                      <span class="bp">self</span><span class="o">.</span><span class="n">selectedChain</span><span class="p">)</span>
        <span class="c1"># END PDB sequence</span>

        <span class="c1"># start user imported target sequence</span>
        <span class="c1"># get target sequence imported by the user</span>

        <span class="n">targetSeqID</span> <span class="o">=</span> <span class="n">userSeq</span><span class="o">.</span><span class="n">getId</span><span class="p">()</span>  <span class="c1"># ID associated to SEQ object (str)</span>
        <span class="n">userSequence</span> <span class="o">=</span> <span class="n">userSeq</span><span class="o">.</span><span class="n">getSequence</span><span class="p">()</span>  <span class="c1"># sequence associated to</span>
        <span class="c1"># that SEQ object (str)</span>
        <span class="c1"># transformation of this sequence (str) in a Bio.Seq.Seq object:</span>
        <span class="n">seqHandler</span> <span class="o">=</span> <span class="n">SequenceHandler</span><span class="p">(</span><span class="n">userSequence</span><span class="p">,</span>
                                     <span class="n">isAminoacid</span><span class="o">=</span><span class="n">userSeq</span><span class="o">.</span><span class="n">getIsAminoacids</span><span class="p">())</span>
        <span class="n">targetSeq</span> <span class="o">=</span> <span class="n">seqHandler</span><span class="o">.</span><span class="n">_sequence</span>  <span class="c1"># Bio.Seq.Seq object</span>

        <span class="c1"># creation of Dic of IDs and sequences</span>
        <span class="n">SeqDic</span> <span class="o">=</span> <span class="n">OrderedDict</span><span class="p">()</span>
        <span class="n">SeqDic</span><span class="p">[</span><span class="n">structSeqID</span><span class="p">]</span> <span class="o">=</span> <span class="n">structureSeq</span>
        <span class="n">SeqDic</span><span class="p">[</span><span class="n">targetSeqID</span><span class="p">]</span> <span class="o">=</span> <span class="n">targetSeq</span>

        <span class="c1"># align sequences and save them to disk, -this will be chimera input-</span>
        <span class="c1"># get all sequences in a fasta file</span>
        <span class="n">inFile</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getInFastaSequencesFile</span><span class="p">(</span><span class="n">inFile</span><span class="p">)</span>
        <span class="n">outFile</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getOutFastaSequencesFile</span><span class="p">(</span><span class="n">outFile</span><span class="p">)</span>

        <span class="c1"># get the alignment of sequences</span>
        <span class="k">if</span> <span class="n">addSeq</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">saveFileSequencesToAlign</span><span class="p">(</span><span class="n">SeqDic</span><span class="p">,</span> <span class="n">inFile</span><span class="p">)</span>
            <span class="n">inputSeqAlign</span> <span class="o">=</span> <span class="kc">None</span>
            <span class="k">if</span> <span class="n">programToAlign1</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="o">==</span> \
                    <span class="bp">self</span><span class="o">.</span><span class="n">ProgramToAlign1</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;Bio.pairwise2&#39;</span><span class="p">):</span>
                <span class="c1"># Only the two first sequences will be included in the alignment</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">alignment</span> <span class="o">=</span> <span class="n">alignBioPairwise2Sequences</span><span class="p">(</span>
                    <span class="n">structSeqID</span><span class="p">,</span> <span class="n">structureSeq</span><span class="p">,</span>
                    <span class="n">targetSeqID</span><span class="p">,</span> <span class="n">targetSeq</span><span class="p">,</span>
                    <span class="n">outFile</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="c1"># All the sequences will be included in the alignment</span>
                <span class="k">if</span> <span class="n">programToAlign1</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="o">==</span> \
                        <span class="bp">self</span><span class="o">.</span><span class="n">ProgramToAlign1</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s1">&#39;Clustal Omega&#39;</span><span class="p">):</span>
                    <span class="n">cline</span> <span class="o">=</span> <span class="n">alignClustalSequences</span><span class="p">(</span><span class="n">inFile</span><span class="p">,</span> <span class="n">outFile</span><span class="p">)</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">cline</span> <span class="o">=</span> <span class="n">alignMuscleSequences</span><span class="p">(</span><span class="n">inFile</span><span class="p">,</span> <span class="n">outFile</span><span class="p">)</span>
                <span class="n">args</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">runJob</span><span class="p">(</span><span class="n">cline</span><span class="p">,</span> <span class="n">args</span><span class="p">)</span>
        <span class="k">elif</span> <span class="n">addSeq</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
            <span class="c1"># if there are additional sequences imported by the user</span>
            <span class="k">if</span> <span class="n">inputSeqAlign</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
                <span class="k">for</span> <span class="n">seq</span> <span class="ow">in</span> <span class="n">inputSeqAlign</span><span class="p">:</span>
                    <span class="n">seq</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
                    <span class="n">ID</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">getId</span><span class="p">()</span>
                    <span class="n">sequence</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">getSequence</span><span class="p">()</span>
                    <span class="n">seqHandler</span> <span class="o">=</span> <span class="n">SequenceHandler</span><span class="p">(</span><span class="n">sequence</span><span class="p">,</span>
                                                 <span class="n">isAminoacid</span><span class="o">=</span><span class="n">seq</span><span class="o">.</span><span class="n">getIsAminoacids</span><span class="p">())</span>
                    <span class="n">otherSeq</span> <span class="o">=</span> <span class="n">seqHandler</span><span class="o">.</span><span class="n">_sequence</span>  <span class="c1"># Bio.Seq.Seq object</span>
                    <span class="n">SeqDic</span><span class="p">[</span><span class="n">ID</span><span class="p">]</span> <span class="o">=</span> <span class="n">otherSeq</span>

            <span class="c1"># align sequences and save them to disk, -this will be chimera input-</span>
            <span class="c1"># get all sequences in a fasta file</span>
            <span class="c1"># inFile = self._getInFastaSequencesFile()</span>
            <span class="n">saveFileSequencesToAlign</span><span class="p">(</span><span class="n">SeqDic</span><span class="p">,</span> <span class="n">inFile</span><span class="p">)</span>
            <span class="c1"># outFile = self._getOutFastaSequencesFile()</span>

            <span class="c1"># All the sequences will be included in the alignment</span>
            <span class="k">if</span> <span class="n">programToAlign2</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">ProgramToAlign2</span><span class="o">.</span><span class="n">index</span><span class="p">(</span>
                    <span class="s1">&#39;Clustal Omega&#39;</span><span class="p">):</span>
                <span class="n">cline</span> <span class="o">=</span> <span class="n">alignClustalSequences</span><span class="p">(</span><span class="n">inFile</span><span class="p">,</span> <span class="n">outFile</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">cline</span> <span class="o">=</span> <span class="n">alignMuscleSequences</span><span class="p">(</span><span class="n">inFile</span><span class="p">,</span> <span class="n">outFile</span><span class="p">)</span>
            <span class="n">args</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">runJob</span><span class="p">(</span><span class="n">cline</span><span class="p">,</span> <span class="n">args</span><span class="p">)</span>

        <span class="k">else</span><span class="p">:</span>
            <span class="n">aligmentFile</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">yourAlignment</span><span class="p">)</span>
            <span class="n">outFile</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">_getExtraPath</span><span class="p">(),</span> <span class="n">aligmentFile</span><span class="p">)</span>
            <span class="n">copyFile</span><span class="p">(</span><span class="n">yourAlignment</span><span class="p">,</span> <span class="n">outFile</span><span class="p">)</span></div>

    <span class="k">def</span> <span class="nf">_readPDB</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span> <span class="o">=</span> <span class="n">AtomicStructHandler</span><span class="p">()</span>
        <span class="n">fileName</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">abspath</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">pdbFileToBeRefined</span><span class="o">.</span><span class="n">get</span><span class="p">(</span>
        <span class="p">)</span><span class="o">.</span><span class="n">getFileName</span><span class="p">())</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">fileName</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">fileName</span>

    <span class="k">def</span> <span class="nf">_getInFastaSequencesFile</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">inFile</span><span class="p">):</span>
        <span class="n">INFILENAME</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getTmpPath</span><span class="p">(</span><span class="n">inFile</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">abspath</span><span class="p">(</span><span class="n">INFILENAME</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">_getOutFastaSequencesFile</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">outFile</span><span class="p">):</span>
        <span class="n">OUTFILENAME</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getExtraPath</span><span class="p">(</span><span class="n">outFile</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">abspath</span><span class="p">(</span><span class="n">OUTFILENAME</span><span class="p">)</span>

    <span class="c1"># def validate(self):</span>
    <span class="c1">#    super(ChimeraModelFromTemplate, self).validate()</span>
    <span class="c1">#    # TODO check if clustal/muscle exists</span>
    <span class="c1">#    #TODO; change help ro installation pages instead of apt-get</span>

    <span class="k">def</span> <span class="nf">_validate</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="c1"># Check that CLUSTALO or MUSCLE program exists</span>
        <span class="n">errors</span> <span class="o">=</span> <span class="nb">super</span><span class="p">(</span><span class="n">ChimeraModelFromTemplate</span><span class="p">,</span> <span class="bp">self</span><span class="p">)</span><span class="o">.</span><span class="n">_validate</span><span class="p">()</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">is_tool</span><span class="p">(</span><span class="n">CLUSTALO</span><span class="p">)</span> <span class="ow">or</span> <span class="bp">self</span><span class="o">.</span><span class="n">is_tool</span><span class="p">(</span><span class="n">MUSCLE</span><span class="p">)):</span>
            <span class="n">errors</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Clustal-omega and MUSCLE programs missing.</span><span class="se">\n</span><span class="s2"> &quot;</span>
                          <span class="s2">&quot;You need at least one of them to run this program.</span><span class="se">\n</span><span class="s2">&quot;</span>
                          <span class="s2">&quot;Please install Clustal-omega and/or MUSCLE:</span><span class="se">\n</span><span class="s2">&quot;</span>
                          <span class="s2">&quot;     sudo apt-get install clustalo</span><span class="se">\n</span><span class="s2">&quot;</span>
                          <span class="s2">&quot;     sudo apt-get install muscle&quot;</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">errors</span></div>
</pre></div>

           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Scipion team

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  
<div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
        <span class="fa fa-book"> Other Versions</span>
        v: release-3.0.0
        <span class="fa fa-caret-down"></span>
    </span>
    <div class="rst-other-versions">
        <dl>
            <dt>Branches</dt>
            <dd><a href="../../../../dm_facilitiesUpdate/index.html">dm_facilitiesUpdate</a></dd>
            <dd><a href="../../../../gh-pages/index.html">gh-pages</a></dd>
            <dd><a href="../../../../release-2.0.0/index.html">release-2.0.0</a></dd>
            <dd><a href="protocol_modeller_search.html">release-3.0.0</a></dd>
            <dd><a href="../../../../scipion3_instalation/index.html">scipion3_instalation</a></dd>
            <dd><a href="../../../../update_scipion_modes/index.html">update_scipion_modes</a></dd>
        </dl>
    </div>
</div>


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>