

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>chimera.protocols.protocol_subtraction_maps &mdash; Scipion 3.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
        <script src="../../../_static/jquery.js"></script>
        <script src="../../../_static/underscore.js"></script>
        <script src="../../../_static/doctools.js"></script>
        <script src="../../../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../_static/custom.css" type="text/css" />
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../index.html" class="icon icon-home"> Scipion
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/scipion-modes/install-from-sources.html">Installing Scipion v2.0</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/user/troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption"><span class="caption-text">User documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/user/user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/user/user-documentation.html#tutorials">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/user/user-documentation.html#processing-how-to-s">Processing How To’s</a></li>
</ul>
<p class="caption"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/facilities/facilities.html">Streaming Processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/facilities/acquisition-simulation.html">Tutorial for Facilities</a></li>
</ul>
<p class="caption"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../api/pyworkflow/pyworkflow.html">pyworkflow package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/pwem/pwem.html">pwem package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/appion/appion.html">appion package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/atomstructutils/atomstructutils.html">atomstructutils package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/atsas/atsas.html">atsas package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/bamfordlab/bamfordlab.html">bamfordlab package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/bsoft/bsoft.html">bsoft package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/ccp4/ccp4.html">ccp4 package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/chimera/chimera.html">chimera package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/cistem/cistem.html">cistem package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/cryoef/cryoef.html">cryoef package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/cryosparc2/cryosparc2.html">cryosparc2 package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/eman2/eman2.html">eman2 package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/emxlib/emxlib.html">emxlib package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/gautomatch/gautomatch.html">gautomatch package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/gctf/gctf.html">gctf package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/localrec/localrec.html">localrec package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/locscale/locscale.html">locscale package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/motioncorr/motioncorr.html">motioncorr package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/nysbc/nysbc.html">nysbc package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/phenix/phenix.html">phenix package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/relion/relion.html">relion package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/resmap/resmap.html">resmap package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/sphire/sphire.html">sphire package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/spider/spider.html">spider package</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/xmipp3/xmipp3.html">xmipp3 package</a></li>
</ul>
<p class="caption"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../docs/misc/contact-us.html">Contact Us</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">Scipion</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../index.html">Docs</a> &raquo;</li>
        
          <li><a href="../../index.html">Module code</a> &raquo;</li>
        
          <li><a href="../../chimera.html">chimera</a> &raquo;</li>
        
      <li>chimera.protocols.protocol_subtraction_maps</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <h1>Source code for chimera.protocols.protocol_subtraction_maps</h1><div class="highlight"><pre>
<span></span><span class="c1"># **************************************************************************</span>
<span class="c1"># *</span>
<span class="c1"># * Authors:     Marta Martinez (mmmtnez@cnb.csic.es)</span>
<span class="c1"># *              Roberto Marabini (roberto@cnb.csic.es)</span>
<span class="c1"># *</span>
<span class="c1"># * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC</span>
<span class="c1"># *</span>
<span class="c1"># * This program is free software; you can redistribute it and/or modify</span>
<span class="c1"># * it under the terms of the GNU General Public License as published by</span>
<span class="c1"># * the Free Software Foundation; either version 2 of the License, or</span>
<span class="c1"># * (at your option) any later version.</span>
<span class="c1"># *</span>
<span class="c1"># * This program is distributed in the hope that it will be useful,</span>
<span class="c1"># * but WITHOUT ANY WARRANTY; without even the implied warranty of</span>
<span class="c1"># * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the</span>
<span class="c1"># * GNU General Public License for more details.</span>
<span class="c1"># *</span>
<span class="c1"># * You should have received a copy of the GNU General Public License</span>
<span class="c1"># * along with this program; if not, write to the Free Software</span>
<span class="c1"># * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA</span>
<span class="c1"># * 02111-1307  USA</span>
<span class="c1"># *</span>
<span class="c1"># *  All comments concerning this program package may be sent to the</span>
<span class="c1"># *  e-mail address &#39;scipion@cnb.csic.es&#39;</span>
<span class="c1"># *</span>
<span class="c1"># **************************************************************************</span>

<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">from</span> <span class="nn">pwem</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">pwem.viewers</span> <span class="kn">import</span> <span class="n">Chimera</span>
<span class="kn">from</span> <span class="nn">pwem.viewers.viewer_chimera</span> <span class="kn">import</span> <span class="n">chimeraScriptFileName</span>
<span class="kn">from</span> <span class="nn">pyworkflow</span> <span class="kn">import</span> <span class="n">VERSION_3_0</span>

<span class="kn">from</span> <span class="nn">pwem.convert.atom_struct</span> <span class="kn">import</span> <span class="n">AtomicStructHandler</span>

<span class="kn">from</span> <span class="nn">pyworkflow.protocol.params</span> <span class="kn">import</span> <span class="p">(</span><span class="n">PointerParam</span><span class="p">,</span>
                                        <span class="n">IntParam</span><span class="p">,</span>
                                        <span class="n">EnumParam</span><span class="p">,</span>
                                        <span class="n">FloatParam</span><span class="p">,</span>
                                        <span class="n">BooleanParam</span><span class="p">,</span>
                                        <span class="n">StringParam</span><span class="p">,</span> <span class="n">MultiPointerParam</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">pwem.convert.sequence</span> <span class="kn">import</span> <span class="p">(</span><span class="n">SequenceHandler</span><span class="p">,</span>
                                   <span class="n">saveFileSequencesToAlign</span><span class="p">,</span>
                                   <span class="n">alignClustalSequences</span><span class="p">,</span>
                                   <span class="n">alignBioPairwise2Sequences</span><span class="p">,</span>
                                   <span class="n">alignMuscleSequences</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">OrderedDict</span>
<span class="kn">from</span> <span class="nn">..constants</span> <span class="kn">import</span> <span class="n">CLUSTALO</span><span class="p">,</span> <span class="n">MUSCLE</span>
<span class="kn">from</span> <span class="nn">pwem.constants</span> <span class="kn">import</span> <span class="p">(</span><span class="n">SYM_DIHEDRAL_X</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">..constants</span> <span class="kn">import</span> <span class="p">(</span><span class="n">CHIMERA_I222</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">..convert</span> <span class="kn">import</span> <span class="n">CHIMERA_LIST</span>
<span class="kn">from</span> <span class="nn">pwem.protocols</span> <span class="kn">import</span> <span class="n">EMProtocol</span>
<span class="kn">from</span> <span class="nn">.protocol_base</span> <span class="kn">import</span> <span class="n">chimeraScriptMain</span><span class="p">,</span> \
    <span class="n">chimeraScriptHeader</span><span class="p">,</span> \
    <span class="n">chimeraMapTemplateFileName</span><span class="p">,</span> <span class="n">chimeraPdbTemplateFileName</span>
<span class="kn">from</span> <span class="nn">pwem.viewers.viewer_chimera</span> <span class="kn">import</span> <span class="p">(</span><span class="n">Chimera</span><span class="p">,</span>
                                         <span class="n">sessionFile</span><span class="p">,</span>
                                         <span class="n">chimeraMapTemplateFileName</span><span class="p">,</span>
                                         <span class="n">chimeraScriptFileName</span><span class="p">,</span>
                                         <span class="n">chimeraPdbTemplateFileName</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">pyworkflow.protocol.constants</span> <span class="kn">import</span> <span class="n">LEVEL_ADVANCED</span>

<div class="viewcode-block" id="ChimeraSubtractionMaps"><a class="viewcode-back" href="../../../api/chimera/chimera.protocols.protocol_subtraction_maps.html#chimera.protocols.protocol_subtraction_maps.ChimeraSubtractionMaps">[docs]</a><span class="k">class</span> <span class="nc">ChimeraSubtractionMaps</span><span class="p">(</span><span class="n">EMProtocol</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;Protocol to subtract two volumes.</span>
<span class="sd">        One of these volumes can be derived from an atomic structure.</span>
<span class="sd">        Execute command *scipionwrite [model #n]* from command line in order</span>
<span class="sd">        to transfer the selected</span>
<span class="sd">        pdb to scipion. Default value is model=#0,</span>
<span class="sd">        model refers to the pdb file.&quot;&quot;&quot;</span>
    <span class="n">_label</span> <span class="o">=</span> <span class="s1">&#39;map subtraction&#39;</span>
    <span class="n">_program</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
    <span class="n">_version</span> <span class="o">=</span> <span class="n">VERSION_3_0</span>

    <span class="n">MODELDERIVEDMAP</span> <span class="o">=</span> <span class="s1">&#39;map_from_model.mrc&#39;</span>
    <span class="n">DIFFMAPNAME</span> <span class="o">=</span> <span class="s1">&#39;diff_map.mrc&#39;</span>
    <span class="n">NORMALIZEDDIFFMAPNAME</span> <span class="o">=</span> <span class="s1">&#39;filtered_diff_map.mrc&#39;</span>
    <span class="n">MAP_OPTIONS</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;3D map&#39;</span><span class="p">,</span> <span class="s1">&#39;atomic structure&#39;</span><span class="p">]</span>
    <span class="n">CHIMERA_FILTERS</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Gaussian&#39;</span><span class="p">,</span> <span class="s1">&#39;Fourier Transform&#39;</span><span class="p">]</span>

    <span class="c1"># --------------------------- DEFINE param functions --------------------</span>
    <span class="k">def</span> <span class="nf">_defineParams</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">form</span><span class="p">,</span> <span class="n">doHelp</span><span class="o">=</span><span class="kc">False</span><span class="p">):</span>

        <span class="n">form</span><span class="o">.</span><span class="n">addSection</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="s1">&#39;Input&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputVolume&#39;</span><span class="p">,</span> <span class="n">PointerParam</span><span class="p">,</span> <span class="n">pointerClass</span><span class="o">=</span><span class="s2">&quot;Volume&quot;</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Input 3D Map&#39;</span><span class="p">,</span>
                      <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;difference 3D map = minuend − subtrahend&quot;</span>
                           <span class="s2">&quot;input here the minuend&quot;</span><span class="p">)</span>
        <span class="c1"># form.addParam(&#39;samplingRate&#39;, FloatParam,</span>
        <span class="c1">#             label=pwutils.Message.LABEL_SAMP_RATE,</span>
        <span class="c1">#             default=-1.0,</span>
        <span class="c1">#              help=&quot;default=-1 -&gt; input 3D Map sampling rate&quot;)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;mapOrModel&#39;</span><span class="p">,</span> <span class="n">EnumParam</span><span class="p">,</span>
                     <span class="n">choices</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">MAP_OPTIONS</span><span class="p">,</span>
                     <span class="n">display</span><span class="o">=</span><span class="n">EnumParam</span><span class="o">.</span><span class="n">DISPLAY_HLIST</span><span class="p">,</span>
                     <span class="n">default</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Subtraction of&#39;</span><span class="p">,</span>
                     <span class="n">help</span><span class="o">=</span><span class="s2">&quot;difference 3D Map = minuend − subtrahend. &quot;</span>
                          <span class="s2">&quot;Subtrahend 3D map may be provided by the user (choose &#39;3D map&#39;) &quot;</span>
                          <span class="s2">&quot;or created from an atomic coordinates file (choose &#39;atomic structure&#39;&quot;</span>
                          <span class="s2">&quot; If 3D Map is chosen, the sampling rate of minuend should be&quot;</span>
                          <span class="s2">&quot; equal to sampling rate of subtrahend&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputVolume2&#39;</span><span class="p">,</span> <span class="n">PointerParam</span><span class="p">,</span> <span class="n">pointerClass</span><span class="o">=</span><span class="s2">&quot;Volume&quot;</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1"> &#39;</span> <span class="o">%</span> <span class="mi">0</span><span class="p">),</span>
                         <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Map to subtract (subtrahend)&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Map that has to be subtracted from the minuend 3D map&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;resolution&#39;</span><span class="p">,</span> <span class="n">FloatParam</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="mi">1</span><span class="p">),</span>
                         <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Map resolution (A):&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Thershold that will be used to low pass the 3D map created &quot;</span>
                              <span class="s2">&quot;from the atomic structure. We recommend half the value of the&quot;</span>
                              <span class="s2">&quot; resolution obtained by FSC&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;pdbFileToBeRefined&#39;</span><span class="p">,</span> <span class="n">PointerParam</span><span class="p">,</span>
                      <span class="n">pointerClass</span><span class="o">=</span><span class="s2">&quot;AtomStruct&quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1"> &#39;</span> <span class="o">%</span> <span class="mi">1</span><span class="p">),</span>
                      <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Atomic structure&#39;</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Atomic structure to derive &quot;</span>
                       <span class="s2">&quot;a 3D map that will be subtracted from the minuend map.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;selectChain&#39;</span><span class="p">,</span> <span class="n">BooleanParam</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="mi">1</span><span class="p">),</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Select a specific chain?&quot;</span><span class="p">,</span>
                         <span class="n">default</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select &#39;Yes&#39; if you want to generate the map&quot;</span>
                              <span class="s2">&quot; from a specific chain of the atomic structure.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;selectStructureChain&#39;</span><span class="p">,</span> <span class="n">StringParam</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1"> and selectChain==True&#39;</span> <span class="o">%</span> <span class="mi">1</span><span class="p">),</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Chain of the atomic structure&quot;</span><span class="p">,</span> <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select a particular chain of the atomic &quot;</span>
                              <span class="s2">&quot;structure.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;removeResidues&#39;</span><span class="p">,</span> <span class="n">BooleanParam</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="mi">1</span><span class="p">),</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Remove residues from the atomic structure?&quot;</span><span class="p">,</span>
                         <span class="n">default</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select &#39;Yes&#39; to remove a certain &quot;</span>
                              <span class="s2">&quot;number of residues of one of the chains &quot;</span>
                              <span class="s2">&quot;of the atomic structure. these removed residues &quot;</span>
                              <span class="s2">&quot;might help you to establish a control of &quot;</span>
                              <span class="s2">&quot;appropriate levels of map density.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputStructureChain&#39;</span><span class="p">,</span> <span class="n">StringParam</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="p">((</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1"> and &#39;</span>
                                    <span class="s1">&#39;removeResidues==True and &#39;</span>
                                    <span class="s1">&#39;selectChain==False&#39;</span><span class="p">)</span> <span class="o">%</span> <span class="mi">1</span><span class="p">),</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Chain &quot;</span><span class="p">,</span> <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select a particular chain of the atomic &quot;</span>
                              <span class="s2">&quot;structure.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;firstResidueToRemove&#39;</span><span class="p">,</span> <span class="n">StringParam</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="p">((</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1"> and &#39;</span>
                                    <span class="s1">&#39;removeResidues==True&#39;</span><span class="p">)</span> <span class="o">%</span> <span class="mi">1</span><span class="p">),</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;First residue to remove &quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
                         <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select the first residue of the selected atomic &quot;</span>
                              <span class="s2">&quot;structure chain that you want to remove.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;lastResidueToRemove&#39;</span><span class="p">,</span> <span class="n">StringParam</span><span class="p">,</span>
                         <span class="n">condition</span><span class="o">=</span><span class="p">((</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1"> and &#39;</span>
                                    <span class="s1">&#39;removeResidues==True&#39;</span><span class="p">)</span> <span class="o">%</span> <span class="mi">1</span><span class="p">),</span>
                         <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Last residue to remove &quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                         <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select the last residue of the selected atomic &quot;</span>
                              <span class="s2">&quot;structure chain that you want to remove.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;applySymmetry&#39;</span><span class="p">,</span> <span class="n">BooleanParam</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="mi">1</span><span class="p">),</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Apply symmetry to the atomic structure:&quot;</span><span class="p">,</span>
                      <span class="n">default</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;&#39;Symmetry = Yes&#39; indicates that symmetry will be applied. &quot;</span>
                           <span class="s2">&quot;This option is recommended is the atomic structure &quot;</span>
                           <span class="s2">&quot;corresponds to the asymmetrical unit and you want to &quot;</span>
                           <span class="s2">&quot;regenerate the structure of the whole map.</span><span class="se">\n</span><span class="s2">&#39;Symmetry = &quot;</span>
                           <span class="s2">&quot;No&#39; indicates that symmetry will not be applied, and then  &quot;</span>
                           <span class="s2">&quot;the map derived from the atomic structure involves the input&quot;</span>
                           <span class="s2">&quot;atomic structure only.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;symmetryGroup&#39;</span><span class="p">,</span> <span class="n">EnumParam</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="p">((</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1"> and applySymmetry==True&#39;</span><span class="p">)</span> <span class="o">%</span> <span class="mi">1</span><span class="p">),</span>
                      <span class="n">choices</span><span class="o">=</span><span class="n">CHIMERA_LIST</span><span class="p">,</span>
                      <span class="n">default</span><span class="o">=</span><span class="n">CHIMERA_I222</span><span class="p">,</span>
                      <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Symmetry&quot;</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;https://scipion-em.github.io/docs/release-2.0.0/docs/developer/symmetries.html?highlight=symmetry&quot;</span>
                           <span class="s2">&quot;Symmetry for a description of the symmetry groups &quot;</span>
                           <span class="s2">&quot;format in CHIMERA.</span><span class="se">\n</span><span class="s2">&quot;</span>
                           <span class="s2">&quot;If no symmetry is present, use _c1_.&quot;</span>
                           <span class="s1">&#39;More information: </span><span class="se">\n</span><span class="s1">&#39;</span>
                           <span class="s1">&#39;https://www.cgl.ucsf.edu/chimera/current/docs/UsersGuide/midas/sym.html&#39;</span>
                      <span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;symmetryOrder&#39;</span><span class="p">,</span> <span class="n">IntParam</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="p">((</span><span class="s1">&#39;mapOrModel==</span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="mi">1</span><span class="p">)</span> <span class="ow">and</span>
                                 <span class="s1">&#39;applySymmetry==True&#39;</span> <span class="ow">and</span>
                                 <span class="p">(</span><span class="s1">&#39;symmetryGroup&lt;=</span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">SYM_DIHEDRAL_X</span><span class="p">)),</span>
                      <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Symmetry Order&#39;</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Select the order of cyclic or dihedral symmetry.&#39;</span><span class="p">)</span>

        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputPdbFiles&#39;</span><span class="p">,</span> <span class="n">MultiPointerParam</span><span class="p">,</span>
                      <span class="n">pointerClass</span><span class="o">=</span><span class="s2">&quot;AtomStruct&quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Other atomic structures&#39;</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;In case you need to load more PDBx/mmCIF files, &quot;</span>
                           <span class="s2">&quot;you can load them here. This files will NOT be used &quot;</span>
                           <span class="s2">&quot;to create the subtrahend&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s2">&quot;filterToApplyToDiffMap&quot;</span><span class="p">,</span> <span class="n">EnumParam</span><span class="p">,</span>
                      <span class="n">expertLevel</span><span class="o">=</span><span class="n">LEVEL_ADVANCED</span><span class="p">,</span>
                      <span class="n">choices</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">CHIMERA_FILTERS</span><span class="p">,</span>
                      <span class="n">display</span><span class="o">=</span><span class="n">EnumParam</span><span class="o">.</span><span class="n">DISPLAY_HLIST</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Filter to apply to the differential map&quot;</span><span class="p">,</span>
                      <span class="n">default</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Choose the filter to clean the background noise &quot;</span>
                           <span class="s2">&quot;of the differential map.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s2">&quot;widthFilter&quot;</span><span class="p">,</span> <span class="n">FloatParam</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;filterToApplyToDiffMap&#39;</span><span class="p">,</span>
                      <span class="n">expertLevel</span><span class="o">=</span><span class="n">LEVEL_ADVANCED</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Gaussian filter width&quot;</span><span class="p">,</span>
                      <span class="n">default</span><span class="o">=</span><span class="mf">1.5</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Set the width of the Gaussian filter.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addSection</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="s1">&#39;Help&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addLine</span><span class="p">(</span><span class="s2">&quot;Step 1:</span><span class="se">\n</span><span class="s2">In the sequence window your target &quot;</span>
                         <span class="s2">&quot;sequence (and other additional sequences that you &quot;</span>
                         <span class="s2">&quot;want to use in  the alignment) will appear aligned to &quot;</span>
                         <span class="s2">&quot;the template&#39;s sequence. Select in the sequence window &quot;</span>
                         <span class="s2">&quot;menu:</span><span class="se">\n</span><span class="s2">Structure -&gt; Modeller (homology)...;</span><span class="se">\n</span><span class="s2">A new  &quot;</span>
                         <span class="s2">&quot;window for Comparative Modeling with Modeller will &quot;</span>
                         <span class="s2">&quot;appear. Select your specific sequence as the sequence &quot;</span>
                         <span class="s2">&quot;to be modeled (target), and the input atomic structure&quot;</span>
                         <span class="o">+</span> <span class="s1">&#39;&#39;&#39;</span>
<span class="s1">        used as template for modeling. Select Run Modeller via web service </span>
<span class="s1">        and write the Modeller license key supplied (Academic users can </span>
<span class="s1">        register free of charge to receive a license key). Finally, press OK.</span>
<span class="s1">        </span><span class="se">\n</span><span class="s1">WAITING TIME: (you may see the status of your job in chimera main </span>
<span class="s1">        window, lower left corner.)</span><span class="se">\n\n</span><span class="s1">Step 2:</span><span class="se">\n</span><span class="s1">When the process finished, </span>
<span class="s1">        5 models will </span>
<span class="s1">        be automatically superimposed onto the template and model scores</span>
<span class="s1">        will appear in Modeller Results window. In Chimera main menu -&gt; </span>
<span class="s1">        Favorites -&gt; Model panel will show you: #0 (coordinate axes); #1 (</span>
<span class="s1">        template); #2.1 to 2.5 (models).Choose the one you like the best, </span>
<span class="s1">        for example model #2.1. To save it in Scipion, we need to change the </span>
<span class="s1">        model ID. In Chimera main menu: Favorites -&gt; Command Line, write </span>
<span class="s1">        *combine #2.1 model #3 close t*. Then, you will see in Model panel </span>
<span class="s1">        that selected model #2.1 renamed to combination with ID #3. Save it </span>
<span class="s1">        as first guess in Scipion by executing the Chimera command </span>
<span class="s1">        *scipionwrite [model #n]*. In our example *scipionwrite model #3*.</span><span class="se">\n</span><span class="s1"> </span>
<span class="s1">        When you use the command line scipionwrite, the Chimera session will </span>
<span class="s1">        be saved by default. Additionally, you can save the Chimera session </span>
<span class="s1">        whenever you want by executing the command *scipionss*. You will be </span>
<span class="s1">        able to restore the saved session by using the protocol chimera </span>
<span class="s1">        restore session (SCIPION menu: Tools/Calculators/chimera restore </span>
<span class="s1">        session). Once you have save your favorite model you can press </span>
<span class="s1">        Quit in the Modeller Results window.&#39;&#39;&#39;</span><span class="p">)</span>

    <span class="c1"># --------------------------- INSERT steps functions --------------------</span>

    <span class="k">def</span> <span class="nf">_insertAllSteps</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;prerequisitesStep&#39;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;runChimeraStep&#39;</span><span class="p">)</span>
        <span class="c1"># self._insertFunctionStep(&#39;createOutput&#39;)</span>


<div class="viewcode-block" id="ChimeraSubtractionMaps.prerequisitesStep"><a class="viewcode-back" href="../../../api/chimera/chimera.protocols.protocol_subtraction_maps.html#chimera.protocols.protocol_subtraction_maps.ChimeraSubtractionMaps.prerequisitesStep">[docs]</a>    <span class="k">def</span> <span class="nf">prerequisitesStep</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="c1"># get the reference map</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">vol</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputVolume</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">fnVolName</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">abspath</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">vol</span><span class="o">.</span><span class="n">getFileName</span><span class="p">())</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Reference map:</span><span class="se">\n</span><span class="s2"> </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">fnVolName</span><span class="p">)</span>

        <span class="c1"># clean form</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mapOrModel</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span> <span class="c1"># 0 -&gt; map</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">subVol</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputVolume2</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">subVolName</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">abspath</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">subVol</span><span class="o">.</span><span class="n">getFileName</span><span class="p">())</span>
            <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Map to self.subVolubtract:</span><span class="se">\n</span><span class="s2"> </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">subVolName</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span> <span class="c1"># 1 -&gt; ATOMIC STRUCT</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">atomStruct</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">pdbFileToBeRefined</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">atomStructName</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">abspath</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">atomStruct</span><span class="o">.</span><span class="n">getFileName</span><span class="p">())</span>
            <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Map to subtract generated from the atomic structure:</span><span class="se">\n</span><span class="s2"> </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span>
                  <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">atomStructName</span><span class="p">)</span></div>

<div class="viewcode-block" id="ChimeraSubtractionMaps.runChimeraStep"><a class="viewcode-back" href="../../../api/chimera/chimera.protocols.protocol_subtraction_maps.html#chimera.protocols.protocol_subtraction_maps.ChimeraSubtractionMaps.runChimeraStep">[docs]</a>    <span class="k">def</span> <span class="nf">runChimeraStep</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="c1"># building script file including the coordinate axes and the input</span>
        <span class="c1"># volume with samplingRate and Origin information</span>
        <span class="n">f</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">_getTmpPath</span><span class="p">(</span><span class="n">chimeraScriptFileName</span><span class="p">),</span> <span class="s2">&quot;w&quot;</span><span class="p">)</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;from chimera import runCommand</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>

        <span class="c1"># # create coherent header</span>
        <span class="c1"># createScriptFile(1,  # model id pdb</span>
        <span class="c1">#                  1,  # model id 3D map</span>
        <span class="c1">#                  self._getExtraPath(chimeraPdbTemplateFileName),</span>
        <span class="c1">#                  self._getExtraPath(chimeraMapTemplateFileName),</span>
        <span class="c1">#                  f,</span>
        <span class="c1">#                  self._getExtraPath(sessionFile),</span>
        <span class="c1">#                  )</span>

        <span class="c1"># building coordinate axes</span>
        <span class="n">dim</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">vol</span><span class="o">.</span><span class="n">getDim</span><span class="p">()[</span><span class="mi">0</span><span class="p">]</span>
        <span class="n">sampling</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">vol</span><span class="o">.</span><span class="n">getSamplingRate</span><span class="p">()</span>
        <span class="n">bildFileName</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">abspath</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">_getTmpPath</span><span class="p">(</span><span class="s2">&quot;axis_input.bild&quot;</span><span class="p">))</span>
        <span class="n">Chimera</span><span class="o">.</span><span class="n">createCoordinateAxisFile</span><span class="p">(</span><span class="n">dim</span><span class="p">,</span> <span class="n">bildFileName</span><span class="o">=</span><span class="n">bildFileName</span><span class="p">,</span>
                                         <span class="n">sampling</span><span class="o">=</span><span class="n">sampling</span><span class="p">)</span>
        <span class="c1"># input volume</span>
        <span class="n">chimeraModelId</span> <span class="o">=</span> <span class="mi">0</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;open </span><span class="si">%s</span><span class="s2"> model-ID </span><span class="si">%d</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">bildFileName</span><span class="p">,</span> <span class="n">chimeraModelId</span><span class="p">))</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;cofr 0,0,0&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>  <span class="c1"># set center of coordinates</span>
        <span class="c1"># origin coordinates</span>
        <span class="n">chimeraModelIdM</span> <span class="o">=</span> <span class="n">chimeraModelId</span> <span class="o">+</span> <span class="mi">1</span> <span class="c1"># 1, Minuend</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;open </span><span class="si">%s</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">fnVolName</span><span class="p">)</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;volume #</span><span class="si">%d</span><span class="s2"> style surface voxelSize </span><span class="si">%f</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span>
                <span class="o">%</span> <span class="p">(</span><span class="n">chimeraModelIdM</span><span class="p">,</span> <span class="n">sampling</span><span class="p">))</span>
        <span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">z</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">vol</span><span class="o">.</span><span class="n">getShiftsFromOrigin</span><span class="p">()</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;volume #</span><span class="si">%d</span><span class="s2"> origin </span><span class="si">%0.2f</span><span class="s2">,</span><span class="si">%0.2f</span><span class="s2">,</span><span class="si">%0.2f</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span>
                <span class="o">%</span> <span class="p">(</span><span class="n">chimeraModelIdM</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">z</span><span class="p">))</span>
        <span class="c1"># input vol with its origin coordinates</span>
        <span class="n">chimeraModelIdS</span> <span class="o">=</span> <span class="n">chimeraModelIdM</span> <span class="o">+</span> <span class="mi">1</span>  <span class="c1"># 2 Subtrahend</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mapOrModel</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;open </span><span class="si">%s</span><span class="s2"> model-ID </span><span class="si">%d</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span>
                    <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">subVolName</span><span class="p">,</span> <span class="n">chimeraModelIdS</span><span class="p">))</span>
            <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;volume #</span><span class="si">%d</span><span class="s2"> style surface voxelSize </span><span class="si">%f</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span>
                    <span class="o">%</span> <span class="p">(</span><span class="n">chimeraModelIdS</span><span class="p">,</span> <span class="n">sampling</span><span class="p">))</span>
            <span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">z</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">subVol</span><span class="o">.</span><span class="n">getShiftsFromOrigin</span><span class="p">()</span>
            <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;volume #</span><span class="si">%d</span><span class="s2"> origin </span><span class="si">%0.2f</span><span class="s2">,</span><span class="si">%0.2f</span><span class="s2">,</span><span class="si">%0.2f</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span>
                        <span class="o">%</span> <span class="p">(</span><span class="n">chimeraModelIdS</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">z</span><span class="p">))</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;open </span><span class="si">%s</span><span class="s2"> model-ID </span><span class="si">%d</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span>
                    <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">atomStructName</span><span class="p">,</span> <span class="n">chimeraModelIdS</span><span class="p">))</span>
            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">selectChain</span> <span class="o">==</span> <span class="kc">True</span><span class="p">:</span>
                <span class="c1"># read PDB</span>
                <span class="n">aSH</span> <span class="o">=</span> <span class="n">AtomicStructHandler</span><span class="p">()</span>
                <span class="n">aSH</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">atomStructName</span><span class="p">)</span>
                <span class="n">aSH</span><span class="o">.</span><span class="n">getStructure</span><span class="p">()</span>
                <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;aSH: &quot;</span><span class="p">,</span> <span class="n">aSH</span><span class="p">)</span>
                <span class="n">modelsLength</span><span class="p">,</span> <span class="n">modelsFirstResidue</span> <span class="o">=</span> <span class="n">aSH</span><span class="o">.</span><span class="n">getModelsChains</span><span class="p">()</span>
                <span class="c1"># model and chain selected</span>
                <span class="n">chain</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">selectStructureChain</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">selectedModel</span> <span class="o">=</span> <span class="n">chain</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">selectedChain</span> <span class="o">=</span> <span class="n">chain</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span>
                <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Selected chain: </span><span class="si">%s</span><span class="s2"> from model: </span><span class="si">%s</span><span class="s2"> from structure: </span><span class="si">%s</span><span class="s2">&quot;</span> \
                        <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">selectedChain</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">selectedModel</span><span class="p">,</span>
                            <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">atomStructName</span><span class="p">)))</span>

                <span class="c1"># get pdb sequence</span>
                <span class="c1">#         if (self.pdbFileToBeRefined.get() is not None) and \</span>
                <span class="c1">#                 (self.selectStructureChain.get() is not None):</span>
                <span class="c1">#             import json</span>
                <span class="c1">#             chainIdDict = json.loads(self.inputStructureChain.get())</span>

                <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;&#39;)&quot;</span><span class="p">)</span>

        <span class="n">chimeraModelIdD</span> <span class="o">=</span> <span class="n">chimeraModelIdS</span> <span class="o">+</span> <span class="mi">1</span>  <span class="c1"># 3 Diff map</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;vop subtract #</span><span class="si">%d</span><span class="s2"> #</span><span class="si">%d</span><span class="s2"> modelId #</span><span class="si">%d</span><span class="s2"> &quot;</span>
                <span class="s2">&quot;minRMS true onGrid #</span><span class="si">%d</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span>
                <span class="o">%</span> <span class="p">(</span><span class="n">chimeraModelIdM</span><span class="p">,</span> <span class="n">chimeraModelIdS</span><span class="p">,</span>
                   <span class="n">chimeraModelIdD</span><span class="p">,</span> <span class="n">chimeraModelIdM</span><span class="p">))</span>
        <span class="n">chimeraModelIdF</span> <span class="o">=</span> <span class="n">chimeraModelIdD</span> <span class="o">+</span> <span class="mi">1</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">filterToApplyToDiffMap</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;vop gaussian #</span><span class="si">%d</span><span class="s2"> sd </span><span class="si">%0.3f</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span>
                    <span class="o">%</span> <span class="p">(</span><span class="n">chimeraModelIdF</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">widthFilter</span><span class="o">.</span><span class="n">get</span><span class="p">()))</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;runCommand(&#39;vop laplacian #</span><span class="si">%d</span><span class="s2">&#39;)</span><span class="se">\n</span><span class="s2">&quot;</span>
                    <span class="o">%</span> <span class="p">(</span><span class="n">chimeraModelIdF</span><span class="p">))</span></div>

<span class="c1"># xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx</span>
<span class="c1">#             if self.inputVolumes is not None:</span>
<span class="c1">#                 f.write(&quot;runCommand(&#39;open %s model-ID %d&#39;)\n&quot; %</span>
<span class="c1">#                         (self.subVolName, chimeraModelId))</span>
<span class="c1">#</span>
<span class="c1">#</span>
<span class="c1">#</span>
<span class="c1">#                 pdbModelCounter += 1</span>
<span class="c1">#                 for vol in self.inputVolumes:</span>
<span class="c1">#                     f.write(&quot;runCommand(&#39;open %s&#39;)\n&quot; % vol.get().getFileName())</span>
<span class="c1">#                     pdbModelCounter += 1</span>
<span class="c1">#</span>
<span class="c1">#             pdbFileToBeRefined = self.pdbFileToBeRefined.get()</span>
<span class="c1">#             f.write(&quot;runCommand(&#39;open %s&#39;)\n&quot; % os.path.abspath(</span>
<span class="c1">#                 pdbFileToBeRefined.getFileName()))</span>
<span class="c1">#             if pdbFileToBeRefined.hasOrigin():</span>
<span class="c1">#                 x, y, z = (pdbFileToBeRefined.getOrigin().getShifts())</span>
<span class="c1">#                 f.write(&quot;runCommand(&#39;move %0.2f,%0.2f,%0.2f model #%d &quot;</span>
<span class="c1">#                         &quot;coord #0&#39;)\n&quot; % (x, y, z, pdbModelCounter))</span>
<span class="c1">#</span>
<span class="c1">#             # Alignment of sequence and structure</span>
<span class="c1">#             if (hasattr(self, &#39;inputSequence&#39;) and</span>
<span class="c1">#                     hasattr(self, &#39;inputStructureChain&#39;)):</span>
<span class="c1">#                 if (self.inputSequence.get() is not None and</span>
<span class="c1">#                         self.inputStructureChain.get() is not None):</span>
<span class="c1">#                     models = self.structureHandler.getModelsChains()</span>
<span class="c1">#                     if len(models) &gt; 1:</span>
<span class="c1">#                         f.write(&quot;runCommand(&#39;select #%d.%s:.%s&#39;)\n&quot;</span>
<span class="c1">#                                 % (pdbModelCounter, str(self.selectedModel),</span>
<span class="c1">#                                    str(self.selectedChain)))</span>
<span class="c1">#                     else:</span>
<span class="c1">#                         f.write(&quot;runCommand(&#39;select #%d:.%s&#39;)\n&quot;</span>
<span class="c1">#                                 % (pdbModelCounter, str(self.selectedChain)))</span>
<span class="c1">#                     # f.write(&quot;runCommand(&#39;sequence selection&#39;)\n&quot;) # To open</span>
<span class="c1">#                     # only the</span>
<span class="c1">#                     # sequence of</span>
<span class="c1">#                     # the selected</span>
<span class="c1">#                     # chain</span>
<span class="c1">#                     if self._getOutFastaSequencesFile is not None:</span>
<span class="c1">#                         alignmentFile = self._getOutFastaSequencesFile()</span>
<span class="c1">#                         f.write(&quot;runCommand(&#39;open %s&#39;)\n&quot; % alignmentFile)</span>
<span class="c1">#</span>
<span class="c1">#             # other pdb files</span>
<span class="c1">#             pdbModelCounter += 1</span>
<span class="c1">#             for pdb in self.inputPdbFiles:</span>
<span class="c1">#                 f.write(&quot;runCommand(&#39;open %s&#39;)\n&quot; % os.path.abspath(pdb.get(</span>
<span class="c1">#                 ).getFileName()))</span>
<span class="c1">#                 if pdb.get().hasOrigin():</span>
<span class="c1">#                     x, y, z = pdb.get().getOrigin().getShifts()</span>
<span class="c1">#                     f.write(&quot;runCommand(&#39;move %0.2f,%0.2f,%0.2f model #%d &quot;</span>
<span class="c1">#                             &quot;coord #0&#39;)\n&quot; % (x, y, z, pdbModelCounter))</span>
<span class="c1">#                 # TODO: Check this this this this this this</span>
<span class="c1">#                 pdbModelCounter += 1</span>
<span class="c1">#</span>
<span class="c1">#             # run the text:</span>
<span class="c1">#             if len(self.extraCommands.get()) &gt; 2:</span>
<span class="c1">#                 f.write(self.extraCommands.get())</span>
<span class="c1">#                 args = &quot; --nogui --script &quot; + self._getTmpPath(</span>
<span class="c1">#                     chimeraScriptFileName)</span>
<span class="c1">#             else:</span>
<span class="c1">#                 args = &quot; --script &quot; + self._getTmpPath(chimeraScriptFileName)</span>
<span class="c1">#</span>
<span class="c1">#             f.close()</span>
<span class="c1">#</span>
<span class="c1">#             self._log.info(&#39;Launching: &#39; + Plugin.getProgram() + &#39; &#39; + args)</span>
<span class="c1">#</span>
<span class="c1">#             # run in the background</span>
<span class="c1">#             Chimera.runProgram(Plugin.getProgram(), args)</span>
<span class="c1">#</span>
<span class="c1">#         # get pdb sequence</span>
<span class="c1">#         if (self.pdbFileToBeRefined.get() is not None) and \</span>
<span class="c1">#                 (self.selectStructureChain.get() is not None):</span>
<span class="c1">#             import json</span>
<span class="c1">#             chainIdDict = json.loads(self.inputStructureChain.get())</span>
<span class="c1">#</span>
<span class="c1">#         # read PDB</span>
<span class="c1">#         self.structureHandler = AtomicStructHandler()</span>
<span class="c1">#         fileName = os.path.abspath(self.pdbFileToBeRefined.get(</span>
<span class="c1">#         ).getFileName())</span>
<span class="c1">#         self.structureHandler.read(fileName)</span>
<span class="c1">#</span>
<span class="c1">#         # get sequence of structure chain with id chainId (selected by the user)</span>
<span class="c1">#         self.selectedModel = chainIdDict[&#39;model&#39;]</span>
<span class="c1">#         self.selectedChain = chainIdDict[&#39;chain&#39;]</span>
<span class="c1">#         # self.selectedModel = chainId.split(&#39;,&#39;)[0].split(&#39;:&#39;)[1].strip()</span>
<span class="c1">#         # self.selectedChain = chainId.split(&#39;,&#39;)[1].split(&#39;:&#39;)[1].strip()</span>
<span class="c1">#         print(&quot;Selected chain: %s from model: %s from structure: %s&quot; \</span>
<span class="c1">#               % (self.selectedChain, self.selectedModel,</span>
<span class="c1">#                  os.path.basename(fileName)))</span>
<span class="c1">#</span>
<span class="c1">#         # Bio.Seq.Seq object</span>
<span class="c1">#         structureSeq = self.structureHandler.getSequenceFromChain(</span>
<span class="c1">#             self.selectedModel, self.selectedChain)</span>
<span class="c1">#</span>
<span class="c1">#         # obtain a seqID for our PDB sequence</span>
<span class="c1">#         structSeqID = self.structureHandler.getFullID(self.selectedModel,</span>
<span class="c1">#                                                       self.selectedChain)</span>
<span class="c1">#         # END PDB sequence</span>
<span class="c1">#</span>
<span class="c1">#         # start user imported target sequence</span>
<span class="c1">#         # get target sequence imported by the user</span>
<span class="c1">#         userSeq = self.inputSequence.get()  # SEQ object from Scipion</span>
<span class="c1">#         targetSeqID = userSeq.getId()  # ID associated to SEQ object (str)</span>
<span class="c1">#         userSequence = userSeq.getSequence()  # sequence associated to</span>
<span class="c1">#         # that SEQ object (str)</span>
<span class="c1">#         # transformation of this sequence (str) in a Bio.Seq.Seq object:</span>
<span class="c1">#         seqHandler = SequenceHandler(userSequence,</span>
<span class="c1">#                                      isAminoacid=userSeq.getIsAminoacids())</span>
<span class="c1">#         targetSeq = seqHandler._sequence  # Bio.Seq.Seq object</span>
<span class="c1">#</span>
<span class="c1">#         # creation of Dic of IDs and sequences</span>
<span class="c1">#         SeqDic = OrderedDict()</span>
<span class="c1">#         SeqDic[structSeqID] = structureSeq</span>
<span class="c1">#         SeqDic[targetSeqID] = targetSeq</span>
<span class="c1">#</span>
<span class="c1">#         # align sequences and save them to disk, -this will be chimera input-</span>
<span class="c1">#         # get all sequences in a fasta file</span>
<span class="c1">#         inFile = self._getInFastaSequencesFile()</span>
<span class="c1">#         outFile = self._getOutFastaSequencesFile()</span>
<span class="c1">#</span>
<span class="c1">#         # get the alignment of sequences</span>
<span class="c1">#         if not self.additionalSequencesToAlign.get():</span>
<span class="c1">#             saveFileSequencesToAlign(SeqDic, inFile)</span>
<span class="c1">#             self.inputSequencesToAlign = None</span>
<span class="c1">#             if self.inputProgramToAlign1.get() == \</span>
<span class="c1">#                     self.ProgramToAlign1.index(&#39;Bio.pairwise2&#39;):</span>
<span class="c1">#                 # Only the two first sequences will be included in the alignment</span>
<span class="c1">#                 self.alignment = alignBioPairwise2Sequences(</span>
<span class="c1">#                     structSeqID, structureSeq,</span>
<span class="c1">#                     targetSeqID, targetSeq,</span>
<span class="c1">#                     outFile)</span>
<span class="c1">#             else:</span>
<span class="c1">#                 # All the sequences will be included in the alignment</span>
<span class="c1">#                 if self.inputProgramToAlign1.get() == \</span>
<span class="c1">#                         self.ProgramToAlign1.index(&#39;Clustal Omega&#39;):</span>
<span class="c1">#                     cline = alignClustalSequences(inFile, outFile)</span>
<span class="c1">#                 else:</span>
<span class="c1">#                     cline = alignMuscleSequences(inFile, outFile)</span>
<span class="c1">#                 args = &#39;&#39;</span>
<span class="c1">#                 self.runJob(cline, args)</span>
<span class="c1">#         else:</span>
<span class="c1">#             # if there are additional sequences imported by the user</span>
<span class="c1">#             if self.inputSequencesToAlign is not None:</span>
<span class="c1">#                 for seq in self.inputSequencesToAlign:</span>
<span class="c1">#                     seq = seq.get()</span>
<span class="c1">#                     ID = seq.getId()</span>
<span class="c1">#                     sequence = seq.getSequence()</span>
<span class="c1">#                     seqHandler = SequenceHandler(sequence,</span>
<span class="c1">#                                                  isAminoacid=seq.getIsAminoacids())</span>
<span class="c1">#                     otherSeq = seqHandler._sequence  # Bio.Seq.Seq object</span>
<span class="c1">#                     SeqDic[ID] = otherSeq</span>
<span class="c1">#</span>
<span class="c1">#             # align sequences and save them to disk, -this will be chimera input-</span>
<span class="c1">#             # get all sequences in a fasta file</span>
<span class="c1">#             # inFile = self._getInFastaSequencesFile()</span>
<span class="c1">#             saveFileSequencesToAlign(SeqDic, inFile)</span>
<span class="c1">#             # outFile = self._getOutFastaSequencesFile()</span>
<span class="c1">#</span>
<span class="c1">#             # All the sequences will be included in the alignment</span>
<span class="c1">#             if self.inputProgramToAlign2.get() == self.ProgramToAlign2.index(</span>
<span class="c1">#                     &#39;Clustal Omega&#39;):</span>
<span class="c1">#                 cline = alignClustalSequences(inFile, outFile)</span>
<span class="c1">#             else:</span>
<span class="c1">#                 cline = alignMuscleSequences(inFile, outFile)</span>
<span class="c1">#             args = &#39;&#39;</span>
<span class="c1">#             self.runJob(cline, args)</span>
<span class="c1">#</span>
<span class="c1">#</span>
<span class="c1">#</span>

    <span class="k">def</span> <span class="nf">_validate</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="n">errors</span> <span class="o">=</span> <span class="nb">super</span><span class="p">(</span><span class="n">ChimeraSubtractionMaps</span><span class="p">,</span> <span class="bp">self</span><span class="p">)</span><span class="o">.</span><span class="n">_validate</span><span class="p">()</span>

        <span class="c1"># Check that the input volume exists</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputVolume</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">errors</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Error: You should provide a map.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>

        <span class="c1"># Check that an additional map or an atomic structure exist</span>
        <span class="k">if</span> <span class="p">(</span><span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputVolume2</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="ow">and</span>
                <span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">pdbFileToBeRefined</span><span class="o">.</span><span class="n">get</span><span class="p">()):</span>
            <span class="n">errors</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Error: You should provide an additional &quot;</span>
                          <span class="s2">&quot;map or an atomic structure.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>

        <span class="k">return</span> <span class="n">errors</span></div>
</pre></div>

           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Scipion team

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>