<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>6. Get the initial model with ModelAngelo &mdash; Scipion 3.0.0 documentation</title>
      <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../../../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../../../../_static/custom.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../../../../" id="documentation_options" src="../../../../../_static/documentation_options.js"></script>
        <script src="../../../../../_static/jquery.js"></script>
        <script src="../../../../../_static/underscore.js"></script>
        <script src="../../../../../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../../../../../_static/doctools.js"></script>
    <script src="../../../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../../search.html" />
    <link rel="next" title="7. Comparison of ModelAngelo generated models with the published traced atomic structure" href="030-compareModel.html" />
    <link rel="prev" title="5. General workflow followed in this tutorial to predict and compare ModelAngelo structures" href="012-generalWorkflow.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../../../../../index.html" class="icon icon-home"> Scipion
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/how-to-install.html">Installing Scipion</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">User documentation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html#image-processing-resources">Image processing resources</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../user-documentation.html#tutorials">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#single-particle-tutorials">Single particle tutorials</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#tomography-tutorials">Tomography tutorials</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../../user-documentation.html#model-building">Model building</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../Homology/index.html">Model Building by Homology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../Alphafold/index.html">Build initial model with Alphafold</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="index.html">Build initial model with ModelAngelo</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="index.html#contents">Contents</a></li>
<li class="toctree-l4"><a class="reference internal" href="index.html#appendices">Appendices</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#flexibility-hub">Flexibility hub</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html#processing-how-to-s">Processing How To’s</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities.html">On-the-fly processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/customize-report.html">Summary Reports</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/scipion-API.html">Scipion API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/plugins-API.html">Plugins API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/xmipp-API.html">Xmipp API</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../misc/contact-us.html">Contact Us</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Licences</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../license.html">Licenses</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../../index.html">Scipion</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content style-external-links">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../../../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../../../user-documentation.html">Graphical interface manuals</a> &raquo;</li>
          <li><a href="index.html">Predicting initial models with ModelAngelo</a> &raquo;</li>
      <li><span class="section-number">6. </span>Get the initial model with ModelAngelo</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../../../../_sources/docs/user/tutorials/modelBuilding/ModelAngelo/020-initialModel.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="get-the-initial-model-with-modelangelo">
<span id="getinitialmodel"></span><h1><span class="section-number">6. </span>Get the initial model with ModelAngelo<a class="headerlink" href="#get-the-initial-model-with-modelangelo" title="Permalink to this heading"></a></h1>
<p>In this section of the Modelangelo tutorial we cover steps 1, 2, 3 and 4 of the small workflow shown before (<code class="xref std std-numref docutils literal notranslate"><span class="pre">model_building_fig1_workflow</span></code>) addressed
to get Modelangelo prediction structures of the human TACAN protein (isoform 1).</p>
<p>ModelAngelo is able to predict the atomic structure of a 3D-EM map with or without protein sequences <span id="id1">[<a class="reference internal" href="../Homology/z010-references.html#id34" title="Kiarash Jamali, Dari Kimanius, and Sjors HW Scheres. A graph neural network approach to automated model building in cryo-em maps. In International Conference on Learning Representations. 2023. URL: https://openreview.net/forum?id=65XDF_nwI61.">Jamali <em>et al.</em>, 2023</a>]</span>. We are going to predict the tracing of the human TACAN protein in both cases,
a) using the protein sequence, and b) in absence of any protein sequence.</p>
<div class="section" id="modelangelo-initial-model-using-3d-map-and-protein-sequence-as-starting-inputs">
<h2><span class="section-number">6.1. </span>Modelangelo initial model using 3D map and protein sequence as starting inputs<a class="headerlink" href="#modelangelo-initial-model-using-3d-map-and-protein-sequence-as-starting-inputs" title="Permalink to this heading"></a></h2>
<p>As described in <a class="reference internal" href="../Homology/040-importInputData.html#importinputdata"><span class="std std-ref">Import Input Data</span></a> section, we are going to take advantage of <em>Scipion</em> software framework to import
the 3D map and the sequence of the human TACAN protein isoform 1 using protocols <strong>import volumes</strong> and <strong>import sequence</strong>. Details about the parameters
of these two protocols are shown in Appendices <a class="reference internal" href="../Homology/a100-importvolume.html#app-importvolume"><span class="std std-ref">Import volumes</span></a> and <a class="reference internal" href="../Homology/a090-importSequence.html#app-importsequence"><span class="std std-ref">Import sequence</span></a>, respectively.</p>
<p>First open the <strong>import volumes</strong> protocol (<a class="reference internal" href="#model-building-modelangelo-import-volume"><span class="std std-numref">Fig. 6.17</span></a>), fill in the form selecting EMDBid (1) to
import the map from the database, and include the ID number for the TACAN homodimer (2). Finally, execute the protocol.</p>
<div class="figure align-center" id="model-building-modelangelo-import-volume">
<a class="reference internal image-reference" href="../../../../../_images/Fig2_import_volume.svg"><img alt="Importing TACAN 3D volume in *Scipion* (workflow step 1; :numref:`model_building_modelangelo_fig1_workflow`)." src="../../../../../_images/Fig2_import_volume.svg" width="95.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 6.17 </span><span class="caption-text">Importing TACAN 3D volume in <em>Scipion</em> (workflow step 1; <a class="reference internal" href="012-generalWorkflow.html#model-building-modelangelo-fig1-workflow"><span class="std std-numref">Fig. 5.6</span></a>).</span><a class="headerlink" href="#model-building-modelangelo-import-volume" title="Permalink to this image"></a></p>
</div>
<p>Then you may visualize the volume with <em>ChimeraX</em> <span id="id2">[<a class="reference internal" href="../Homology/z010-references.html#id8" title="T. D. Goddard, C. C. Huang, E. C. Meng, E. F. Pettersen, G. S. Couch, J. H. Morris, and T. E. Ferrin. UCSF ChimeraX: Meeting modern challenges in visualization and analysis. Protein Sci., 27(1):14–25, 01 2018.">Goddard <em>et al.</em>, 2018</a>]</span> clicking <strong>Analyze Results</strong>. The 3D map of the TACAN homodimer
(<a class="reference internal" href="011-problem2solve.html#model-building-tacan-figure"><span class="std std-numref">Fig. 4.11</span></a>, A) appears inserted in the lipid nanodisc on the right side of the coordinate axes.</p>
<p>The sequence of the human TACAN protein isoform 1 will be downloaded from <a class="reference external" href="https://www.uniprot.org">UniprotKB</a>. First of all,
open the form of <strong>import sequence</strong> protocol (<a class="reference internal" href="#model-building-modelangelo-import-sequence"><span class="std std-numref">Fig. 6.18</span></a>), then complete the form to download
TACAN (1) aminoacid sequence (2) with <em>UniprotKB</em> (3) accession code <a class="reference external" href="https://www.uniprot.org/uniprot/Q9BXJ8">Q9BXJ8</a>.
Execute the process and visualize the sequence in a text editor. The sequence will appear in <em>fasta</em> format.</p>
<div class="figure align-center" id="model-building-modelangelo-import-sequence">
<a class="reference internal image-reference" href="../../../../../_images/Fig3_import_sequence.svg"><img alt="Importing TACAN *UniprotKB* sequence in *Scipion* (workflow step 2; :numref:`model_building_modelangelo_fig1_workflow`)." src="../../../../../_images/Fig3_import_sequence.svg" width="95.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 6.18 </span><span class="caption-text">Importing TACAN <em>UniprotKB</em> sequence in <em>Scipion</em> (workflow step 2; <a class="reference internal" href="012-generalWorkflow.html#model-building-modelangelo-fig1-workflow"><span class="std std-numref">Fig. 5.6</span></a>).</span><a class="headerlink" href="#model-building-modelangelo-import-sequence" title="Permalink to this image"></a></p>
</div>
<p>Once we have map and sequence we can run <strong>modelangelo - model builder</strong> protocol (<a class="reference internal" href="#model-building-modelangelo-prediction"><span class="std std-numref">Fig. 6.19</span></a>). Details about the parameters
of this protocol are shown in Appendix <a class="reference internal" href="a010-modelAngeloPredictionProtocol.html#app-modelangelopredictionprotocol"><span class="std std-ref">ModelAngelo</span></a>. Complete the form selecting the GPU
that you plan to use (1), the 3D map (2) and the <em>UniprotKB</em> sequence (3) of the TACAN protein. Finally, execute the process.</p>
<div class="figure align-center" id="model-building-modelangelo-prediction">
<a class="reference internal image-reference" href="../../../../../_images/Fig4_modelangelo.svg"><img alt="Completing the *Scipion* protocol **modelangelo - model builder** to predict the atomic structure of TACAN protein, isoform 1 (workflow step 3; :numref:`model_building_modelangelo_fig1_workflow`)." src="../../../../../_images/Fig4_modelangelo.svg" width="95.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 6.19 </span><span class="caption-text">Completing the <em>Scipion</em> protocol <strong>modelangelo - model builder</strong> to predict the atomic structure of TACAN protein,
isoform 1 (workflow step 3; <a class="reference internal" href="012-generalWorkflow.html#model-building-modelangelo-fig1-workflow"><span class="std std-numref">Fig. 5.6</span></a>).</span><a class="headerlink" href="#model-building-modelangelo-prediction" title="Permalink to this image"></a></p>
</div>
<p>Then, we can visualize the predicted atomic structure based on the 3D map clicking <strong>Analyze Results</strong>. <em>ChimeraX</em>
<span id="id3">[<a class="reference internal" href="../Homology/z010-references.html#id8" title="T. D. Goddard, C. C. Huang, E. C. Meng, E. F. Pettersen, G. S. Couch, J. H. Morris, and T. E. Ferrin. UCSF ChimeraX: Meeting modern challenges in visualization and analysis. Protein Sci., 27(1):14–25, 01 2018.">Goddard <em>et al.</em>, 2018</a>]</span> viewer will open by default (<a class="reference internal" href="#model-building-modelangelo-results-1"><span class="std std-numref">Fig. 6.20</span></a>, A).</p>
<div class="figure align-center" id="model-building-modelangelo-results-1">
<a class="reference internal image-reference" href="../../../../../_images/Fig5_modelangelo_results_1.svg"><img alt="Prediction of TACAN protein (isoform 1) structure obtained using *ModelAngelo* with 3D map and sequence as inputs. **A**: Upper view of the homodimer of the human TACAN protein as it appears in *ChimeraX* GUI main window showing the b-factor value for each atom colored according AlphaFold prediction criteria. **B**: *Idem*, frontal view.  **C**, Models panel. Arrows highlight raw prediction model (1), detailed in **D**, and final postprocessed pruned model (2) detailed in **E**." src="../../../../../_images/Fig5_modelangelo_results_1.svg" width="95.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 6.20 </span><span class="caption-text">Prediction of TACAN protein (isoform 1) structure obtained using <em>ModelAngelo</em> with 3D map and sequence as inputs. <strong>A</strong>: Upper view of the homodimer of the
human TACAN protein as it appears in <em>ChimeraX</em> GUI main window showing the b-factor value for each atom colored according
AlphaFold prediction criteria. <strong>B</strong>: <em>Idem</em>, frontal view.  <strong>C</strong>, Models panel. Arrows highlight raw prediction model (1), detailed
in <strong>D</strong>, and final postprocessed pruned model (2) detailed in <strong>E</strong>.</span><a class="headerlink" href="#model-building-modelangelo-results-1" title="Permalink to this image"></a></p>
</div>
<p>As you can see (<a class="reference internal" href="#model-building-modelangelo-results-1"><span class="std std-numref">Fig. 6.20</span></a>), parts of the predicted raw model (D) have been removed in the pruned one (F).
These removed portions do not have good matches to the sequence based on a hidden Markov model alignment <span id="id4">[<a class="reference internal" href="../Homology/z010-references.html#id34" title="Kiarash Jamali, Dari Kimanius, and Sjors HW Scheres. A graph neural network approach to automated model building in cryo-em maps. In International Conference on Learning Representations. 2023. URL: https://openreview.net/forum?id=65XDF_nwI61.">Jamali <em>et al.</em>, 2023</a>]</span>.
You can observe the sequences included both in the raw and the pruned predictions simply by clicking in the upper menu of <em>ChimeraX</em> GUI -&gt; <em>Tools</em>
-&gt; <em>Sequence</em> -&gt; <em>Show Sequence Viewer</em>, and a window similar to shown in <a class="reference internal" href="#model-building-modelangelo-results-2"><span class="std std-numref">Fig. 6.21</span></a> (A)
will display all sequences.</p>
<div class="figure align-center" id="model-building-modelangelo-results-2">
<a class="reference internal image-reference" href="../../../../../_images/Fig6_modelangelo_results_2.svg"><img alt="Sequence fragments from the UniProtKB ID Q9BXJ8 traced in the 3D map. **A**: *ChimeraX* selection sequence panel. Pruned prediction sequences appear in the lower blue shadow of the panel. **B**: Clustal Omega alignment highlighting matching residues of sequences of the pruned prediction and the whole sequence of the UniProtKB protein Q9BXJ8." src="../../../../../_images/Fig6_modelangelo_results_2.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 6.21 </span><span class="caption-text">Sequence fragments from the UniProtKB ID Q9BXJ8 traced in the 3D map. <strong>A</strong>:
<em>ChimeraX</em> selection sequence panel. Pruned prediction sequences appear in the lower blue shadow of the panel. <strong>B</strong>: Clustal Omega alignment
highlighting matching residues of sequences of the pruned prediction and the whole sequence of the UniProtKB protein Q9BXJ8.</span><a class="headerlink" href="#model-building-modelangelo-results-2" title="Permalink to this image"></a></p>
</div>
<p>As observed in <a class="reference internal" href="#model-building-modelangelo-results-2"><span class="std std-numref">Fig. 6.21</span></a> (B), sequences Ad and Ae, with a higher number of residues (104 and 155, respectively),
contribute the most to the final pruned prediction. In this example, 74% of the whole Q9BXJ8 protein sequence has been covered by the pruned
prediction. The rest of the map, containing regions with lower resolution, has to be modeled manually.</p>
</div>
<div class="section" id="modelangelo-initial-model-using-only-the-3d-map-as-starting-input">
<h2><span class="section-number">6.2. </span>Modelangelo initial model using only the 3D map as starting input<a class="headerlink" href="#modelangelo-initial-model-using-only-the-3d-map-as-starting-input" title="Permalink to this heading"></a></h2>
<p>We are going to start this example running again the <strong>modelangelo - model builder</strong> protocol.
As we have seen above (<a class="reference internal" href="#model-building-modelangelo-prediction"><span class="std std-numref">Fig. 6.19</span></a>), this time we have to complete the form selecting the GPU
that you plan to use (1) and the 3D map (2), to finally execute the process (workflow step 4; <a class="reference internal" href="012-generalWorkflow.html#model-building-modelangelo-fig1-workflow"><span class="std std-numref">Fig. 5.6</span></a>).</p>
<p>After executing the protocol, we can visualize the results in a similar way as we have done before. Clicking <strong>Analyze Results</strong>, <em>ChimeraX</em>
<span id="id5">[<a class="reference internal" href="../Homology/z010-references.html#id8" title="T. D. Goddard, C. C. Huang, E. C. Meng, E. F. Pettersen, G. S. Couch, J. H. Morris, and T. E. Ferrin. UCSF ChimeraX: Meeting modern challenges in visualization and analysis. Protein Sci., 27(1):14–25, 01 2018.">Goddard <em>et al.</em>, 2018</a>]</span> viewer will open by default (<a class="reference internal" href="#model-building-modelangelo-results-3"><span class="std std-numref">Fig. 6.22</span></a>).</p>
<div class="figure align-center" id="model-building-modelangelo-results-3">
<a class="reference internal image-reference" href="../../../../../_images/Fig7_modelangelo_results_3.svg"><img alt="Prediction of TACAN protein (isoform 1) structure obtained using *ModelAngelo* with only the 3D map as input. **A**: Atomic structure generated. **B**: Models panel. Arrow highlights the unique prediction model generated." src="../../../../../_images/Fig7_modelangelo_results_3.svg" width="95.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 6.22 </span><span class="caption-text">Prediction of TACAN protein (isoform 1) structure obtained using <em>ModelAngelo</em> with only the 3D map as input. <strong>A</strong>:
Atomic structure generated. <strong>B</strong>: Models panel. Arrow highlights the unique prediction model generated.</span><a class="headerlink" href="#model-building-modelangelo-results-3" title="Permalink to this image"></a></p>
</div>
<p>Although the prediction (<a class="reference internal" href="#model-building-modelangelo-results-3"><span class="std std-numref">Fig. 6.22</span></a>, A) is aparently similar to the previous one (<a class="reference internal" href="#model-building-modelangelo-results-1"><span class="std std-numref">Fig. 6.20</span></a>, D),
some differences can be observed between the results displayed. First of all, only one prediction has been generated (B, red arrow).
This unique prediction is similar to the raw prediction obtained when the sequence is included as input also. Postprocessing or pruning depends
on the sequence matching. Then, the pruned structure thus cannot be generated. In addition, the whole structure is constituted by a high number
of small fragments that have to be manually assessed. The goodness of this prediction of sequence and structure will depend on the map resolution.
The better the resolution, the higher number of residues could be inequivocally assigned to the map density. Since the map <a class="reference external" href="https://www.ebi.ac.uk/emdb/EMD-31441">EMD-31441</a>
has already been traced (<a class="reference external" href="https://www.rcsb.org/structure/7F3U">PDB 7F3U</a>), we can <a class="reference internal" href="030-compareModel.html#compareinitialmodel"><span class="std std-ref">compare</span></a> the published structure with our ModelAngelo prediction
in the background of the map density (<a class="reference internal" href="#model-building-modelangelo-results-4"><span class="std std-numref">Fig. 6.23</span></a>) .</p>
<blockquote>
<div><div class="figure align-center" id="model-building-modelangelo-results-4">
<a class="reference internal image-reference" href="../../../../../_images/Fig8_modelangelo_results_4.svg"><img alt="Comparison of PDB ID 7F3U atomic structure (chain A, residues 217-235, blue) with ModelAngelo prediction (chain Af, residues 315-334, pink). EMDB ID 31441 map density in grey." src="../../../../../_images/Fig8_modelangelo_results_4.svg" width="50.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 6.23 </span><span class="caption-text">Comparison of PDB ID 7F3U atomic structure (chain A, residues 217-235, blue) with ModelAngelo prediction (chain Af, residues 315-334, pink).
EMDB ID 31441 map density in grey.</span><a class="headerlink" href="#model-building-modelangelo-results-4" title="Permalink to this image"></a></p>
</div>
</div></blockquote>
<p>Red arrows in <a class="reference internal" href="#model-building-modelangelo-results-4"><span class="std std-numref">Fig. 6.23</span></a> point to three Phenylalanine residues located in the <a class="reference external" href="https://www.uniprot.org/uniprot/Q9BXJ8">Q9BXJ8</a> sequence
217-QK <strong>F</strong> RNQ <strong>F</strong> LSFSMYQS <strong>F</strong> VQF-235 that have been correctly assigned. In general, residues with big lateral side chains are better assigned than others and they can be a good starting point
to assign the rest of the sequence residues.</p>
</div>
</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="012-generalWorkflow.html" class="btn btn-neutral float-left" title="5. General workflow followed in this tutorial to predict and compare ModelAngelo structures" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="030-compareModel.html" class="btn btn-neutral float-right" title="7. Comparison of ModelAngelo generated models with the published traced atomic structure" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Scipion team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>