<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>1. ModelAngelo Initial Model Prediction protocol &mdash; Scipion 3.0.0 documentation</title>
      <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../../../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../../../../_static/custom.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../../../../" id="documentation_options" src="../../../../../_static/documentation_options.js"></script>
        <script src="../../../../../_static/jquery.js"></script>
        <script src="../../../../../_static/underscore.js"></script>
        <script src="../../../../../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../../../../../_static/doctools.js"></script>
    <script src="../../../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../../search.html" />
    <link rel="next" title="On-the-fly processing" href="../../../../facilities/facilities.html" />
    <link rel="prev" title="7. Comparison of ModelAngelo generated models with the published traced atomic structure" href="030-compareModel.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../../../../../index.html" class="icon icon-home"> Scipion
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/how-to-install.html">Installing Scipion</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">User documentation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html#image-processing-resources">Image processing resources</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../user-documentation.html#tutorials">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#single-particle-tutorials">Single particle tutorials</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#tomography-tutorials">Tomography tutorials</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../../user-documentation.html#model-building">Model Building</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../Homology/index.html">Model Building by Homology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../Alphafold/index.html">Build initial model with Alphafold</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="index.html">Build initial model with ModelAngelo</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="index.html#contents">Contents</a></li>
<li class="toctree-l4 current"><a class="reference internal" href="index.html#appendices">Appendices</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html#processing-how-to-s">Processing How To’s</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities.html">On-the-fly processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/customize-report.html">Summary Reports</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/scipion-API.html">Scipion API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/plugins-API.html">Plugins API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/xmipp-API.html">Xmipp API</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../misc/contact-us.html">Contact Us</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Licences</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../license.html">Licenses</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../../index.html">Scipion</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content style-external-links">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../../../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../../../user-documentation.html">Graphical interface manuals</a> &raquo;</li>
          <li><a href="index.html">Predicting initial models with ModelAngelo</a> &raquo;</li>
      <li><span class="section-number">1. </span>ModelAngelo Initial Model Prediction protocol</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../../../../_sources/docs/user/tutorials/modelBuilding/ModelAngelo/a010-modelAngeloPredictionProtocol.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="modelangelo-initial-model-prediction-protocol">
<span id="app-modelangelopredictionprotocol"></span><h1><span class="section-number">1. </span>ModelAngelo Initial Model Prediction protocol<a class="headerlink" href="#modelangelo-initial-model-prediction-protocol" title="Permalink to this heading"></a></h1>
<p>Protocol <strong>modelangelo - model builder</strong> from <strong>scipion-em-modelangelo</strong> plugin has been designed to get <a class="reference external" href="https://github.com/3dem/model-angelo">ModelAngelo</a> <em>de novo</em>
atomic structure predictions using a 3D density map as restraint <span id="id1">[<a class="reference internal" href="../Homology/z010-references.html#id34" title="Kiarash Jamali, Dari Kimanius, and Sjors HW Scheres. A graph neural network approach to automated model building in cryo-em maps. In International Conference on Learning Representations. 2023. URL: https://openreview.net/forum?id=65XDF_nwI61.">Jamali <em>et al.</em>, 2023</a>]</span>.
This method is useful to trace the structure of map areas with quite good resolution (commonly lower than 4 angstroms), and to assign residues to the density that
can be starting clues to localize a protein that can be manually traced.</p>
<ul>
<li><div class="line-block">
<div class="line">Requirements to run this protocol and visualize results:</div>
</div>
<ul>
<li><div class="line-block">
<div class="line"><em>Scipion</em> plugin: <strong>scipion-em</strong></div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Scipion</em> plugin: <strong>scipion-em-modelangelo</strong></div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Scipion</em> plugin: <strong>scipion-em-chimera</strong></div>
</div>
</li>
</ul>
</li>
<li><div class="line-block">
<div class="line"><em>Scipion</em> menu: <em>Model building -&gt; Initial model</em> (<a class="reference internal" href="#model-building-modelangelo-protocol-1"><span class="std std-numref">Fig. 1.5</span></a> (A))</div>
</div>
<div class="figure align-center" id="model-building-modelangelo-protocol-1">
<a class="reference internal image-reference" href="../../../../../_images/Fig1a_modelangelo_protocol.svg"><img alt="Protocol **scipion-em-modelangelo**. A: Protocol location in *Scipion* menu. B: Protocol form to generate *ModelAngelo* structure predictions." src="../../../../../_images/Fig1a_modelangelo_protocol.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 1.5 </span><span class="caption-text">Protocol <strong>scipion-em-modelangelo</strong>. A: Protocol location in <em>Scipion</em> menu. B: Protocol form to generate <em>ModelAngelo</em> structure predictions.</span><a class="headerlink" href="#model-building-modelangelo-protocol-1" title="Permalink to this image"></a></p>
</div>
</li>
<li><div class="line-block">
<div class="line">Protocol form parameters (<a class="reference internal" href="#model-building-modelangelo-protocol-1"><span class="std std-numref">Fig. 1.5</span></a> (B)):</div>
</div>
<div class="line-block">
<div class="line"><em>Input</em> section:</div>
</div>
<ul>
<li><div class="line-block">
<div class="line"><strong>Refined volume</strong>: Mandatory param to load the electron density <em>map</em> previously downloaded or generated by refinement in <em>Scipion</em>.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>Protein sequences</strong>: Optional param to load one or several protein sequences experimentally identified, for example from a mass spectrometry assay. When sequence information
is added, <em>ModelAngelo</em> will try to fit it in the map. If no sequence at all is provided, <em>ModelAngelo</em> will try to guess a fitting sequence according to the map density for each residue.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>Volume mask</strong>: Optional param for loading a mask to restrict the computation to predict the initial model to the area covered by the non-zero values of the mask.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>Configuration File</strong>: Advanced param that allows to modify params included in the <a class="reference external" href="https://github.com/3dem/model-angelo/blob/main/model_angelo/config.json">configuration file</a>.</div>
</div>
</li>
</ul>
</li>
<li><div class="line-block">
<div class="line">Protocol execution:</div>
</div>
<blockquote>
<div><div class="line-block">
<div class="line">Adding specific protocol label is recommended in <em>Run name</em>
section, at the form top. To add the label, open the protocol form,
press the pencil symbol at the right side of <em>Run name</em> box,
complete the label in the new opened window, press OK and, finally,
close the protocol. This label will be shown in the output summary
content (see below). If you want to run again this protocol, do not
forget to set to <em>Restart</em> the <em>Run mode</em>.</div>
</div>
<div class="line-block">
<div class="line">To accelerate the execution, this protocol runs using a GPU with at least 8GB of memory (see <a class="reference external" href="https://github.com/3dem/model-angelo">documentation</a>). Optionally,
if you do not have access to GPU devices the protocol is able to run without them also. Nevertheless, it takes much more time. Anyway, do not forget to select in the
form <em>Run</em> window <strong>Yes</strong> or <strong>No</strong> in FGPU IDs, as well as writing the number of the GPU used (<strong>0</strong> by default).</div>
</div>
<div class="line-block">
<div class="line">Press the <em>Execute</em> red button at the form bottom.</div>
</div>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line">Visualization of protocol results:</div>
</div>
<blockquote>
<div><div class="line-block">
<div class="line">After executing the protocol, results can be visualized in <em>Chimera</em> GUI clicking <strong>Analyze Results</strong>. Besides the coordinate axes, two different types of results will be
observed depending on the inputs. If sequence information has been included as input, the atomic structure prediction will be displayed in two ways, raw and pruned.
Raw or unpruned structures, directly obtained as protocol output, include every structural element and display minimal post-processing. Most of the model has
thus been built. However, lower resolution areas of the model are often wrong. A manual inspection of these areas is basic to manually remove incorrect assigned residues.
In pruned predictions, instead, this clearance task has already been performed in an automatic way. Every portion of the model that do not have
good matches to the sequence based on a hidden Markov model alignment is removed. In this case, the part of the interpreted map is lower but most of residues are
correctly assigned to the density. Go to <a class="reference internal" href="020-initialModel.html#model-building-modelangelo-results-1"><span class="std std-numref">Fig. 6.20</span></a> to see an example of this protocol output.</div>
</div>
<div class="line-block">
<div class="line">When no sequence information has been included as protocol input, only a raw structure is generated as <em>ModelAngelo</em> prediction (example in
<a class="reference internal" href="020-initialModel.html#model-building-modelangelo-results-3"><span class="std std-numref">Fig. 6.22</span></a>). Without sequence information, <em>ModelAngelo</em> itself assigns the residues according
to the density trying to cover the whole map. Then, the user should inspect the structure and perform the task of clearence of residues incorrectly assigned.</div>
</div>
<div class="line-block">
<div class="line">Prediction outputs are colored according to the values of the B-factor column in the structure file using Alphafold color criteria and <em>Color key</em> appears
in the GUI main window. <strong>REMARK</strong> that good matches to the sequence and not those B-factor values are the criteria for pruning residues.</div>
</div>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line">Summary content:</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line">Protocol output (below framework):</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line">modelangelo - model builder protein name -&gt; raw         AtomStruct(pseudoatoms=False, volume=False)</div>
</div>
</li>
<li><div class="line-block">
<div class="line">modelangelo - model builder protein name -&gt; pruned      AtomStruct(pseudoatoms=False, volume=False)</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line"><em>SUMMARY</em> box:</div>
<div class="line">No summary information.</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
</ul>
</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="030-compareModel.html" class="btn btn-neutral float-left" title="7. Comparison of ModelAngelo generated models with the published traced atomic structure" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../../../../facilities/facilities.html" class="btn btn-neutral float-right" title="On-the-fly processing" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Scipion team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>