

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>19. Model from Template protocol &mdash; Scipion 3.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../../../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
        <script data-url_root="../../../../" id="documentation_options" src="../../../../_static/documentation_options.js"></script>
        <script src="../../../../_static/jquery.js"></script>
        <script src="../../../../_static/underscore.js"></script>
        <script src="../../../../_static/doctools.js"></script>
    
    <script type="text/javascript" src="../../../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/banner.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/custom.css" type="text/css" />
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../../index.html" class="icon icon-home"> Scipion
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../scipion-modes/how-to-install.html">Installing Scipion</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">User documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../user-documentation.html#tutorials">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../user-documentation.html#processing-how-to-s">Processing How To’s</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../facilities/facilities.html">On-the-fly processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../facilities/customize-report.html">Summary Reports</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../api/scipion-API.html">Scipion API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/plugins-API.html">Plugins API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../api/xmipp-API.html">Xmipp API</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../misc/contact-us.html">Contact Us</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Licences</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../license.html">Licenses</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">Scipion</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../index.html">Docs</a> &raquo;</li>
        
      <li><span class="section-number">19. </span>Model from Template protocol</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../../../../_sources/docs/user/tutorials/modelBuilding/a110-modelFromTemplate.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="model-from-template-protocol">
<span id="app-modelfromtemplate"></span><h1><span class="section-number">19. </span>Model from Template protocol<a class="headerlink" href="#model-from-template-protocol" title="Permalink to this headline">¶</a></h1>
<div class="line-block">
<div class="line">Protocol designed to obtain a structure model for a target sequence in
. Target structure is predicted by sequence homology using
<a href="#id1"><span class="problematic" id="id2">:raw-latex:`\citep{sali1993}`</span></a> web service in .</div>
<div class="line">WARNING: Working with requires a license key, which can be requested
free of charge for academic users. Try to have this license key before
starting the protocol execution.</div>
</div>
<ul>
<li><p>Requirements to run this protocol and visualize results:</p>
<ul class="simple">
<li><p>plugin:</p></li>
<li><p>plugin:</p></li>
<li><p>Multiple sequence alignment tools: ,</p></li>
</ul>
</li>
<li><p>menu: ( (A))</p></li>
<li><p>Protocol form parameters ( (B)):</p>
<ul>
<li><p>section</p>
<ul>
<li><p>: Select if you have found your in a previous similarity
searching step. Select if you do not have any to start the
homology modeling and you would like to search for one.</p></li>
<li><p>Option ( (B))</p>
<ul>
<li><p>: Atomic structure previously downloaded in . This structure
was selected by sequence homology, i.e. by looking for the
structurally characterized sequence more similar (with
higher identity) to the target sequence.</p></li>
<li><p>: Specific monomer of the macromolecule that has to be used
as structure of the . Use the wizard on the right side of
parameter to select that chain.</p></li>
<li><p>: Sequence previously downloaded in . This sequence has to
be modeled following the structure skeleton of the selected
.</p></li>
<li><p>: <span class="math">Modeller</span> provides structural models of the based
on a sequence alignment, in which at least sequences of and
have to be included. Three options can be considered to
improve this alignment:</p>
<ol class="arabic">
<li><div class="line-block">
<div class="line">: No more sequences are going to be included in the
alignment except and sequences. Correlative param:</div>
<div class="line">*** : Select one of the three available alignment
methods, (by default), , .</div>
</div>
</li>
<li><div class="line-block">
<div class="line">if you want to perform a multiple sequence alignment
adding other sequences already downloaded in .
Additional sequences, others than and sequences, are
required to accomplish this multiple alignment.
Correlative params:</div>
<div class="line">*** : Box to complete with the additional sequences
used to perform the multiple sequence alignment. All of
them were previously downloaded in .</div>
<div class="line">*** : Select between and methods.</div>
</div>
</li>
<li><div class="line-block">
<div class="line">: If you want to include other sequences in the
alignment by providing your own sequence alignment.
Correlative param:</div>
<div class="line">*** : Complete this box with the help of the right
side browser including the sequence alignment file that
you already have saved in your computer. Different
alignment formats are available
(<a class="reference external" href="https://www.cgl.ucsf.edu/chimerax/docs/user/commands/open.html">https://www.cgl.ucsf.edu/chimerax/docs/user/commands/open.html</a>).
An example of alignment in format can be seen below
(Use case 3).</div>
</div>
</li>
</ol>
</li>
<li><p>: Select if you’d like to obtain a multimer by using two
sequences and the same multimer . The params to complete the
option are identical to those already shown, with the
exception of the , already completed. However, no one of
those params will appear in case you select in order to
obtain a by using only one sequence.</p></li>
</ul>
</li>
<li><p>Option ( (C))</p>
<ul class="simple">
<li><p>: Sequence previously downloaded in . This sequence has to
be modeled following the structure skeleton of the that you
are going to select among the retrieved entries found by the
similarity searching tool.</p></li>
<li><p>: Select one of the two suggested protein sequence
databases, PDB and NR. Press the symbol on the right to see
the meaning of each one. Remark that the NR database allows
you to get entries with, as well as without, atomic
structure associated. These ones, which do not provide ,
could be useful to build a better sequence alignment.</p></li>
<li><p>: Select one of the “substitution matrix” to assign a score
to any couple of residues in the alignment
(<a class="reference external" href="https://www.ncbi.nlm.nih.gov/blast/html/sub_matrix.html">https://www.ncbi.nlm.nih.gov/blast/html/sub_matrix.html</a>).</p></li>
<li><p>: Maximum statistic value required to include a retrieved
element in the hit list.</p></li>
<li><p>that you’d like to retrieve from the database.</p></li>
</ul>
</li>
</ul>
</li>
<li><p>section</p>
<p>Follow this section steps to run <span class="math">Modeller</span> via web service
in and to select and save one of the retrieved models in
framework.</p>
</li>
</ul>
</li>
<li><p>Protocol execution:</p>
<div class="line-block">
<div class="line">Adding specific template-target label is recommended in section, at
the form top. To add the label, open the protocol form, press the
pencil symbol on the right side of box, complete the label in the
new opened window, press OK and, finally, close the protocol. This
label will be shown in the output summary content (see below). If
you want to run again this protocol, do not forget to set to the .</div>
<div class="line">Press the red button at the form bottom.</div>
<div class="line">Several windows will be opened after executing the protocol with
different contents according to the distinct form param options.
Although we are going to detail some of them through several use
cases (see below), designed to ilustrate different applications of
this protocol, as well as the procedure to follow in each case, in
general we can predict the opening of graphics window and a
sequence alignment window. Usually, in both windows the sequence is
green highlighted (see an example of these windows in ). Main steps
to follow ahead are:</div>
</div>
<ul>
<li><p>Ask for model(s) to <span class="math">Modeller</span> by selecting in the main menu
of graphics window.</p></li>
<li><p>Complete the new window opened for with the sequence alignment
that includes the and with the (s) sequence(s), <span class="math">Modeller</span>
license key, multichain model, number of models retrieved by , and
options like the building of models with hydrogens, as well as
<span class="math">model</span> inclusion of heteroatoms or water molecules. An
example of completed <span class="math">Modeller</span> window can be observed in
(A). By pressing the computation starts. The status of the job can
be checked in the lower left corner of graphics window.</p></li>
<li><p>After a while a new panel window will show the retrieved models of
the sequence ( (B)). Two statistics assess these models: ,
statistical potentials derived-score, and , normalized Discrete
Optimized Protein Energy, atomic distance depending-score.
Reliable models show values higher than 0.7, and negative values
correspond to better models. Retrieved models can be checked in .
One of them should be selected ( (C)).</p></li>
<li><div class="line-block">
<div class="line">Rename the selected model, for example to with the command line:</div>
</div>
</li>
<li><div class="line-block">
<div class="line">Save the retrieved model selected according to the new model
number in the track system () shown in by writing in command
line:</div>
</div>
</li>
</ul>
</li>
<li><p>Visualization of protocol results:</p>
<p>After executing the protocol, press and graphics window will be
opened by default. Atomic structures are referred to the origin of
coordinates in . To show the relative position of the atomic
structure, the three coordinate axes are represented; X axis (red), Y
axis (yellow), and Z axis (blue) (). Coordinate axes and selected
atomic structure are model numbers and , respectively, in panel if
only one structure has been saved.</p>
</li>
<li><p>Summary content:</p>
<ul>
<li><div class="line-block">
<div class="line">Protocol output (below framework):</div>
<div class="line">;</div>
<div class="line">.</div>
<div class="line">Pseudoatoms is set to when the structure is made of pseudoatoms
instead of atoms. Volume is set to when an electron density map
is associated to the atomic structure.</div>
</div>
</li>
<li><div class="line-block">
<div class="line">box:</div>
<div class="line">Produced files:</div>
<div class="line">we have some result</div>
</div>
</li>
</ul>
</li>
</ul>
<div class="section" id="use-cases">
<h2><span class="section-number">19.1. </span>USE CASES<a class="headerlink" href="#use-cases" title="Permalink to this headline">¶</a></h2>
<ul>
<li><div class="line-block">
<div class="line"><br /></div>
<div class="line">Aim: To model a sequence using one chain of a homologous atomic
structure as , using only the sequences of and in the sequence
alignment.</div>
</div>
<div class="line-block">
<div class="line">Protocol execution: Complete the protocol form as indicated in (A).
Follow the general procedure shown above (Protocol execution
section). Windows (B) and (C) will appear. Open and complete the
panel as indicated in (D) and wait for a while. After getting the
retrieved models, if you want to select, for example, the model ,
write in the command line:</div>
<div class="line">And to close the protocol. Visualize your results.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><br /></div>
<div class="line">Aim: To model simultaneously two sequences to obtain a multichain
model using two chains of a homologous atomic structure as , using
other additional sequences than and in the sequence alignment.</div>
</div>
<div class="line-block">
<div class="line">Protocol execution: Complete the protocol form as indicated in (A).
Follow the general procedure shown above (Protocol execution
section). Windows (C), (D) and (E) will appear. Open and complete
the panel as indicated in (B) and wait for a while. After getting
the retrieved models, if you want to select, for example, the model
, write in the command line:</div>
<div class="line">And to close the protocol. Visualize your results.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><br /></div>
<div class="line">Aim: To model simultaneously two sequences to obtain a multichain
model using two chains of a homologous atomic structure as , using
your own sequence alignment.</div>
</div>
<div class="line-block">
<div class="line">Protocol execution: Complete the protocol form as indicated in (A).
Follow the general procedure shown above (Protocol execution
section). Remark that you already have a sequence alignment file
for each saved in your computer. An example can be seen in (B).
Windows (C), (D) and (E) of previous . Open and complete the panel
as indicated in (B) and wait for a while. After getting the
retrieved models, if you want to select, for example, the model ,
write in the command line:</div>
<div class="line">And to close the protocol. Visualize your results.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><br /></div>
<div class="line">Aim: To model a sequence without previous information of a possible
atomic structure .</div>
</div>
<div class="line-block">
<div class="line">Protocol execution: Complete the protocol form as indicated in (A).
Follow the general procedure shown above (Protocol execution
section). Remark that in this case there is no atomic structure.
Instead, an empty graphics window will appear (C) together with the
sequence that we’d like to model and retrieved results (E). Note
that it could take some seconds the opening of the panel. Have a
look to these results and select one of them as possible of your
sequence. In this particular case, for example, we are going to
choose the first one (). Open this atomic structure en by writing
in the command line:</div>
<div class="line">At this point, open the panel and complete it as indicated in (B).
Wait for a while. After getting the retrieved models, if you want
to select, for example, the model , write in the command line:</div>
<div class="line">And to close the protocol. Visualize your results.</div>
</div>
</li>
</ul>
</div>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Scipion team

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  
<div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
        <span class="fa fa-book"> Other Versions</span>
        v: rm_model_building
        <span class="fa fa-caret-down"></span>
    </span>
    <div class="rst-other-versions">
        <dl>
            <dt>Branches</dt>
            <dd><a href="../../../../../gh-pages/index.html">gh-pages</a></dd>
            <dd><a href="../../../../../release-2.0.0/index.html">release-2.0.0</a></dd>
            <dd><a href="../../../../../release-3.0.0/docs/user/tutorials/modelBuilding/a110-modelFromTemplate.html">release-3.0.0</a></dd>
            <dd><a href="a110-modelFromTemplate.html">rm_model_building</a></dd>
        </dl>
    </div>
</div>


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>