

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>9. Morph the predicted model onto the docked fragments and rebuild the missing parts of the model in the map density &mdash; Scipion 3.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../../../../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../../../" src="../../../../../_static/documentation_options.js"></script>
        <script data-url_root="../../../../../" id="documentation_options" src="../../../../../_static/documentation_options.js"></script>
        <script src="../../../../../_static/jquery.js"></script>
        <script src="../../../../../_static/underscore.js"></script>
        <script src="../../../../../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../../../../../_static/doctools.js"></script>
    
    <script type="text/javascript" src="../../../../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/banner.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/custom.css" type="text/css" />
    <link rel="index" title="Index" href="../../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../../search.html" />
    <link rel="next" title="10. Three in one: Processing, docking and rebuilding the model in the map density" href="060-dockAndRebuildModel.html" />
    <link rel="prev" title="8. Dock the domains obtained into your cryo-EM map or unit cell" href="040-dockModel.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../../../index.html" class="icon icon-home"> Scipion
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/how-to-install.html">Installing Scipion</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">User documentation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../user-documentation.html#tutorials">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#single-particle-tutorials">Single particle tutorials</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#tomography-tutorials">Tomography tutorials</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../../user-documentation.html#model-building">Model Building</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../Homology/index.html">Model Building by Homology</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="index.html">Build initial model with Alphafold</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="index.html#contents">Contents</a></li>
<li class="toctree-l4"><a class="reference internal" href="index.html#appendices">Appendices</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html#processing-how-to-s">Processing How To’s</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities.html">On-the-fly processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/customize-report.html">Summary Reports</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/scipion-API.html">Scipion API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/plugins-API.html">Plugins API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/xmipp-API.html">Xmipp API</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../misc/contact-us.html">Contact Us</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Licences</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../license.html">Licenses</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../../index.html">Scipion</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../../index.html">Docs</a> &raquo;</li>
        
          <li><a href="../../../user-documentation.html">Graphical interface manuals</a> &raquo;</li>
        
          <li><a href="index.html">Predicting initial models with AlphaFold</a> &raquo;</li>
        
      <li><span class="section-number">9. </span>Morph the predicted model onto the docked fragments and rebuild the missing parts of the model in the map density</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../../../../../_sources/docs/user/tutorials/modelBuilding/Alphafold/050-rebuildModel.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="morph-the-predicted-model-onto-the-docked-fragments-and-rebuild-the-missing-parts-of-the-model-in-the-map-density">
<span id="rebuildmodel"></span><h1><span class="section-number">9. </span>Morph the predicted model onto the docked fragments and rebuild the missing parts of the model in the map density<a class="headerlink" href="#morph-the-predicted-model-onto-the-docked-fragments-and-rebuild-the-missing-parts-of-the-model-in-the-map-density" title="Permalink to this heading">¶</a></h1>
<p>In this section of the AlphaFold tutorial we are going to rebuild the docked isoform CRA_a of TACAN monomer structure finally obtained in the previous <a class="reference internal" href="040-dockModel.html#dockinitialmodel"><span class="std std-ref">section</span></a>.</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig20-workflow">
<a class="reference internal image-reference" href="../../../../../_images/Fig20_workflow_4.svg"><img alt="Part of the general Scipion workflow performed to rebuild the AlphaFold2 prediction structure of the isoform CRA_a of human TACAN complex." src="../../../../../_images/Fig20_workflow_4.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.23 </span><span class="caption-text">Part of the general Scipion workflow performed to rebuild the AlphaFold2 prediction structure of the isoform CRA_a of human TACAN complex.</span><a class="headerlink" href="#model-building-fig20-workflow" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>We are going to take advantage of the protocol <strong>phenix-rebuild predicted model</strong> to rebuild the docked structure prediction into the map (protocol <a class="reference internal" href="a040-rebuildPredictedModelProtocol.html#app-rebuildpredictedmodelprotocol"><span class="std std-ref">appendix</span></a>). As the title of this section indicates, the rebuild function morphs the predicted model onto the docked fragments and rebuilds the missing parts of the model in the map density. The result, however, depends very much on the map we are working with. In any case, the result can still be improved by additional rounds of processing.</p>
<p>Let us start by opening (<a class="reference internal" href="#model-building-fig24-rebuild"><span class="std std-numref">Fig. 9.24</span></a> (1)) and filling in the protocol form, adding the previously obtained <a class="reference internal" href="020-initialModel.html#getinitialmodel"><span class="std std-ref">predicted AlphaFold2</span></a> (2) and <a class="reference internal" href="040-dockModel.html#dockinitialmodel"><span class="std std-ref">docked</span></a> models (3), and the map (4). After fixing the map resolution limit, increase the number of threads (5) and finally, execute the protocol (6).</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig24-rebuild">
<a class="reference internal image-reference" href="../../../../../_images/Fig24_protocol_rebuild_model_1.svg"><img alt="Completing the protocol **phenix-rebuild predicted model** (workflow step 11; :numref:`model_building_fig20_workflow`)." src="../../../../../_images/Fig24_protocol_rebuild_model_1.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.24 </span><span class="caption-text">Completing the protocol <strong>phenix-rebuild predicted model</strong> (workflow step 11; <a class="reference internal" href="#model-building-fig20-workflow"><span class="std std-numref">Fig. 9.23</span></a>).</span><a class="headerlink" href="#model-building-fig24-rebuild" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>After a quick running, the protocol finishes. Probably due to a file format incompatibility, the tool was unable to morph the predicted structure, as you can check in the results clicking <code class="docutils literal notranslate"><span class="pre">Analyze</span> <span class="pre">Results</span></code>. Nevertheless, we can use the recently obtained docked model (<a class="reference internal" href="040-dockModel.html#model-building-fig23-dock-model"><span class="std std-numref">Fig. 8.13</span></a>) as template to get a new AlphaFold prediction using this option in the <em>Phenix</em> Colab Notebook. With this aim, complete again the <strong>chimerax-alphafold prediction</strong> protocol (<a class="reference internal" href="#model-building-fig25-rebuild"><span class="std std-numref">Fig. 9.25</span></a> (1)) selecting <em>Google Colab</em> (2), the <em>Phenix Colab Notebook</em> (3), the isoform CRA_a of TACAN protein sequence (4) and the previous docked model (5). Then, execute the protocol (6).</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig25-rebuild">
<a class="reference internal image-reference" href="../../../../../_images/Fig25_protocol_rebuild_model_3.svg"><img alt="Completing the protocol **chimerax-alphafold prediction** using the Phenix Colab Notebook with template option (workflow step 12; :numref:`model_building_fig20_workflow`)." src="../../../../../_images/Fig25_protocol_rebuild_model_3.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.25 </span><span class="caption-text">Completing the protocol <strong>chimerax-alphafold prediction</strong> using the Phenix Colab Notebook with template option (workflow step 12; <a class="reference internal" href="#model-building-fig20-workflow"><span class="std std-numref">Fig. 9.23</span></a>).</span><a class="headerlink" href="#model-building-fig25-rebuild" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>Soon after executing the protocol, the browser window of <em>Phenix</em> Colab Notebok opens and stars to run. <code class="docutils literal notranslate"><span class="pre">REMARK</span></code> that when a template is used, this template has to be manually uploaded. This task requires clicking the key <code class="docutils literal notranslate"><span class="pre">Choose</span> <span class="pre">Files</span></code> in the Cell #3 of the browser (<a class="reference internal" href="#model-building-fig26-rebuild"><span class="std std-numref">Fig. 9.26</span></a> (1)).</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig26-rebuild">
<a class="reference internal image-reference" href="../../../../../_images/Fig26_protocol_rebuild_model_4.svg"><img alt="Cell 3 of the browser of *Phenix* Colab Notebook completed and running." src="../../../../../_images/Fig26_protocol_rebuild_model_4.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.26 </span><span class="caption-text">Cell 3 of the browser of <em>Phenix</em> Colab Notebook completed and running.</span><a class="headerlink" href="#model-building-fig26-rebuild" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>After a while the prediction process finishes and the AlphaFold2 predicted structure is shown in <em>ChimeraX</em>. <a class="reference internal" href="#model-building-fig27-rebuild"><span class="std std-numref">Fig. 9.27</span></a> details the new prediction (A) and the PAE plot (B). Comparing the accuracy of residues and PAE error with the predictions of monomers and dimers that we have obtained <a class="reference internal" href="020-initialModel.html#getinitialmodel"><span class="std std-ref">before</span></a>, we can observe that the new one is the best one.</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig27-rebuild">
<a class="reference internal image-reference" href="../../../../../_images/Fig27_protocol_rebuild_model_5.svg"><img alt="Prediction of isoform CRA_a of human TACAN protein structure obtained using a template in the *Phenix* Colab Notebook. **A**: Protein structure in *ChimeraX* GUI. **B**: PAE plot." src="../../../../../_images/Fig27_protocol_rebuild_model_5.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.27 </span><span class="caption-text">Prediction of isoform CRA_a of human TACAN protein structure obtained using a template in the <em>Phenix</em> Colab Notebook. <strong>A</strong>: Protein structure in <em>ChimeraX</em> GUI. <strong>B</strong>: PAE plot.</span><a class="headerlink" href="#model-building-fig27-rebuild" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>At this point we can follow the common workflow to process, dock and rebuild the new AlphaFold2 prediction. Starting with the model processing, open the protocol <strong>phenix-process predicted model</strong> (<a class="reference internal" href="#model-building-fig28-rebuild"><span class="std std-numref">Fig. 9.28</span></a> (1)) with the predicted model (2), the B-value column values to consider as minimum to process (3), the PAE file (4), and then execute the protocol (5).</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig28-rebuild">
<a class="reference internal image-reference" href="../../../../../_images/Fig28_protocol_rebuild_model_6.svg"><img alt="Completing the protocol **phenix-process predicted model** to process the AlphaFold2 prediction obtained using the Phenix Colab Notebook with template option (workflow step 13; :numref:`model_building_fig20_workflow`)." src="../../../../../_images/Fig28_protocol_rebuild_model_6.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.28 </span><span class="caption-text">Completing the protocol <strong>phenix-process predicted model</strong> to process the AlphaFold2 prediction obtained using the Phenix Colab Notebook with template option (workflow step 13; <a class="reference internal" href="#model-building-fig20-workflow"><span class="std std-numref">Fig. 9.23</span></a>).</span><a class="headerlink" href="#model-building-fig28-rebuild" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>After executing the protocol press <code class="docutils literal notranslate"><span class="pre">Analyze</span> <span class="pre">Results</span></code> to check the results menu (<a class="reference internal" href="#model-building-fig29-rebuild"><span class="std std-numref">Fig. 9.29</span></a> (A)).</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig29-rebuild">
<a class="reference internal image-reference" href="../../../../../_images/Fig29_protocol_rebuild_model_7.svg"><img alt="Results obtained with the Scipion protocol **phenix-process predicted model**. **A**: Menu of results. **B**: Processed model opened (monomer) in ChimeraX GUI viewer. **C**: Remainder sequence fragment excluded from the processed structure." src="../../../../../_images/Fig29_protocol_rebuild_model_7.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.29 </span><span class="caption-text">Results obtained with the Scipion protocol <strong>phenix-process predicted model</strong>. <strong>A</strong>: Menu of results. <strong>B</strong>: Processed model opened (monomer) in ChimeraX GUI viewer. <strong>C</strong>: Remainder sequence fragment excluded from the processed structure.</span><a class="headerlink" href="#model-building-fig29-rebuild" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>Clicking <em>Structures in ChimeraX</em> (<a class="reference internal" href="#model-building-fig29-rebuild"><span class="std std-numref">Fig. 9.29</span></a> (A, 1)), the predicted and the processed structures will be simultaneously opened. Whereas the processed structure is still colored according to the AlphaFold2 confidence score, the processed structure is pink colored (<a class="reference internal" href="#model-building-fig29-rebuild"><span class="std std-numref">Fig. 9.29</span></a> (B)). Remark that this processed structure is constituted by <strong>only one domain (chain)</strong> (model #3).  Concerning the remaining fragments of sequence removed, open <em>Remaining Sequences</em> in the results’ menu (<a class="reference internal" href="#model-building-fig29-rebuild"><span class="std std-numref">Fig. 9.29</span></a> (A, 2)) to observe the text file (C). The sequence is quite similar to the sequence shown in <a class="reference internal" href="030-processModel.html#model-building-fig15-alphafold-process"><span class="std std-numref">Fig. 7.18</span></a> (C)).</p>
<p>Next, run the protocol <strong>phenix-dock predicted model</strong> to dock the structure prediction into the map (protocol <a class="reference internal" href="a030-dockPredictedModelProtocol.html#app-dockpredictedmodelprotocol"><span class="std std-ref">appendix</span></a>). Open the protocol (<a class="reference internal" href="#model-building-fig30-rebuild"><span class="std std-numref">Fig. 9.30</span></a> (1)) and complete it with the new AlphaFold2 prediction obtained with <em>Phenix</em> notebook (2) (<a class="reference internal" href="#model-building-fig27-rebuild"><span class="std std-numref">Fig. 9.27</span></a>). Include the respective processed prediction (3) (<a class="reference internal" href="#model-building-fig29-rebuild"><span class="std std-numref">Fig. 9.29</span></a> (B)). Fill in the <em>Input map</em> box with the recently downloaded map (4) (<a class="reference internal" href="040-dockModel.html#model-building-fig14-alphafold-docking"><span class="std std-numref">Fig. 8.11</span></a>). Update the resolution value of the map, increase the number of processors to speed up the process (5) and, finally, execute the protocol (6).</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig30-rebuild">
<a class="reference internal image-reference" href="../../../../../_images/Fig30_protocol_rebuild_model_8.svg"><img alt="Scipion Protocol **phenix-dock predicted model** completed to dock the isoform CRA_a of human TACAN protein prediction (workflow step 14; :numref:`model_building_fig20_workflow`)." src="../../../../../_images/Fig30_protocol_rebuild_model_8.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.30 </span><span class="caption-text">Scipion Protocol <strong>phenix-dock predicted model</strong> completed to dock the isoform CRA_a of human TACAN protein prediction (workflow step 14; <a class="reference internal" href="#model-building-fig20-workflow"><span class="std std-numref">Fig. 9.23</span></a>).</span><a class="headerlink" href="#model-building-fig30-rebuild" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>Visualize the results clicking <code class="docutils literal notranslate"><span class="pre">Analyze</span> <span class="pre">Results</span></code> when the execution finishes. The <em>ChimeraX</em> GUI window will open to show the map and several models (<a class="reference internal" href="#model-building-fig31-rebuild"><span class="std std-numref">Fig. 9.31</span></a>). Not shown in <a class="reference internal" href="#model-building-fig31-rebuild"><span class="std std-numref">Fig. 9.31</span></a>, models #3 and #4 are the new AlphaFold2 prediction and the processed models, respectively. Pink model #5 is the docked model.</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig31-rebuild">
<a class="reference internal image-reference" href="../../../../../_images/Fig31_protocol_rebuild_model_9.svg"><img alt="ChimeraX GUI opened to visualize the docked model in the map EMD-31441." src="../../../../../_images/Fig31_protocol_rebuild_model_9.svg" width="90.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.31 </span><span class="caption-text">ChimeraX GUI opened to visualize the docked model in the map EMD-31441.</span><a class="headerlink" href="#model-building-fig31-rebuild" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>Select the appropriate orientation of the map, the density threshold and the appearance as mesh to better visualize the docked structure.</p>
<p>At this point we take advantage of the protocol <strong>phenix-rebuild predicted model</strong> to rebuild the docked structure prediction into the map (protocol <a class="reference internal" href="a040-rebuildPredictedModelProtocol.html#app-rebuildpredictedmodelprotocol"><span class="std std-ref">appendix</span></a>). Rebuild function morphs the predicted model onto the docked fragments and rebuilds the missing parts of the model in the map density. Then, open the protocol (<a class="reference internal" href="#model-building-fig24-rebuild"><span class="std std-numref">Fig. 9.24</span></a> (1)) and fill in the protocol form, adding the AlphaFold2 prediction previously obtained (<a class="reference internal" href="#model-building-fig27-rebuild"><span class="std std-numref">Fig. 9.27</span></a>) (2), the docked model (<a class="reference internal" href="#model-building-fig31-rebuild"><span class="std std-numref">Fig. 9.31</span></a>) (3) and the map (4). After fixing the map resolution limit, increase the number of threads (5) and finally, execute the protocol (6).</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig32-rebuild">
<a class="reference internal image-reference" href="../../../../../_images/Fig32_protocol_rebuild_model_10.svg"><img alt="Completing the protocol **phenix-rebuild predicted model** (workflow step 15; :numref:`model_building_fig20_workflow`)." src="../../../../../_images/Fig32_protocol_rebuild_model_10.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.32 </span><span class="caption-text">Completing the protocol <strong>phenix-rebuild predicted model</strong> (workflow step 15; <a class="reference internal" href="#model-building-fig20-workflow"><span class="std std-numref">Fig. 9.23</span></a>).</span><a class="headerlink" href="#model-building-fig32-rebuild" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>The protocol finishes after a while and clicking <code class="docutils literal notranslate"><span class="pre">Analyze</span> <span class="pre">Results</span></code> results can be checked (<a class="reference internal" href="#model-building-fig33-rebuild"><span class="std std-numref">Fig. 9.33</span></a>).</p>
<blockquote>
<div><div class="figure align-center" id="model-building-fig33-rebuild">
<a class="reference internal image-reference" href="../../../../../_images/Fig33_protocol_rebuild_model_11.svg"><img alt="ChimeraX GUI opened to visualize the rebuilt model according to the map EMD-31441. **A**: Frontal view. **B**: Lateral view. The frame details structure fitting." src="../../../../../_images/Fig33_protocol_rebuild_model_11.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 9.33 </span><span class="caption-text">ChimeraX GUI opened to visualize the rebuilt model according to the map EMD-31441. <strong>A</strong>: Frontal view. <strong>B</strong>: Lateral view. The frame details structure fitting.</span><a class="headerlink" href="#model-building-fig33-rebuild" title="Permalink to this image">¶</a></p>
</div>
</div></blockquote>
<p>The two different views shown in <a class="reference internal" href="#model-building-fig33-rebuild"><span class="std std-numref">Fig. 9.33</span></a> detail the morphing both of horizontal alpha-helices (A, 1) and vertical ones (B, 2). Remark that the new rebuilt model (pink) fits better to the map than the previous docked one (green). The rest of the sequence, however, was unable to adapt its architecture to the map and remains mainly disordered. This result makes sense due to the sequence differences in the C-terminal part of the protein between the two isoforms of the human TACAN protein. Only the sequence identical in both isoforms has been predicted by Alphafold2 and finally morphed according to the map of the first isoform. The rest of the structure of the second isoform remains unpredicted and seems to adopt a quite different architecture than the first isoform. The experimental map of this second isoform CRA_a is thus essential to characterize the structure of the C-terminal part of the second isoform.</p>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="060-dockAndRebuildModel.html" class="btn btn-neutral float-right" title="10. Three in one: Processing, docking and rebuilding the model in the map density" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="040-dockModel.html" class="btn btn-neutral float-left" title="8. Dock the domains obtained into your cryo-EM map or unit cell" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2022, Scipion team

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  
<div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
        <span class="fa fa-book"> Other Versions</span>
        v: release-3.0.0
        <span class="fa fa-caret-down"></span>
    </span>
    <div class="rst-other-versions">
        <dl>
            <dt>Branches</dt>
            <dd><a href="050-rebuildModel.html">release-3.0.0</a></dd>
        </dl>
    </div>
</div>


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>