

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>1. AlphaFold2 Initial Model Prediction protocol &mdash; Scipion 3.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../../../../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../../../" src="../../../../../_static/documentation_options.js"></script>
        <script data-url_root="../../../../../" id="documentation_options" src="../../../../../_static/documentation_options.js"></script>
        <script src="../../../../../_static/jquery.js"></script>
        <script src="../../../../../_static/underscore.js"></script>
        <script src="../../../../../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../../../../../_static/doctools.js"></script>
    
    <script type="text/javascript" src="../../../../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/banner.css" type="text/css" />
  <link rel="stylesheet" href="../../../../../_static/custom.css" type="text/css" />
    <link rel="index" title="Index" href="../../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../../search.html" />
    <link rel="next" title="2. Phenix Process Predicted Model protocol" href="a020-processPredictedProtocol.html" />
    <link rel="prev" title="9. Morph the predicted model onto the docked fragments and rebuild the missing parts of the model in the map density" href="050-rebuildModel.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../../../index.html" class="icon icon-home"> Scipion
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/how-to-install.html">Installing Scipion</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">User documentation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../user-documentation.html#tutorials">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#single-particle-tutorials">Single particle tutorials</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../user-documentation.html#tomography-tutorials">Tomography tutorials</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../../user-documentation.html#model-building">Model Building</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../Homology/index.html">Model Building by Homology</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="index.html">Build initial model with Alphafold</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="index.html#contents">Contents</a></li>
<li class="toctree-l4 current"><a class="reference internal" href="index.html#appendices">Appendices</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-documentation.html#processing-how-to-s">Processing How To’s</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities.html">On-the-fly processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../facilities/customize-report.html">Summary Reports</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/scipion-API.html">Scipion API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/plugins-API.html">Plugins API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../api/xmipp-API.html">Xmipp API</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../misc/contact-us.html">Contact Us</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Licences</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../license.html">Licenses</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../../index.html">Scipion</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../../index.html">Docs</a> &raquo;</li>
        
          <li><a href="../../../user-documentation.html">Graphical interface manuals</a> &raquo;</li>
        
          <li><a href="index.html">Predicting initial models with AlphaFold</a> &raquo;</li>
        
      <li><span class="section-number">1. </span>AlphaFold2 Initial Model Prediction protocol</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../../../../../_sources/docs/user/tutorials/modelBuilding/Alphafold/a010-alphafoldPredictionProtocol.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="alphafold2-initial-model-prediction-protocol">
<span id="app-alphafoldpredictionprotocol"></span><h1><span class="section-number">1. </span>AlphaFold2 Initial Model Prediction protocol<a class="headerlink" href="#alphafold2-initial-model-prediction-protocol" title="Permalink to this heading">¶</a></h1>
<p><em>ChimeraX</em>-based protocol designed to get AlphaFold2 atomic structure predictions <span id="id1">[<a class="reference internal" href="../Homology/z010-references.html#id28" title="J. Jumper, R. Evans, A. Pritzel, T. Green, M. Figurnov, O. Ronneberger, K. Tunyasuvunakool, R. Bates, A. Žídek, A. Potapenko, A. Bridgland, C. Meyer, S. A. A. Kohl, A. J. Ballard, A. Cowie, B. Romera-Paredes, S. Nikolov, R. Jain, J. Adler, T. Back, S. Petersen, D. Reiman, E. Clancy, M. Zielinski, M. Steinegger, M. Pacholska, T. Berghammer, S. Bodenstein, D. Silver, O. Vinyals, A. W. Senior, K. Kavukcuoglu, P. Kohli, and D. Hassabis. Highly accurate protein structure prediction with AlphaFold. Nature, 596(7873):583–589, 08 2021.">Jumper <em>et al.</em>, 2021</a>]</span>, <span id="id2">[<a class="reference internal" href="../Homology/z010-references.html#id33" title="R. Evans, Michael O'Neill, Alexander Pritzel, Natasha Antropova, Andrew Senior, Tim Green, Augustin Žídek, Russ Bates, Sam Blackwell, Jason Yim, Olaf Ronneberger, Sebastian Bodenstein, Michal Zielinski, Alex Bridgland, Anna Potapenko, Andrew Cowie, Kathryn Tunyasuvunakool, Rishub Jain, Ellen Clancy, Pushmeet Kohli, John Jumper, and Demis Hassabis. Protein complex prediction with AlphaFold-Multimer. bioRxiv, 2021. URL: https://www.biorxiv.org/content/early/2021/10/04/2021.10.04.463034, arXiv:https://www.biorxiv.org/content/early/2021/10/04/2021.10.04.463034.full.pdf, doi:10.1101/2021.10.04.463034.">Evans <em>et al.</em>, 2021</a>]</span>. Although included in the <strong>scipion-em-chimera</strong> plugin, the protocol <strong>chimerax-alphafold prediction</strong> allows to use, in addition to <em>ChimeraX</em> <a class="reference external" href="https://www.rbvi.ucsf.edu/chimerax/data/alphafold-nov2021/af_sbgrid.html">tools</a>, the <em>PHENIX</em> <a class="reference external" href="https://phenix-online.org/documentation/reference/alphafold.html">options</a> to access to Google Colab notebook system.</p>
<ul>
<li><div class="line-block">
<div class="line">Requirements to run this protocol and visualize results:</div>
</div>
<ul>
<li><div class="line-block">
<div class="line"><em>Scipion</em> plugin: <strong>scipion-em</strong></div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Scipion</em> plugin: <strong>scipion-em-chimera</strong></div>
</div>
</li>
</ul>
</li>
<li><div class="line-block">
<div class="line"><em>Scipion</em> menu: <em>Model building -&gt; Initial model</em> (<a class="reference internal" href="#model-building-app-protocol-alphafold-prediction-1"><span class="std std-numref">Fig. 1.1</span></a> (A))</div>
</div>
<div class="figure align-center" id="model-building-app-protocol-alphafold-prediction-1">
<a class="reference internal image-reference" href="../../../../../_images/Fig6_alphaFoldPrediction.svg"><img alt="Protocol **chimerax-alphafold prediction**. A: Protocol location in *Scipion* menu. B: Protocol form to retrieve AlphaFold2 structure predictions from EMDB (identical sequence). C: Protocol form to retrieve AlphaFold2 structure predictions from EMDB (homologous sequence) D: Protocol form to predict atomic structures using Google Colab notebooks to run AlphaFold2. E: Protocol form to predict atomic structures running AlphaFold2 in your own computer." src="../../../../../_images/Fig6_alphaFoldPrediction.svg" width="100.0%" /></a>
<p class="caption"><span class="caption-number">Fig. 1.1 </span><span class="caption-text">Protocol <strong>chimerax-alphafold prediction</strong>. A: Protocol location in <em>Scipion</em> menu. B: Protocol form to retrieve AlphaFold2 structure predictions from EMDB (identical sequence). C: Protocol form to retrieve AlphaFold2 structure predictions from EMDB (homologous sequence). D: Protocol form to predict atomic structures using Google Colab notebooks to run AlphaFold2. E: Protocol form to predict atomic structures running AlphaFold2 in your own computer.</span><a class="headerlink" href="#model-building-app-protocol-alphafold-prediction-1" title="Permalink to this image">¶</a></p>
</div>
</li>
<li><div class="line-block">
<div class="line">Protocol form parameters (<a class="reference internal" href="#model-building-app-protocol-alphafold-prediction-1"><span class="std std-numref">Fig. 1.1</span></a> (B, C, D, E)):</div>
</div>
<div class="line-block">
<div class="line"><em>Input</em> section:</div>
</div>
<ul>
<li><div class="line-block">
<div class="line"><em>Source to retrieve the AlphaFold2 model</em>:</div>
<div class="line">Click to select one of the next four options:</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line"><em>EBI Database (identical sequence)</em>: The <a class="reference external" href="https://alphafold.ebi.ac.uk/">AlphaFold Protein Structure Database</a> has been developed by DeepMind and EMBL-EBI with the aim of providing researchers with the AlphaFold2 predictions for the human proteome and other key organisms. They plan to extent the database to the majority of catalogued proteins. This work is still in progress and many proteins included as UniProt IDs have already been predicted and those predictions are freely available to the scientific community. To retrieve one of those protein predictions, write the alphanumerical code of the respective UniProt entry (<a class="reference internal" href="#model-building-app-protocol-alphafold-prediction-1"><span class="std std-numref">Fig. 1.1</span></a> (B):</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line"><em>UniProt name/ID</em></div>
</div>
<div class="line-block">
<div class="line">The protocol will fail if the written code doesn’t exist in UniProt DB or if  despite being a valid UniProt ID, the prediction has not been included in the <em>AlphaFold Protein Structure Database</em> yet.</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line"><em>EBI Database (homologous sequence)</em>: For all those cases in which the AlphaFold2 prediction hasn’t been included in the <em>AlphaFold Protein Structure Database</em> yet, the protocol <strong>chimerax-alphafold prediction</strong> gives the oportunity of retrieving the prediction performed for the most similar protein, identified through BLAST searching taking advantage of <em>ChimeraX</em>. Selecting this option you’ll be interrogated about (<a class="reference internal" href="#model-building-app-protocol-alphafold-prediction-1"><span class="std std-numref">Fig. 1.1</span></a> (C):</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line"><em>Reference sequence</em>: Protein sequence to perform the BLAST searching. This sequence has been previously imported to <em>Scipion</em> by using the protocol <a class="reference internal" href="../Homology/a090-importSequence.html#app-importsequence"><span class="std std-ref">Import sequence</span></a>.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>similarity-matrix</em>: Advanced param to select one of the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/blast/html/sub_matrix.html">substitution matrices</a> to assign a score to any couple of residues in the alignment.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>cutoff</em>: Advanced param to select the maximum statistic value required to include a retrieved element in the hit list.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Hide help popup window</em>: With this param you can choose between hide (<em>YES</em>) or show (<em>NO</em>) a help message to select and save the atomic structure using the <em>ChimeraX</em> command <a class="reference external" href="help:user/commands/scipionwrite.html">scipionwrite</a>. Selecting <em>YES</em> that help information will be shown also in red bold in the <em>ChimeraX</em> GUI (Log). Take into account that you might not retrieve any homologous structure. Check the box <em>Blast Protein Results</em> on the right. An empty box means no homologous structures retrieved for a particular substitution matrix and cutoff value.</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line"><em>Google  Colab</em>: In case no AlphaFold2 predictions can be retrieved from <em>EBI Database</em> (considering both identical and homologous sequences) the user has the opportunity to generate his/her own structure prediction taking advantage of two different Google Colaboratory notebooks that allows the user to run the program for free in the Google cloud.  Look at this <cite>video &lt;https://www.youtube.com/watch?v=inN8seMm7UI&gt;’_ for an introduction on Google Colaboratory. There are, nevertheless, some limitations in the computing resources that can be used. In particular, there is a daily limit (about 2 hours), the GPUs assigned by Google Colab might not have enough memory for predicting the structure of long sequences (higher than 800 amino acids approximately) and only one active session will be allowed. Compared with local AlphaFold, there are additional differences concerning the size of sequence databases since Google Colab uses reduced sequence databases and this could generate lower quality predictions. Selecting the option of retrieving structures from Google Colab the user will be interrogated about (:numref:`model_building_app_protocol_alphafold_prediction_1</cite> (D):</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line"><em>Colab Notebook</em>: Two different notebooks are available in this protocol, designed from <em>ChimeraX</em> and <em>PHENIX</em>:</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line"><em>Chimera21</em>: With almost no parameters, <em>ChimeraX</em> notebook runs AlphaFold2 with 5 alternative sets of params and retrieves five different models using <em>HMMER</em> for computing sequence alignments and searching in a reduced sequence database constituted by <em>UniRef90</em>, <em>SmallBFD</em> and <em>MGNIFY</em> sequence databases. For more information look at <a class="reference external" href="https://www.rbvi.ucsf.edu/chimerax/data/stanford-sep2021/alphafold.html">ChimeraX help</a> . Other params appear when the user selects the option <em>Chimera21</em>:</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line"><em>Skip Amber relaxation</em>: By default this notebook computes the final energy-minimization <em>Amber relaxation</em> step for the most confident prediction (from 5 structures retrieved) using the high-performance toolkit for molecular simulation <em>OpenMM</em> and the <em>Amber</em> forcefield. The option <em>Yes</em> has been selected by default in the protocol to save time and avoid failures due to insufficient computational resources to complete the job. However, to finish with the relaxation step, the user can also select the option <em>No</em> in this case.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Reference sequences</em>: Since the <em>ChimeraX</em> notebook gives the possibility of retrieving AlphaFold2 predictions from a monomer or a multimer the user should include the sequence of only one protein or several sequences covering all proteins in the multimer, respectively.</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line"><em>Phenix</em>: <em>PHENIX</em> notebook incorporates a higher number of parameters and the option of including several templates from <em>PDB</em> and a user’s template to improve a previous prediction according to the experimental map restrictions. Despite this advantage, only one model is retrieved running AlphaFold2 with the <em>PHENIX</em> notebook using <em>MMSEQS2</em> for calculating sequence alignments. Although <em>MMSEQS2</em> is very precise, it might find less hits compared to <em>HMMER</em>. Additional params of this option:</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line"><em>Reference sequence</em>: Include here only one sequence since the <em>PHENIX</em> notebook allows to retrieve predictions exclusively for monomers.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Use this template</em>: Include here the template that you would like to rebuilt an AlphaFold2 prediction. This template would probably corresponds to a previous structure prediction that has been modified according to the experimental map constraints. To encourage AlphaFold to use your rebuilt template select the option <code class="docutils literal notranslate"><span class="pre">skip_all_msa</span></code> (click this option at the bottom of the second cell <em>2. Options</em>). This way your template information and intrinsic structural information will be used.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Use templates from PDB</em>: Maximum number of templates used from PDB.</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line"><em>Test</em>: This third option is not a Google Colab notebook although it has been included here for debugging. <code class="docutils literal notranslate"><span class="pre">WARNING:</span></code> <em>Do not select this option</em>.</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line"><em>Show results in ChimeraX</em>: By default the <em>ChimeraX</em> GUI window will be opened to show the AlphaFold2 prediction structures. This gives the user the opportunity to modify any of the predictions and save this model using the command <a class="reference external" href="help:user/commands/scipionwrite.html">scipionwrite</a>. See also the <em>ChimeraX Operate</em> protocol <a class="reference external" href="https://scipion-em.github.io/docs/docs/user/tutorials/modelBuilding/Homology/a020-chimeraOperate.html">appendix</a>.</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line"><em>Local AlphaFold</em>: It is possible to run AlphaFold2 locally if you can fit to the requirements of hardware system configuration. In particular, you need a minimum storage capacity of 2.5 TB (recommended 3 TB) since the Alphafold database size is 2.2 TB (uncompressed), and a GPU since otherwise the computation will be very slow. GPU of 24 Gb is recommended to run predictions for sequences between 3,000 and 4,000 amino acids. The protocol has been tested for AlphaFold v2.1.1, able to predict multimers <span id="id4">[<a class="reference internal" href="../Homology/z010-references.html#id33" title="R. Evans, Michael O'Neill, Alexander Pritzel, Natasha Antropova, Andrew Senior, Tim Green, Augustin Žídek, Russ Bates, Sam Blackwell, Jason Yim, Olaf Ronneberger, Sebastian Bodenstein, Michal Zielinski, Alex Bridgland, Anna Potapenko, Andrew Cowie, Kathryn Tunyasuvunakool, Rishub Jain, Ellen Clancy, Pushmeet Kohli, John Jumper, and Demis Hassabis. Protein complex prediction with AlphaFold-Multimer. bioRxiv, 2021. URL: https://www.biorxiv.org/content/early/2021/10/04/2021.10.04.463034, arXiv:https://www.biorxiv.org/content/early/2021/10/04/2021.10.04.463034.full.pdf, doi:10.1101/2021.10.04.463034.">Evans <em>et al.</em>, 2021</a>]</span>. To learn more about software instalation and requirements follow this <a class="reference external" href="https://github.com/kalininalab/alphafold_non_docker">link</a>. Additional params with this option (<a class="reference internal" href="#model-building-app-protocol-alphafold-prediction-1"><span class="std std-numref">Fig. 1.1</span></a> (E):</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line"><em>Reference sequences</em>: Since AlphaFold2 is able to predict the structure of both monomers and multimers, the user should include here the protein sequence(s) contained in the particular complex.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Use Template until</em>: Option useful to avoid using as template a particular structure present in PDB. Then, the user should select a date before the release date of the structure. The date appears as <em>year-month-day</em>.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Is prokaryote?</em>: This option particularizes the searching of multimer predictions for prokaryotes. This option was removed in AlphaFold v2.2.0 to pair eukaryotes and prokaryotes in the same way.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>Use GPU acceleration?</em>: Although not mandatory, using a GPU accelerates very much the prediction process. By seleccting <em>Yes</em> an additional param will be shown:</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line"><em>Which GPUs to use</em>: Write here the GPU you want to use or a list of GPUs available (GPUs separated by commas). Otherwise the job itself will try to allocate available GPU resources.</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line"><em>Extra flags</em>: Include here additional arguments. Besides those indicated in the protocol help, look at <a class="reference external" href="https://github.com/deepmind/alphafold/blob/main/run_alphafold.py">run_alphafold.py</a> to seek the list of flags.</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
</ul>
</li>
<li><div class="line-block">
<div class="line">Protocol execution:</div>
</div>
<blockquote>
<div><div class="line-block">
<div class="line">Adding specific protocol label is recommended in <em>Run name</em>
section, at the form top. To add the label, open the protocol form,
press the pencil symbol at the right side of <em>Run name</em> box,
complete the label in the new opened window, press OK and, finally,
close the protocol. This label will be shown in the output summary
content (see below). If you want to run again this protocol, do not
forget to set to <em>Restart</em> the <em>Run mode</em>.</div>
</div>
<div class="line-block">
<div class="line">Press the <em>Execute</em> red button at the form bottom.</div>
</div>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line">Visualization of protocol results:</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line">Option <em>EBI Database (identical sequence)</em> (<a class="reference internal" href="#model-building-app-protocol-alphafold-prediction-1"><span class="std std-numref">Fig. 1.1</span></a> (B): The prediction requested from EBI will be save in the protocol and can be visualized in <em>ChimeraX</em> pressing <code class="docutils literal notranslate"><span class="pre">Analyze</span> <span class="pre">Results</span></code>. <em>ChimeraX</em> graphics window will be opened. Atomic structures are referred to the origin of coordinates in <em>ChimeraX</em>. To show the relative position of atomic structures, the three coordinate axes are represented; X axis (red), Y axis (yellow), and Z axis (blue) (<a class="reference internal" href="../Homology/a020-chimeraOperate.html#model-building-app-protocol-volume-3"><span class="std std-numref">Fig. 5.3</span></a>). Coordinate axes, predicted model and <em>Color key</em> of  AlphaFold2 confidence scores for individual residues (bfactor (<em>LDDT</em> values) palette of <em>ChimeraX</em>) are model numbers <em>#1</em>, <em>#2</em> and <em>#3</em>, respectively, in <em>ChimeraX Models</em> panel.</div>
</div>
</li>
<li><div class="line-block">
<div class="line">Option <em>EBI Database (homologous sequence)</em> (<a class="reference internal" href="#model-building-app-protocol-alphafold-prediction-1"><span class="std std-numref">Fig. 1.1</span></a> (C): After pressing the <em>Execute</em> button the <em>ChimeraX</em> GUI window will be opened. In case the user selected protocol options by default a pop up window appears with a help message to suggest the steps to follow to select a particular homologous sequence and save the retrieved structure. Pressing <em>OK</em> this window desappears and Blast results, if any has been found, are shown at the bottom rigth corner of <em>ChimeraX</em> GUI. Click on the selected line hit and the homologous structure will be retrieved to the main <em>ChimeraX</em> window. To save the homologous structure use the <em>ChimeraX</em> command <a class="reference external" href="help:user/commands/scipionwrite.html">scipionwrite</a>. After exiting <em>ChimeraX</em> GUI, the user can visualize the saved result as it has been indicated in the previous case.</div>
</div>
</li>
<li><div class="line-block">
<div class="line">Option <em>Google Colab</em> (<a class="reference internal" href="#model-building-app-protocol-alphafold-prediction-1"><span class="std std-numref">Fig. 1.1</span></a> (D): After pressing the <em>Execute</em> button a specific window browser will be opened especific both for <em>Chimera21</em> (<code class="docutils literal notranslate"><span class="pre">alphafold21_predict_colab.jpynb</span></code>) and <em>Phenix</em> (<code class="docutils literal notranslate"><span class="pre">AlphaFold2.jpynb</span></code>) notebooks.  Before starting try to be familiar with some basic functions to run and stop executions (Runtime options). In the case of <em>Phenix</em> notebook, besides the global running, there are several cells that can be executed sucesively and indepently. Options selected in each case should appear in each notebook. In this case (<em>Phenix</em> notebook), if the user selects the option <em>Use this template</em>, (s)he has to run independently each cell and click the option <em>Upload template</em> to effectively upload it (cell number 3). Failures are not uncommon and re-run processes or cells could be a good option if you have enough computational resources. After a while the process finishes and the prediction(s) will be directly in the <em>ChimeraX</em> GUI if the protocol option <em>Show results in ChimeraX</em> was selected. The <em>PAE</em> (predicted alignment error) plot will be opened at the same time. In the main window of <em>ChimeraX</em> the user has the possibility of modifying and saving the structure(s) with the <em>ChimeraX</em> command <a class="reference external" href="help:user/commands/scipionwrite.html">scipionwrite</a>. After exiting <em>ChimeraX</em> GUI, the user can visualize the saved result as it has been indicated in the previous case. In addition to the <em>ChimeraX</em> viewer window, the <em>PAE</em> plot will be opened, as well as the plot of <em>Number of aligned sequences with no gap for each residue position</em>.</div>
</div>
</li>
<li><div class="line-block">
<div class="line">Option <em>Local AlphaFold</em> (<a class="reference internal" href="#model-building-app-protocol-alphafold-prediction-1"><span class="std std-numref">Fig. 1.1</span></a> (E):</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line">Summary content:</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line">Protocol output (below framework):</div>
</div>
<blockquote>
<div><ul>
<li><div class="line-block">
<div class="line">Option <em>EBI Database (identical sequence)</em>:</div>
</div>
<blockquote>
<div><div class="line-block">
<div class="line"><em>chimerax - alphafold prediction -&gt; UniProt_ID</em>;</div>
</div>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line">Option <em>EBI Database (homologous sequence)</em>:</div>
</div>
<blockquote>
<div><div class="line-block">
<div class="line"><em>chimerax - alphafold prediction -&gt; Name of the structure saved by the user with the scipionwrite command</em>;</div>
</div>
</div></blockquote>
</li>
<li><div class="line-block">
<div class="line">Option <em>Google  Colab</em>:</div>
</div>
<blockquote>
<div><div class="line-block">
<div class="line"><em>chimerax - alphafold prediction -&gt; best_model</em>;</div>
<div class="line"><em>chimerax - alphafold prediction -&gt; best_model_pae</em>;</div>
</div>
</div></blockquote>
</li>
</ul>
</div></blockquote>
<div class="line-block">
<div class="line"><em>AtomStruct (pseudoatoms=True/ False, volume=True/ False)</em>.</div>
<div class="line">Pseudoatoms is set to <em>True</em> when the structure is made of
pseudoatoms instead of atoms.</div>
<div class="line">PAE</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><em>SUMMARY</em> box:</div>
<div class="line">PLDDTs (monomer) or iptm+ptm (multimer)</div>
<div class="line">alphafold ranking not yet computed</div>
</div>
</li>
</ul>
</div></blockquote>
</li>
</ul>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="a020-processPredictedProtocol.html" class="btn btn-neutral float-right" title="2. Phenix Process Predicted Model protocol" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="050-rebuildModel.html" class="btn btn-neutral float-left" title="9. Morph the predicted model onto the docked fragments and rebuild the missing parts of the model in the map density" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2022, Scipion team

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  
<div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
        <span class="fa fa-book"> Other Versions</span>
        v: release-3.0.0
        <span class="fa fa-caret-down"></span>
    </span>
    <div class="rst-other-versions">
        <dl>
            <dt>Branches</dt>
            <dd><a href="a010-alphafoldPredictionProtocol.html">release-3.0.0</a></dd>
        </dl>
    </div>
</div>


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>