

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>pwem.protocols.protocol_import.sequence &mdash; Scipion 3.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../../../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../../../../_static/jquery.js"></script>
        <script type="text/javascript" src="../../../../_static/underscore.js"></script>
        <script type="text/javascript" src="../../../../_static/doctools.js"></script>
        <script type="text/javascript" src="../../../../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../../../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/banner.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/custom.css" type="text/css" />
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../../index.html" class="icon icon-home"> Scipion
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Set up</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/scipion-modes/how-to-install.html">Installing Scipion v3.0</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/scipion-modes/scipion-configuration.html">Scipion configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/scipion-modes/host-configuration.html">Host configuration</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/user/troubleshooting.html">Troubleshooting</a></li>
</ul>
<p class="caption"><span class="caption-text">User documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/user/user-documentation.html">Graphical interface manuals</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/user/user-documentation.html#tutorials">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/user/user-documentation.html#processing-how-to-s">Processing How Toâ€™s</a></li>
</ul>
<p class="caption"><span class="caption-text">Facilities</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/facilities/facilities.html">On-the-fly processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/facilities/facilities-workflows.html">Streaming workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/facilities/customize-report.html">Summary Reports</a></li>
</ul>
<p class="caption"><span class="caption-text">Developer documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/developer/developers.html">Developers Page</a></li>
</ul>
<p class="caption"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/api/scipion-API.html">Scipion API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/api/plugins-API.html">Plugins API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/api/xmipp-API.html">Xmipp API</a></li>
</ul>
<p class="caption"><span class="caption-text">Get in touch</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../docs/misc/contact-us.html">Contact Us</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">Scipion</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../index.html">Docs</a> &raquo;</li>
        
          <li><a href="../../../index.html">Module code</a> &raquo;</li>
        
          <li><a href="../../../pwem.html">pwem</a> &raquo;</li>
        
      <li>pwem.protocols.protocol_import.sequence</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <h1>Source code for pwem.protocols.protocol_import.sequence</h1><div class="highlight"><pre>
<span></span><span class="c1"># -*- coding: utf-8 -*-</span>
<span class="c1"># **************************************************************************</span>
<span class="c1"># *</span>
<span class="c1"># * Authors:     Marta Martinez (mmmtnez@cnb.csic.es)</span>
<span class="c1"># *              Roberto Marabini (roberto@cnb.csic.es)</span>
<span class="c1"># *</span>
<span class="c1"># * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC</span>
<span class="c1"># *</span>
<span class="c1"># * This program is free software; you can redistribute it and/or modify</span>
<span class="c1"># * it under the terms of the GNU General Public License as published by</span>
<span class="c1"># * the Free Software Foundation; either version 3 of the License, or</span>
<span class="c1"># * (at your option) any later version.</span>
<span class="c1"># *</span>
<span class="c1"># * This program is distributed in the hope that it will be useful,</span>
<span class="c1"># * but WITHOUT ANY WARRANTY; without even the implied warranty of</span>
<span class="c1"># * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the</span>
<span class="c1"># * GNU General Public License for more details.</span>
<span class="c1"># *</span>
<span class="c1"># * You should have received a copy of the GNU General Public License</span>
<span class="c1"># * along with this program; if not, write to the Free Software</span>
<span class="c1"># * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA</span>
<span class="c1"># * 02111-1307  USA</span>
<span class="c1"># *</span>
<span class="c1"># *  All comments concerning this program package may be sent to the</span>
<span class="c1"># *  e-mail address &#39;scipion@cnb.csic.es&#39;</span>
<span class="c1"># *</span>
<span class="c1"># **************************************************************************</span>

<span class="kn">from</span> <span class="nn">os.path</span> <span class="k">import</span> <span class="n">exists</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">pyworkflow.protocol.params</span> <span class="k">as</span> <span class="nn">params</span>

<span class="kn">from</span> <span class="nn">pwem.objects.data</span> <span class="k">import</span> <span class="n">Sequence</span>
<span class="kn">import</span> <span class="nn">pwem.convert</span> <span class="k">as</span> <span class="nn">emconv</span>
<span class="kn">from</span> <span class="nn">pwem.convert</span> <span class="k">import</span> <span class="n">AtomicStructHandler</span>

<span class="kn">from</span> <span class="nn">.base</span> <span class="k">import</span> <span class="n">ProtImportFiles</span>


<div class="viewcode-block" id="ProtImportSequence"><a class="viewcode-back" href="../../../../api/pwem/pwem.protocols.protocol_import.sequence.html#pwem.protocols.protocol_import.sequence.ProtImportSequence">[docs]</a><span class="k">class</span> <span class="nc">ProtImportSequence</span><span class="p">(</span><span class="n">ProtImportFiles</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot; Protocol to import an aminoacid/nucleotide sequence file to the</span>
<span class="sd">    project&quot;&quot;&quot;</span>
    <span class="n">_label</span> <span class="o">=</span> <span class="s1">&#39;import sequence&#39;</span>
    <span class="c1"># SEQUENCEFILENAME = &#39;_sequence.fasta&#39;</span>
    <span class="n">IMPORT_FROM_PLAIN_TEXT</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">IMPORT_FROM_STRUCTURE</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="n">IMPORT_FROM_FILES</span> <span class="o">=</span> <span class="mi">2</span>
    <span class="n">IMPORT_FROM_UNIPROT</span> <span class="o">=</span> <span class="mi">3</span>
    <span class="n">IMPORT_FROM_NUCLEOTIDE_PLAIN_TEXT</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">IMPORT_FROM_NUCLEOTIDE_STRUCTURE</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="n">IMPORT_FROM_NUCLEOTIDE_FILES</span> <span class="o">=</span> <span class="mi">2</span>
    <span class="n">IMPORT_FROM_GENEBANK</span> <span class="o">=</span> <span class="mi">3</span>
    <span class="n">IMPORT_STRUCTURE_FROM_ID</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">IMPORT_STRUCTURE_FROM_FILES</span> <span class="o">=</span> <span class="mi">1</span>

    <span class="n">url</span> <span class="o">=</span> <span class="s2">&quot;http://www.uniprot.org/uniprot/&quot;</span>

    <span class="k">def</span> <span class="nf">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="o">**</span><span class="n">args</span><span class="p">):</span>
        <span class="n">ProtImportFiles</span><span class="o">.</span><span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="o">**</span><span class="n">args</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">_defineParams</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">form</span><span class="p">):</span>

        <span class="n">form</span><span class="o">.</span><span class="n">addSection</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="s1">&#39;Input&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputSequenceID&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">StringParam</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Sequence ID&quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Write a sequence ID. Otherwise, if the &quot;</span>
                           <span class="s2">&quot;sequence derives from GeneBank/UniProt/PDB &quot;</span>
                           <span class="s2">&quot;databases, the respective database ID will be &quot;</span>
                           <span class="s2">&quot;selected as starting sequence ID; examples: if &quot;</span>
                           <span class="s2">&quot;you select GeneBank accession AJ520101, SCIPION &quot;</span>
                           <span class="s2">&quot;will assign AJ520101 as sequence ID; if &quot;</span>
                           <span class="s2">&quot;you select UniProt accession P12345, SCIPION will &quot;</span>
                           <span class="s2">&quot;assign P12345 as sequence ID; if you &quot;</span>
                           <span class="s2">&quot;select atomic structure 3lqd.cif, chain B, &quot;</span>
                           <span class="s2">&quot;SCIPION will assign 3lqd_B as sequence ID. In &quot;</span>
                           <span class="s2">&quot;the rest of cases, the Sequence name &quot;</span>
                           <span class="s2">&quot;will be selected as starting Sequence ID.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputSequenceName&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">StringParam</span><span class="p">,</span> <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Sequence name&quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Write a sequence name.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputSequenceDescription&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">StringParam</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Sequence description&quot;</span><span class="p">,</span>
                      <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Write a description for your sequence. Otherwise, &quot;</span>
                           <span class="s2">&quot;if the &quot;</span>
                           <span class="s2">&quot;sequence derives from GeneBank/UniProt/PDB &quot;</span>
                           <span class="s2">&quot;databases, the respective database description &quot;</span>
                           <span class="s2">&quot;will be &quot;</span>
                           <span class="s2">&quot;selected as starting sequence description. In &quot;</span>
                           <span class="s2">&quot;the rest of cases, no sequence description will &quot;</span>
                           <span class="s2">&quot;be added.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputSequence&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="p">,</span>
                      <span class="n">pointerClass</span><span class="o">=</span><span class="s1">&#39;Sequence&#39;</span><span class="p">,</span>
                      <span class="n">choices</span><span class="o">=</span><span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE</span><span class="p">,</span>
                      <span class="n">display</span><span class="o">=</span><span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="o">.</span><span class="n">DISPLAY_HLIST</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Import sequence of &quot;</span><span class="p">,</span>
                      <span class="n">default</span><span class="o">=</span><span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Select the type of sequence to import.&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputProteinSequence&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="p">,</span>
                      <span class="n">choices</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;plain text&#39;</span><span class="p">,</span> <span class="s1">&#39;atomic structure&#39;</span><span class="p">,</span> <span class="s1">&#39;file&#39;</span><span class="p">,</span>
                               <span class="s1">&#39;UniProt ID&#39;</span><span class="p">],</span>
                      <span class="n">display</span><span class="o">=</span><span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="o">.</span><span class="n">DISPLAY_HLIST</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;inputSequence == </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;From &quot;</span><span class="p">,</span>
                      <span class="n">default</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_PLAIN_TEXT</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Select one of the four options: write the &#39;</span>
                           <span class="s1">&#39;aminoacid sequence or import it &#39;</span>
                           <span class="s1">&#39;from a previously loaded atomic structure, a local &#39;</span>
                           <span class="s1">&#39;file or an online server.&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;proteinIUPACalphabet&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="p">,</span>
                      <span class="n">choices</span><span class="o">=</span><span class="n">emconv</span><span class="o">.</span><span class="n">IUPAC_PROTEIN_ALPHABET</span><span class="p">,</span>
                      <span class="n">display</span><span class="o">=</span><span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="o">.</span><span class="n">DISPLAY_HLIST</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;inputSequence == </span><span class="si">%d</span><span class="s1"> and &#39;</span>
                                <span class="s1">&#39;inputProteinSequence == </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span>
                                <span class="p">(</span><span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">,</span>
                                 <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_PLAIN_TEXT</span><span class="p">),</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;IUPAC Protein alphabet: &quot;</span><span class="p">,</span>
                      <span class="n">default</span><span class="o">=</span><span class="n">emconv</span><span class="o">.</span><span class="n">EXTENDED_PROTEIN_ALPHABET</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Your raw sequence will be cleaned according &#39;</span>
                           <span class="s1">&#39;a certain alphabet, i.e., only the letters &#39;</span>
                           <span class="s1">&#39;contained in the alphabet will be maintained in &#39;</span>
                           <span class="s1">&#39;the sequence. Select thus the type of protein &#39;</span>
                           <span class="s1">&#39;alphabet in order to accomplish the &#39;</span>
                           <span class="s1">&#39;cleaning:</span><span class="se">\n\n</span><span class="s1">Protein alphabet: IUPAC protein &#39;</span>
                           <span class="s1">&#39;alphabet of the 20 standard amino acids; uppercase&#39;</span>
                           <span class="s1">&#39; and single letter: *ACDEFGHIKLMNPQRSTVWY*.</span><span class="se">\n\n</span><span class="s1">&#39;</span>
                           <span class="s1">&#39;Extended Protein alphabet: Extended uppercase &#39;</span>
                           <span class="s1">&#39;IUPAC &#39;</span>
                           <span class="s1">&#39;protein single letter alphabet including X etc.</span><span class="se">\n</span><span class="s1">&#39;</span>
                           <span class="s1">&#39;In addition to the standard 20 single letter &#39;</span>
                           <span class="s1">&#39;protein codes, this includes:</span><span class="se">\n</span><span class="s1">*B = Asx*; &#39;</span>
                           <span class="s1">&#39;Aspartic acid (R) or Asparagine (N)</span><span class="se">\n</span><span class="s1">*X = Xxx*&quot;; &#39;</span>
                           <span class="s1">&#39;Unknown or other amino acid</span><span class="se">\n</span><span class="s1">*Z = Glx*; Glutamic &#39;</span>
                           <span class="s1">&#39;acid (E) or Glutamine (Q)</span><span class="se">\n</span><span class="s1">*J = Xle*; Leucine (&#39;</span>
                           <span class="s1">&#39;L) or Isoleucine (I), used in mass-spec (NMR)</span><span class="se">\n</span><span class="s1">&#39;</span>
                           <span class="s1">&#39;*U = Sec*; Selenocysteine</span><span class="se">\n</span><span class="s1">*O = Pyl*; &#39;</span>
                           <span class="s1">&#39;Pyrrolysine</span><span class="se">\n</span><span class="s1">This alphabet is not intended to be &#39;</span>
                           <span class="s1">&#39;used with X for Selenocysteine (an ad-hoc standard&#39;</span>
                           <span class="s1">&#39; prior to the IUPAC adoption of U instead).</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;uniProtSequence&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">StringParam</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;inputSequence == </span><span class="si">%d</span><span class="s1"> and &#39;</span>
                                <span class="s1">&#39;inputProteinSequence == </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span>
                                <span class="p">(</span><span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">,</span>
                                 <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_UNIPROT</span><span class="p">),</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;UniProt name/ID &quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Write a UniProt ID (six or ten alphanumeric &#39;</span>
                           <span class="s1">&#39;characters; examples: A2BC19, P12345, &#39;</span>
                           <span class="s1">&#39;A0A022YWF9, DGAL_ECOLI).</span><span class="se">\n</span><span class="s1"> You can convert other &#39;</span>
                           <span class="s1">&#39;database identifiers to UniProt accession codes &#39;</span>
                           <span class="s1">&#39;by using the &quot;ID Mapping&quot; tab on &#39;</span>
                           <span class="s1">&#39;https://www.uniprot.org/&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputNucleotideSequence&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="p">,</span>
                      <span class="n">choices</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;plain text&#39;</span><span class="p">,</span> <span class="s1">&#39;atomic structure&#39;</span><span class="p">,</span> <span class="s1">&#39;file&#39;</span><span class="p">,</span>
                               <span class="s1">&#39;GeneBank&#39;</span><span class="p">],</span>
                      <span class="n">display</span><span class="o">=</span><span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="o">.</span><span class="n">DISPLAY_HLIST</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;inputSequence == </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_NUCLEOTIDES</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;From &quot;</span><span class="p">,</span>
                      <span class="n">default</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_PLAIN_TEXT</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Select one of the four options: write the &#39;</span>
                           <span class="s1">&#39;nucleic acid sequence or import it &#39;</span>
                           <span class="s1">&#39;from a local file or an online server.&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;nucleotideIUPACalphabet&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="p">,</span>
                      <span class="n">choices</span><span class="o">=</span><span class="n">emconv</span><span class="o">.</span><span class="n">IUPAC_NUCLEOTIDE_ALPHABET</span><span class="p">,</span>
                      <span class="n">display</span><span class="o">=</span><span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="o">.</span><span class="n">DISPLAY_HLIST</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;inputSequence == </span><span class="si">%d</span><span class="s1"> and &#39;</span>
                                <span class="s1">&#39;inputNucleotideSequence == </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span>
                                <span class="p">(</span><span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_NUCLEOTIDES</span><span class="p">,</span>
                                 <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_PLAIN_TEXT</span><span class="p">),</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;IUPAC Nucleic acid alphabet: &quot;</span><span class="p">,</span>
                      <span class="n">default</span><span class="o">=</span><span class="n">emconv</span><span class="o">.</span><span class="n">EXTENDED_DNA_ALPHABET</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Your raw sequence will be cleaned according &#39;</span>
                           <span class="s1">&#39;a certain alphabet, i.e., only the letters &#39;</span>
                           <span class="s1">&#39;contained in the alphabet will be maintained in &#39;</span>
                           <span class="s1">&#39;the sequence. Select thus the type of nucleic &#39;</span>
                           <span class="s1">&#39;acid alphabet in order to accomplish the &#39;</span>
                           <span class="s1">&#39;cleaning:</span><span class="se">\n\n</span><span class="s1"> Ambiguous DNA alphabet: Uppercase &#39;</span>
                           <span class="s1">&#39;IUPAC ambiguous DNA: *GATCRYWSMKHBVDN*.</span><span class="se">\n\n</span><span class="s1">&#39;</span>
                           <span class="s1">&#39;Unambiguous DNA alphabet: Uppercase IUPAC unambiguous DNA &#39;</span>
                           <span class="s1">&#39;(letters *GATC* only).</span><span class="se">\n\n</span><span class="s1">Extended DNA: Extended &#39;</span>
                           <span class="s1">&#39;IUPAC DNA alphabet.</span><span class="se">\n</span><span class="s1">In addition to the standard letter &#39;</span>
                           <span class="s1">&#39;codes GATC, this includes:</span><span class="se">\n</span><span class="s1">*B* = 5-bromouridine</span><span class="se">\n</span><span class="s1">&#39;</span>
                           <span class="s1">&#39;*D* = 5,6-dihydrouridine</span><span class="se">\n</span><span class="s1">*S* = thiouridine</span><span class="se">\n</span><span class="s1">*W* &#39;</span>
                           <span class="s1">&#39;= wyosine</span><span class="se">\n\n</span><span class="s1">Ambiguous RNA: Uppercase IUPAC &#39;</span>
                           <span class="s1">&#39;ambiguous RNA; *GAUCRYWSMKHBVDN*</span><span class="se">\n\n</span><span class="s1">Unambigous &#39;</span>
                           <span class="s1">&#39;RNA alphabet: Generic single letter RNA &#39;</span>
                           <span class="s1">&#39;alphabet.</span><span class="se">\n\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputRawSequence&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">StringParam</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;(inputSequence == </span><span class="si">%d</span><span class="s1"> and &#39;</span>
                                <span class="s1">&#39;inputProteinSequence == </span><span class="si">%d</span><span class="s1">) or &#39;</span>
                                <span class="s1">&#39;(inputSequence == </span><span class="si">%d</span><span class="s1"> and &#39;</span>
                                <span class="s1">&#39;inputNucleotideSequence == </span><span class="si">%d</span><span class="s1">) &#39;</span> <span class="o">%</span>
                                <span class="p">(</span><span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">,</span>
                                 <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_PLAIN_TEXT</span><span class="p">,</span>
                                 <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_NUCLEOTIDES</span><span class="p">,</span>
                                 <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_PLAIN_TEXT</span><span class="p">),</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Write your sequence here:&quot;</span><span class="p">,</span> <span class="n">important</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Write the aminoacid or nucleotide raw sequence.</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputStructureSequence&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="p">,</span>
                      <span class="n">choices</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;id&#39;</span><span class="p">,</span> <span class="s1">&#39;file&#39;</span><span class="p">],</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;inputProteinSequence == </span><span class="si">%d</span><span class="s1"> or &#39;</span>
                                <span class="s1">&#39;inputNucleotideSequence == </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span>
                                <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_STRUCTURE</span><span class="p">,</span>
                                 <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_STRUCTURE</span><span class="p">),</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Atomic structure from&quot;</span><span class="p">,</span>
                      <span class="n">default</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_STRUCTURE_FROM_ID</span><span class="p">,</span>
                      <span class="n">display</span><span class="o">=</span><span class="n">params</span><span class="o">.</span><span class="n">EnumParam</span><span class="o">.</span><span class="n">DISPLAY_HLIST</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Import structure data from online server or local &#39;</span>
                           <span class="s1">&#39;file&#39;</span><span class="p">,</span>
                      <span class="n">pointerClass</span><span class="o">=</span><span class="s1">&#39;AtomStruct&#39;</span><span class="p">,</span>
                      <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;pdbId&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">StringParam</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;(inputProteinSequence == </span><span class="si">%d</span><span class="s1"> or &#39;</span>
                                <span class="s1">&#39;inputNucleotideSequence == </span><span class="si">%d</span><span class="s1">) and &#39;</span>
                                <span class="s1">&#39;inputStructureSequence == </span><span class="si">%d</span><span class="s1">&#39;</span>
                                <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_STRUCTURE</span><span class="p">,</span>
                                   <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_STRUCTURE</span><span class="p">,</span>
                                   <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_STRUCTURE_FROM_ID</span><span class="p">),</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Atomic structure ID &quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Type a structure ID (four alphanumeric &#39;</span>
                           <span class="s1">&#39;characters).&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;pdbFile&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">PathParam</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;File path&quot;</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;(inputProteinSequence == </span><span class="si">%d</span><span class="s1"> or &#39;</span>
                                <span class="s1">&#39;inputNucleotideSequence == </span><span class="si">%d</span><span class="s1">) and &#39;</span>
                                <span class="s1">&#39;inputStructureSequence == </span><span class="si">%d</span><span class="s1">&#39;</span>
                                <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_STRUCTURE</span><span class="p">,</span>
                                   <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_STRUCTURE</span><span class="p">,</span>
                                   <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_STRUCTURE_FROM_FILES</span><span class="p">),</span>
                      <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Specify a path to desired atomic structure.&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;inputStructureChain&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">StringParam</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;inputProteinSequence == </span><span class="si">%d</span><span class="s1"> or &#39;</span>
                                <span class="s1">&#39;inputNucleotideSequence == </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span>
                                <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_STRUCTURE</span><span class="p">,</span>
                                 <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_STRUCTURE</span><span class="p">),</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Chain &quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s2">&quot;Select a particular chain of the atomic &quot;</span>
                           <span class="s2">&quot;structure.&quot;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;fileSequence&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">PathParam</span><span class="p">,</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;File path&quot;</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;inputProteinSequence == </span><span class="si">%d</span><span class="s1"> or &#39;</span>
                                <span class="s1">&#39;inputNucleotideSequence == </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span>
                                <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_FILES</span><span class="p">,</span>
                                 <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_FILES</span><span class="p">),</span>
                      <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Specify a path to desired aminoacid or &#39;</span>
                           <span class="s1">&#39;nucleic acid sequence &#39;</span>
                           <span class="s1">&#39;file.</span><span class="se">\n</span><span class="s1">If your file contains more than one &#39;</span>
                           <span class="s1">&#39;sequence, only the first one will be considered.&#39;</span><span class="p">)</span>
        <span class="n">form</span><span class="o">.</span><span class="n">addParam</span><span class="p">(</span><span class="s1">&#39;geneBankSequence&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">.</span><span class="n">StringParam</span><span class="p">,</span>
                      <span class="n">condition</span><span class="o">=</span><span class="s1">&#39;inputSequence == </span><span class="si">%d</span><span class="s1"> and &#39;</span>
                                <span class="s1">&#39;inputNucleotideSequence == </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span>
                                <span class="p">(</span><span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_NUCLEOTIDES</span><span class="p">,</span>
                                 <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_GENEBANK</span><span class="p">),</span>
                      <span class="n">label</span><span class="o">=</span><span class="s2">&quot;GeneBank accession &quot;</span><span class="p">,</span> <span class="n">allowsNull</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                      <span class="n">help</span><span class="o">=</span><span class="s1">&#39;Write a GeneBank accession.</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">_insertAllSteps</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceName</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>

        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequence</span> <span class="o">==</span> <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">:</span>
            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProteinSequence</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_PLAIN_TEXT</span><span class="p">:</span>
                <span class="n">rawSequence</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputRawSequence</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;getRawSequenceStep&#39;</span><span class="p">,</span> <span class="n">rawSequence</span><span class="p">)</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProteinSequence</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_STRUCTURE</span><span class="p">:</span>
                <span class="n">chainId</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputStructureChain</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;getSequenceOfChainStep&#39;</span><span class="p">,</span> <span class="n">chainId</span><span class="p">)</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProteinSequence</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_UNIPROT</span><span class="p">:</span>
                <span class="n">sequenceDB</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getUniProtID</span><span class="p">()</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;sequenceDatabaseDownloadStep&#39;</span><span class="p">,</span>
                                         <span class="n">sequenceDB</span><span class="p">)</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProteinSequence</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_FILES</span><span class="p">:</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">sequenceFile</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">fileSequence</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
                <span class="n">sequenceFile</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">sequenceFile</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;fileDownloadStep&#39;</span><span class="p">,</span> <span class="n">sequenceFile</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputNucleotideSequence</span> <span class="o">==</span> \
                    <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_PLAIN_TEXT</span><span class="p">:</span>
                <span class="n">rawSequence</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputRawSequence</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;getRawSequenceStep&#39;</span><span class="p">,</span> <span class="n">rawSequence</span><span class="p">)</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputNucleotideSequence</span> <span class="o">==</span> \
                    <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_STRUCTURE</span><span class="p">:</span>
                <span class="n">chainId</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputStructureChain</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;getSequenceOfChainStep&#39;</span><span class="p">,</span> <span class="n">chainId</span><span class="p">)</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputNucleotideSequence</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_GENEBANK</span><span class="p">:</span>
                <span class="n">sequenceDB</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getGeneBankID</span><span class="p">()</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;sequenceDatabaseDownloadStep&#39;</span><span class="p">,</span>
                                         <span class="n">sequenceDB</span><span class="p">)</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputNucleotideSequence</span> <span class="o">==</span> \
                    <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_FILES</span><span class="p">:</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">sequenceFile</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">fileSequence</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
                <span class="n">sequenceFile</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">sequenceFile</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;fileDownloadStep&#39;</span><span class="p">,</span> <span class="n">sequenceFile</span><span class="p">)</span>

        <span class="bp">self</span><span class="o">.</span><span class="n">_insertFunctionStep</span><span class="p">(</span><span class="s1">&#39;createOutputStep&#39;</span><span class="p">)</span>

<div class="viewcode-block" id="ProtImportSequence.getRawSequenceStep"><a class="viewcode-back" href="../../../../api/pwem/pwem.protocols.protocol_import.sequence.html#pwem.protocols.protocol_import.sequence.ProtImportSequence.getRawSequenceStep">[docs]</a>    <span class="k">def</span> <span class="nf">getRawSequenceStep</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">rawSequence</span><span class="p">):</span>
        <span class="c1"># user types sequence</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceID</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceID</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">name</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">alphabet</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getAlphabet</span><span class="p">()</span>  <span class="c1"># index number</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">sequence</span> <span class="o">=</span> <span class="n">emconv</span><span class="o">.</span><span class="n">cleanSequenceScipion</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">inputSequence</span> <span class="o">==</span>
                                                    <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">,</span>
                                                    <span class="bp">self</span><span class="o">.</span><span class="n">alphabet</span><span class="p">,</span> <span class="n">rawSequence</span><span class="p">)</span></div>

<div class="viewcode-block" id="ProtImportSequence.getSequenceOfChainStep"><a class="viewcode-back" href="../../../../api/pwem/pwem.protocols.protocol_import.sequence.html#pwem.protocols.protocol_import.sequence.ProtImportSequence.getSequenceOfChainStep">[docs]</a>    <span class="k">def</span> <span class="nf">getSequenceOfChainStep</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">chainId</span><span class="p">):</span>
        <span class="c1"># sequece is obtained from PDB file</span>

        <span class="c1"># form has a wizard that creates label with the format</span>
        <span class="c1"># [model: x, chain: x, xxx residues]</span>
        <span class="kn">import</span> <span class="nn">json</span>
        <span class="n">chainIdDict</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">loads</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">inputStructureChain</span><span class="o">.</span><span class="n">get</span><span class="p">())</span>

        <span class="n">selectedModel</span> <span class="o">=</span> <span class="n">chainIdDict</span><span class="p">[</span><span class="s1">&#39;model&#39;</span><span class="p">]</span>
        <span class="n">selectedChain</span> <span class="o">=</span> <span class="n">chainIdDict</span><span class="p">[</span><span class="s1">&#39;chain&#39;</span><span class="p">]</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span> <span class="o">=</span> <span class="n">AtomicStructHandler</span><span class="p">()</span>

        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">pdbId</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="c1"># PDB from remote database</span>
            <span class="n">pdbID</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">pdbId</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
            <span class="n">tmpFilePath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="s2">&quot;/tmp&quot;</span><span class="p">,</span> <span class="n">pdbID</span> <span class="o">+</span> <span class="s2">&quot;.cif&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span>
            <span class="k">if</span> <span class="n">exists</span><span class="p">(</span><span class="n">tmpFilePath</span><span class="p">):</span>
                <span class="c1"># wizard already downloaded the file</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">tmpFilePath</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="c1"># wizard has not used and the file has not been downloaded yet</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span><span class="o">.</span><span class="n">readFromPDBDatabase</span><span class="p">(</span><span class="n">pdbID</span><span class="p">,</span> <span class="nb">dir</span><span class="o">=</span><span class="s2">&quot;/tmp&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="c1"># PDB from file</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">pdbFile</span><span class="o">.</span><span class="n">get</span><span class="p">())</span>

        <span class="n">_sequence</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span><span class="o">.</span><span class="n">getSequenceFromChain</span><span class="p">(</span>
            <span class="n">selectedModel</span><span class="p">,</span> <span class="n">selectedChain</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">sequence</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">_sequence</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">alphabet</span> <span class="o">=</span> <span class="n">emconv</span><span class="o">.</span><span class="n">alphabetToIndex</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">inputSequence</span> <span class="o">==</span>
                                               <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">,</span>
                                               <span class="n">_sequence</span><span class="o">.</span><span class="n">alphabet</span><span class="p">)</span>

        <span class="c1"># Assignation of sequence ID: if the user has provided a specific</span>
        <span class="c1">#  ID, this will be adopted by default; otherwise, a sequence ID</span>
        <span class="c1"># related with the starting structure will be selected.</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceID</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceID</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span><span class="o">.</span><span class="n">getFullID</span><span class="p">(</span>
                <span class="n">selectedModel</span><span class="p">,</span> <span class="n">selectedChain</span><span class="p">)</span>

        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Selected chain: </span><span class="si">%s</span><span class="s2"> from model: </span><span class="si">%s</span><span class="s2"> from structure: </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span>
              <span class="p">(</span><span class="n">selectedChain</span><span class="p">,</span> <span class="n">selectedModel</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">structureHandler</span><span class="o">.</span><span class="n">structure</span><span class="o">.</span><span class="n">get_id</span><span class="p">()))</span></div>

<div class="viewcode-block" id="ProtImportSequence.sequenceDatabaseDownloadStep"><a class="viewcode-back" href="../../../../api/pwem/pwem.protocols.protocol_import.sequence.html#pwem.protocols.protocol_import.sequence.ProtImportSequence.sequenceDatabaseDownloadStep">[docs]</a>    <span class="k">def</span> <span class="nf">sequenceDatabaseDownloadStep</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">sequenceDB</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;Download UniProt/GeneBank sequence from its respective database</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="c1"># sequenceDB = str(sequenceDB)</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">uniProtSequence</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">seqHandler</span> <span class="o">=</span> <span class="n">emconv</span><span class="o">.</span><span class="n">SequenceHandler</span><span class="p">()</span>

        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getGeneBankID</span><span class="p">()</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">seqHandler</span> <span class="o">=</span> <span class="n">emconv</span><span class="o">.</span><span class="n">SequenceHandler</span><span class="p">(</span><span class="n">isAminoacid</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

        <span class="n">record</span><span class="p">,</span> <span class="n">error</span> <span class="o">=</span> <span class="n">seqHandler</span><span class="o">.</span><span class="n">downloadSeqFromDatabase</span><span class="p">(</span><span class="n">sequenceDB</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">record</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Error: &quot;</span><span class="p">,</span> <span class="n">error</span><span class="p">)</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">setAborted</span><span class="p">()</span>
            <span class="n">exit</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>

        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceID</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceID</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
        <span class="k">elif</span> <span class="n">sequenceDB</span> <span class="o">!=</span> <span class="s1">&#39;&#39;</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="n">sequenceDB</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">name</span>
        <span class="k">if</span> <span class="n">record</span><span class="o">.</span><span class="n">description</span> <span class="o">!=</span> <span class="s1">&#39;&#39;</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">description</span> <span class="o">=</span> <span class="n">record</span><span class="o">.</span><span class="n">description</span>

        <span class="bp">self</span><span class="o">.</span><span class="n">sequence</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">seq</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">alphabet</span> <span class="o">=</span> <span class="n">emconv</span><span class="o">.</span><span class="n">alphabetToIndex</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">inputSequence</span> <span class="o">==</span>
                                               <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">,</span> <span class="n">record</span><span class="o">.</span><span class="n">seq</span><span class="o">.</span><span class="n">alphabet</span><span class="p">)</span></div>

<div class="viewcode-block" id="ProtImportSequence.fileDownloadStep"><a class="viewcode-back" href="../../../../api/pwem/pwem.protocols.protocol_import.sequence.html#pwem.protocols.protocol_import.sequence.ProtImportSequence.fileDownloadStep">[docs]</a>    <span class="k">def</span> <span class="nf">fileDownloadStep</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">sequenceFile</span><span class="p">):</span>
        <span class="c1"># If sequencePath contains more than one sequence, only</span>
        <span class="c1"># the first one will be considered</span>
        <span class="n">seqHandler</span> <span class="o">=</span> <span class="n">emconv</span><span class="o">.</span><span class="n">SequenceHandler</span><span class="p">()</span>
        <span class="n">record</span> <span class="o">=</span> <span class="n">seqHandler</span><span class="o">.</span><span class="n">downloadSeqFromFile</span><span class="p">(</span><span class="n">sequenceFile</span><span class="p">,</span>
                                                <span class="nb">type</span><span class="o">=</span><span class="s2">&quot;fasta&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceID</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceID</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
        <span class="k">elif</span> <span class="n">record</span><span class="o">.</span><span class="n">id</span> <span class="o">!=</span> <span class="s1">&#39;&#39;</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="n">record</span><span class="o">.</span><span class="n">id</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">name</span>
        <span class="k">if</span> <span class="n">record</span><span class="o">.</span><span class="n">description</span> <span class="o">!=</span> <span class="s1">&#39;&#39;</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">description</span> <span class="o">=</span> <span class="n">record</span><span class="o">.</span><span class="n">description</span>

        <span class="bp">self</span><span class="o">.</span><span class="n">sequence</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">seq</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">alphabet</span> <span class="o">=</span> <span class="n">emconv</span><span class="o">.</span><span class="n">alphabetToIndex</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">inputSequence</span> <span class="o">==</span>
                                               <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">,</span> <span class="n">record</span><span class="o">.</span><span class="n">seq</span><span class="o">.</span><span class="n">alphabet</span><span class="p">)</span></div>

<div class="viewcode-block" id="ProtImportSequence.createOutputStep"><a class="viewcode-back" href="../../../../api/pwem/pwem.protocols.protocol_import.sequence.html#pwem.protocols.protocol_import.sequence.ProtImportSequence.createOutputStep">[docs]</a>    <span class="k">def</span> <span class="nf">createOutputStep</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot; Register the output object. &quot;&quot;&quot;</span>

        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceDescription</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">description</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceDescription</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
        <span class="k">elif</span> <span class="nb">hasattr</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="s1">&#39;description&#39;</span><span class="p">):</span>
            <span class="k">pass</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">description</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>

        <span class="n">seq</span> <span class="o">=</span> <span class="n">Sequence</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span>
                       <span class="n">sequence</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">sequence</span><span class="p">,</span>
                       <span class="n">alphabet</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">alphabet</span><span class="p">,</span>
                       <span class="n">isAminoacids</span><span class="o">=</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">inputSequence</span> <span class="o">==</span>
                                     <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">),</span>
                       <span class="nb">id</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="n">description</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">description</span><span class="p">)</span>
        <span class="n">outputs</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;outputSequence&#39;</span><span class="p">:</span> <span class="n">seq</span><span class="p">}</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_defineOutputs</span><span class="p">(</span><span class="o">**</span><span class="n">outputs</span><span class="p">)</span></div>

    <span class="k">def</span> <span class="nf">_summary</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="n">summary</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequenceName</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
        <span class="n">uniProtId</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getUniProtID</span><span class="p">()</span>
        <span class="n">geneBankID</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getGeneBankID</span><span class="p">()</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequence</span> <span class="o">==</span> <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">:</span>
            <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;Sequence of aminoacids:</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProteinSequence</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_PLAIN_TEXT</span><span class="p">:</span>
                <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Sequence *</span><span class="si">%s</span><span class="s2">* imported from plain text</span><span class="se">\n</span><span class="s2">&quot;</span>
                               <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">)</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProteinSequence</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_STRUCTURE</span><span class="p">:</span>
                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputStructureSequence</span> <span class="o">==</span> \
                        <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_STRUCTURE_FROM_ID</span><span class="p">:</span>
                    <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Sequence *</span><span class="si">%s</span><span class="s2">* imported from atomic &quot;</span>
                                   <span class="s2">&quot;structure *</span><span class="si">%s</span><span class="s2">.cif*</span><span class="se">\n</span><span class="s2">&quot;</span>
                                   <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">pdbId</span><span class="o">.</span><span class="n">get</span><span class="p">()))</span>
                <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputStructureSequence</span> <span class="o">==</span> \
                        <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_STRUCTURE_FROM_FILES</span><span class="p">:</span>
                    <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Sequence *</span><span class="si">%s</span><span class="s2">* imported from file *</span><span class="si">%s</span><span class="s2">*</span><span class="se">\n</span><span class="s2">&quot;</span>
                                   <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">pdbFile</span><span class="o">.</span><span class="n">get</span><span class="p">()))</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProteinSequence</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_UNIPROT</span><span class="p">:</span>
                <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Sequence *</span><span class="si">%s</span><span class="s2">* imported from UniProt ID &quot;</span>
                               <span class="s2">&quot;*</span><span class="si">%s</span><span class="s2">*</span><span class="se">\n</span><span class="s2">&quot;</span>
                               <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="n">uniProtId</span><span class="p">))</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputProteinSequence</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_FILES</span><span class="p">:</span>
                <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Sequence *</span><span class="si">%s</span><span class="s2">* imported from file name: &quot;</span>
                               <span class="s2">&quot;*</span><span class="si">%s</span><span class="s2">*</span><span class="se">\n</span><span class="s2">&quot;</span>
                               <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">fileSequence</span><span class="o">.</span><span class="n">get</span><span class="p">()))</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;Sequence of nucleotides:</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputNucleotideSequence</span> <span class="o">==</span> \
                    <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_PLAIN_TEXT</span><span class="p">:</span>
                <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Sequence *</span><span class="si">%s</span><span class="s2">* imported from plain text</span><span class="se">\n</span><span class="s2">&quot;</span>
                               <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">)</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputNucleotideSequence</span> <span class="o">==</span> \
                    <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_STRUCTURE</span><span class="p">:</span>
                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputStructureSequence</span> <span class="o">==</span> \
                        <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_STRUCTURE_FROM_ID</span><span class="p">:</span>
                    <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Sequence *</span><span class="si">%s</span><span class="s2">* imported from atomic &quot;</span>
                                   <span class="s2">&quot;structure *</span><span class="si">%s</span><span class="s2">.cif*</span><span class="se">\n</span><span class="s2">&quot;</span>
                                   <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">pdbId</span><span class="o">.</span><span class="n">get</span><span class="p">()))</span>
                <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputStructureSequence</span> <span class="o">==</span> \
                        <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_STRUCTURE_FROM_FILES</span><span class="p">:</span>
                    <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Sequence *</span><span class="si">%s</span><span class="s2">* imported from file *</span><span class="si">%s</span><span class="s2">*</span><span class="se">\n</span><span class="s2">&quot;</span>
                                   <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">pdbFile</span><span class="o">.</span><span class="n">get</span><span class="p">()))</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputNucleotideSequence</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_GENEBANK</span><span class="p">:</span>
                <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Sequence *</span><span class="si">%s</span><span class="s2">* imported from geneBank ID &quot;</span>
                               <span class="s2">&quot;*</span><span class="si">%s</span><span class="s2">*</span><span class="se">\n</span><span class="s2">&quot;</span>
                               <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="n">geneBankID</span><span class="p">))</span>
            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputNucleotideSequence</span> <span class="o">==</span> \
                    <span class="bp">self</span><span class="o">.</span><span class="n">IMPORT_FROM_NUCLEOTIDE_FILES</span><span class="p">:</span>
                <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;Sequence *</span><span class="si">%s</span><span class="s2">* imported from file name: &quot;</span>
                               <span class="s2">&quot;*</span><span class="si">%s</span><span class="s2">*</span><span class="se">\n</span><span class="s2">&quot;</span>
                               <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">fileSequence</span><span class="o">.</span><span class="n">get</span><span class="p">()))</span>
        <span class="k">return</span> <span class="n">summary</span>

    <span class="k">def</span> <span class="nf">_validate</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="n">errors</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">return</span> <span class="n">errors</span>

    <span class="k">def</span> <span class="nf">_getSequenceName</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="k">pass</span>

    <span class="k">def</span> <span class="nf">_getUniProtID</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">uniProtSequence</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>

    <span class="k">def</span> <span class="nf">_getGeneBankID</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">geneBankSequence</span>

    <span class="k">def</span> <span class="nf">_getAlphabet</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">inputSequence</span> <span class="o">==</span> <span class="n">emconv</span><span class="o">.</span><span class="n">SEQ_TYPE_AMINOACIDS</span><span class="p">:</span>
            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">proteinIUPACalphabet</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">nucleotideIUPACalphabet</span><span class="o">.</span><span class="n">get</span><span class="p">()</span></div>
</pre></div>

           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Scipion team

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  
<div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
        <span class="fa fa-book"> Other Versions</span>
        v: vtune
        <span class="fa fa-caret-down"></span>
    </span>
    <div class="rst-other-versions">
        <dl>
            <dt>Branches</dt>
            <dd><a href="../../../../../dependabot_pip_jinja2-2.10.1/index.html">dependabot/pip/jinja2-2.10.1</a></dd>
            <dd><a href="../../../../../gh-pages/index.html">gh-pages</a></dd>
            <dd><a href="../../../../../release-2.0.0/index.html">release-2.0.0</a></dd>
            <dd><a href="../../../../../release-3.0.0/index.html">release-3.0.0</a></dd>
            <dd><a href="../../../../../rsanchezgarc-patch-1/index.html">rsanchezgarc-patch-1</a></dd>
            <dd><a href="sequence.html">vtune</a></dd>
        </dl>
    </div>
</div>


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>